// @nylas/web-elements/dist/cdn/google-logo/google-logo.es.js
const T = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let E, M, A, oe = !1, B = !1, W = !1, v = !1, G = null, D = !1;
const x = (e, t = "") => () => {
}, Se = "slot-fb{display:contents}slot-fb[hidden]{display:none}", I = "http://www.w3.org/1999/xlink", Q = {}, ke = "http://www.w3.org/2000/svg", me = "http://www.w3.org/1999/xhtml", be = (e) => e != null, z = (e) => (e = typeof e, e === "object" || e === "function");
function Te(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const S = (e, t, ...n) => {
  let s = null, l = null, c = null, o = !1, r = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof e != "function" && !z(s)) && (s = String(s)), o && r ? i[i.length - 1].$text$ += s : i.push(o ? H(null, s) : s), r = o);
  };
  if (u(n), t) {
    t.key && (l = t.key), t.name && (c = t.name);
    {
      const a = t.className || t.class;
      a && (t.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, Ee);
  const f = H(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = c, f;
}, H = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Le = {}, xe = (e) => e && e.$tag$ === Le, Ee = {
  forEach: (e, t) => e.map(X).forEach(t),
  map: (e, t) => e.map(X).map(t).map(Ce)
}, X = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), Ce = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), S(e.vtag, n, ...e.vchildren || []);
  }
  const t = H(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, _e = (e) => tt.map((t) => t(e)).find((t) => !!t), Ae = (e, t) => e != null && !z(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, Z = /* @__PURE__ */ new WeakMap(), Re = (e, t, n) => {
  let s = U.get(e);
  ot && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, U.set(e, s);
}, Oe = (e, t, n) => {
  var s;
  const l = ie(t, n), c = U.get(l);
  if (e = e.nodeType === 11 ? e : k, c)
    if (typeof c == "string") {
      e = e.head || e;
      let o = Z.get(e), r;
      if (o || Z.set(e, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          r = k.createElement("style"), r.innerHTML = c;
          const i = (s = g.$nonce$) !== null && s !== void 0 ? s : Te(k);
          i != null && r.setAttribute("nonce", i), e.insertBefore(r, e.querySelector("link"));
        }
        t.$flags$ & 4 && (r.innerHTML += Se), o && o.add(l);
      }
    } else e.adoptedStyleSheets.includes(c) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, c]);
  return l;
}, we = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, l = x("attachStyles", t.$tagName$), c = Oe(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = c, n.classList.add(c + "-h"), s & 2 && n.classList.add(c + "-s")), l();
}, ie = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), K = (e, t, n, s, l, c) => {
  if (n !== s) {
    let o = te(e, t), r = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = J(n), f = J(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(j, r) ? t = r.slice(2) : t = r[2] + t.slice(3), n || s) {
          const i = t.endsWith(ce);
          t = t.replace(Be, ""), n && g.rel(e, t, n, i), s && g.ael(e, t, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? o = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        r !== (r = r.replace(/^xlink\:?/, "")) && (t = r, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(I, t) : e.removeAttribute(t)) : (!o || c & 4 || l) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(I, t, s) : e.setAttribute(t, s));
      }
  }
}, Fe = /\s/, J = (e) => e ? e.split(Fe) : [], ce = "Capture", Be = new RegExp(ce + "$"), re = (e, t, n, s) => {
  const l = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, c = e && e.$attrs$ || Q, o = t.$attrs$ || Q;
  for (s in c)
    s in o || K(l, s, c[s], void 0, n, t.$flags$);
  for (s in o)
    K(l, s, c[s], o[s], n, t.$flags$);
}, P = (e, t, n, s) => {
  const l = t.$children$[n];
  let c = 0, o, r, i;
  if (oe || (W = !0, l.$tag$ === "slot" && (E && s.classList.add(E + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = k.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = k.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = k.createElementNS(v ? ke : me, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), re(null, l, v), be(E) && o["s-si"] !== E && o.classList.add(o["s-si"] = E), l.$children$)
      for (c = 0; c < l.$children$.length; ++c)
        r = P(e, l, c, o), r && o.appendChild(r);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = A, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = M, o["s-sn"] = l.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === l.$tag$ && e.$elm$ && R(e.$elm$, !1)), o;
}, R = (e, t) => {
  g.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== A && l["s-ol"] && (fe(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), t && R(l, t);
  }
  g.$flags$ &= -2;
}, ae = (e, t, n, s, l, c) => {
  let o = e["s-cr"] && e["s-cr"].parentNode || e, r;
  for (o.shadowRoot && o.tagName === A && (o = o.shadowRoot); l <= c; ++l)
    s[l] && (r = P(null, n, l, e), r && (s[l].$elm$ = r, o.insertBefore(r, q(t))));
}, $e = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const l = e[s];
    if (l) {
      const c = l.$elm$;
      he(l), c && (B = !0, c["s-ol"] ? c["s-ol"].remove() : R(c, !0), c.remove());
    }
  }
}, He = (e, t, n, s, l = !1) => {
  let c = 0, o = 0, r = 0, i = 0, u = t.length - 1, f = t[0], a = t[u], d = s.length - 1, $ = s[0], h = s[d], y, p;
  for (; c <= u && o <= d; )
    if (f == null)
      f = t[++c];
    else if (a == null)
      a = t[--u];
    else if ($ == null)
      $ = s[++o];
    else if (h == null)
      h = s[--d];
    else if (F(f, $, l))
      C(f, $, l), f = t[++c], $ = s[++o];
    else if (F(a, h, l))
      C(a, h, l), a = t[--u], h = s[--d];
    else if (F(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && R(f.$elm$.parentNode, !1), C(f, h, l), e.insertBefore(f.$elm$, a.$elm$.nextSibling), f = t[++c], h = s[--d];
    else if (F(a, $, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && R(a.$elm$.parentNode, !1), C(a, $, l), e.insertBefore(a.$elm$, f.$elm$), a = t[--u], $ = s[++o];
    else {
      for (r = -1, i = c; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === $.$key$) {
          r = i;
          break;
        }
      r >= 0 ? (p = t[r], p.$tag$ !== $.$tag$ ? y = P(t && t[o], n, r, e) : (C(p, $, l), t[r] = void 0, y = p.$elm$), $ = s[++o]) : (y = P(t && t[o], n, o, e), $ = s[++o]), y && fe(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  c > u ? ae(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && $e(t, c, u);
}, F = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, q = (e) => e && e["s-ol"] || e, fe = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, C = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, l = e.$children$, c = t.$children$, o = t.$tag$, r = t.$text$;
  let i;
  r === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || re(e, t, v), l !== null && c !== null ? He(s, l, t, c, n) : c !== null ? (e.$text$ !== null && (s.textContent = ""), ae(s, null, t, c, 0, c.length - 1)) : l !== null && $e(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = r : e.$text$ !== r && (s.data = r);
}, de = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of t)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      de(n);
    }
}, m = [], ue = (e) => {
  let t, n, s;
  for (const l of e.childNodes) {
    if (l["s-sr"] && (t = l["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const c = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== l["s-hn"] && !T.experimentalSlotFixes)
          if (Y(t, c)) {
            let o = m.find((r) => r.$nodeToRelocate$ === t);
            B = !0, t["s-sn"] = t["s-sn"] || c, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (t["s-sh"] = l["s-hn"], m.push({
              $slotRefNode$: l,
              $nodeToRelocate$: t
            })), t["s-sr"] && m.map((r) => {
              Y(r.$nodeToRelocate$, t["s-sn"]) && (o = m.find((i) => i.$nodeToRelocate$ === t), o && !r.$slotRefNode$ && (r.$slotRefNode$ = o.$slotRefNode$));
            });
          } else m.some((o) => o.$nodeToRelocate$ === t) || m.push({
            $nodeToRelocate$: t
          });
    }
    l.nodeType === 1 && ue(l);
  }
}, Y = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", he = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(he);
}, Pe = (e, t, n = !1) => {
  var s, l, c, o, r;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || H(null, null), a = xe(t) ? t : S(null, null, t);
  if (A = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, e.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, E = i["s-sc"], M = i["s-cr"], oe = (u.$flags$ & 1) !== 0, B = !1, C(f, a, n);
  {
    if (g.$flags$ |= 1, W) {
      ue(a.$elm$);
      for (const d of m) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = k.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of m) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let p = h.nextSibling;
          {
            let L = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; L; ) {
              let b = (l = L["s-nr"]) !== null && l !== void 0 ? l : null;
              if (b && b["s-sn"] === $["s-sn"] && y === b.parentNode && (b = b.nextSibling, !b || !b["s-nr"])) {
                p = b;
                break;
              }
              L = L.previousSibling;
            }
          }
          (!p && y !== $.parentNode || $.nextSibling !== p) && $ !== p && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), y.insertBefore($, p), $.nodeType === 1 && ($.hidden = (c = $["s-ih"]) !== null && c !== void 0 ? c : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (o = $.hidden) !== null && o !== void 0 ? o : !1), $.hidden = !0);
      }
    }
    B && de(a.$elm$), g.$flags$ &= -2, m.length = 0;
  }
  if (T.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== A && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (r = d.hidden) !== null && r !== void 0 ? r : !1), d.hidden = !0);
  M = void 0;
}, Ue = (e, t) => {
}, ge = (e, t) => (e.$flags$ |= 16, Ue(e, e.$ancestorComponent$), rt(() => je(e, t))), je = (e, t) => {
  const n = e.$hostElement$, s = x("scheduleUpdate", e.$cmpMeta$.$tagName$), l = n;
  let c;
  return t ? c = _(l, "componentWillLoad") : c = _(l, "componentWillUpdate"), c = V(c, () => _(l, "componentWillRender")), s(), V(c, () => De(e, l, t));
}, V = (e, t) => Me(e) ? e.then(t) : t(), Me = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", De = async (e, t, n) => {
  const s = e.$hostElement$, l = x("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && we(e);
  const c = x("render", e.$cmpMeta$.$tagName$);
  Ne(e, t, s, n), c(), l(), We(e);
}, Ne = (e, t, n, s) => {
  try {
    G = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (T.hasRenderFn || T.reflect) && (T.vdomRender || T.reflect) && (T.hydrateServerSide || Pe(e, t, s));
  } catch (i) {
    w(i, e.$hostElement$);
  }
  return G = null, null;
}, We = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = x("postUpdate", t), l = n;
  e.$ancestorComponent$, _(l, "componentDidRender"), e.$flags$ & 64 ? (_(l, "componentDidUpdate"), s()) : (e.$flags$ |= 64, _(l, "componentDidLoad"), s());
}, _ = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      w(s);
    }
}, ze = (e, t) => O(e).$instanceValues$.get(t), qe = (e, t, n, s) => {
  const l = O(e), c = e, o = l.$instanceValues$.get(t), r = l.$flags$, i = c;
  n = Ae(n, s.$members$[t][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && r & 128) {
        const a = s.$watchers$[t];
        a && a.map((d) => {
          try {
            i[d](n, o, t);
          } catch ($) {
            w($, c);
          }
        });
      }
      if ((r & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, t) === !1)
          return;
        ge(l, !1);
      }
    }
  }
}, Ge = (e, t, n) => {
  var s;
  const l = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const c = Object.entries(t.$members$);
    c.map(([o, [r]]) => {
      (r & 31 || r & 32) && Object.defineProperty(l, o, {
        get() {
          return ze(this, o);
        },
        set(i) {
          qe(this, o, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(r, i, u) {
        g.jmp(() => {
          var f;
          const a = o.get(r);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (l.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = O(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const y = this, p = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[r];
                p == null || p.forEach((L) => {
                  y[L] != null && y[L].call(y, u, i, r);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...c.filter(
          ([r, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([r, i]) => {
          var u;
          const f = i[1] || r;
          return o.set(f, r), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([r, f])), f;
        })
      ]));
    }
  }
  return e;
}, Ie = async (e, t, n, s) => {
  let l;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, l = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[t.$modeName$ = _e(e)]);
    const r = ie(n, t.$modeName$);
    if (!U.has(r)) {
      const i = x("registerStyles", n.$tagName$);
      Re(r, o, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, ge(t, !0);
}, ee = (e) => {
}, Qe = (e) => {
  if (!(g.$flags$ & 1)) {
    const t = O(e), n = t.$cmpMeta$, s = x("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (pe(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xe(e), n.$members$ && Object.entries(n.$members$).map(([l, [c]]) => {
      if (c & 31 && e.hasOwnProperty(l)) {
        const o = e[l];
        delete e[l], e[l] = o;
      }
    }), Ie(e, t, n)), s();
  }
}, Xe = (e) => {
  const t = e["s-cr"] = k.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Ze = async (e) => {
  if (!(g.$flags$ & 1)) {
    const t = O(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Ke = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [], !se && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = e.prototype.connectedCallback, l = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      et(this, n);
    },
    connectedCallback() {
      Qe(this), s && s.call(this);
    },
    disconnectedCallback() {
      Ze(this), l && l.call(this);
    },
    __attachShadow() {
      se ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), e.is = n.$tagName$, Ge(e, n);
}, pe = (e, t, n, s) => {
  n && n.map(([l, c, o]) => {
    const r = Ye(e, l), i = Je(t, o), u = Ve(l);
    g.ael(r, c, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => g.rel(r, c, i, u));
  });
}, Je = (e, t) => (n) => {
  try {
    T.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    w(s);
  }
}, Ye = (e, t) => t & 4 ? k : t & 8 ? j : t & 16 ? k.body : e, Ve = (e) => nt ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ye = /* @__PURE__ */ new WeakMap(), O = (e) => ye.get(e), et = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return pe(e, n, t.$listeners$), ye.set(e, n);
}, te = (e, t) => t in e, w = (e, t) => (0, console.error)(e, t), U = /* @__PURE__ */ new Map(), tt = [], j = typeof window < "u" ? window : {}, k = j.document || { head: {} }, st = j.HTMLElement || class {
}, g = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, se = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), nt = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    k.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), lt = (e) => Promise.resolve(e), ot = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), ne = [], ve = [], it = (e, t) => (n) => {
  e.push(n), D || (D = !0, t && g.$flags$ & 4 ? ct(N) : g.raf(N));
}, le = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      w(n);
    }
  e.length = 0;
}, N = () => {
  le(ne), le(ve), (D = ne.length > 0) && g.raf(N);
}, ct = (e) => lt().then(e), rt = /* @__PURE__ */ it(ve, !0), at = ".sc-google-logo-icon-h{display:flex}", $t = at, ft = Ke(class extends st {
  constructor() {
    super(), this.__registerHost(), this.width = "24", this.height = "24";
  }
  render() {
    return S("svg", { key: "3ca599e18ce6095779ded84b497b5296f17c10d2", width: this.width, height: this.height, viewBox: "0 0 24 24", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, S("g", { key: "88b3f5f83ede3ed85a770467f48ad836405b990a", "clip-path": "url(#clip0_959_12452)" }, S("path", { key: "4e15a1ff9fd5ed8752a01653cc0313519c3979ca", d: "M23.745 12.27C23.745 11.48 23.675 10.73 23.555 10H12.255V14.51H18.725C18.435 15.99 17.585 17.24 16.325 18.09V21.09H20.185C22.445 19 23.745 15.92 23.745 12.27Z", fill: "#4285F4" }), S("path", { key: "1d5b9ac448e9067003693fc996d84e4a6a6fb424", d: "M12.255 24C15.495 24 18.205 22.92 20.185 21.09L16.325 18.09C15.245 18.81 13.875 19.25 12.255 19.25C9.12504 19.25 6.47505 17.14 5.52505 14.29H1.54504V17.38C3.51504 21.3 7.56504 24 12.255 24Z", fill: "#34A853" }), S("path", { key: "60c45db031014875d1f0c51e4d96d233ff4d739e", d: "M5.52501 14.2901C5.27501 13.5701 5.145 12.8001 5.145 12.0001C5.145 11.2001 5.28501 10.4301 5.52501 9.71012V6.62012H1.545C0.725004 8.24012 0.255005 10.0601 0.255005 12.0001C0.255005 13.9401 0.725004 15.7601 1.545 17.3801L5.52501 14.2901Z", fill: "#FBBC05" }), S("path", { key: "c11e6c275b10368721e0d520d98a8f57bc38acc5", d: "M12.255 4.75C14.025 4.75 15.605 5.36 16.855 6.55L20.275 3.13C18.205 1.19 15.495 0 12.255 0C7.56504 0 3.51504 2.7 1.54504 6.62L5.52505 9.71C6.47505 6.86 9.12504 4.75 12.255 4.75Z", fill: "#EA4335" })), S("defs", { key: "9ed8d3df0e31854c32fe754fc79877ca48ea5ab9" }, S("clipPath", { key: "7b54dd3d596dc38980237b59f21184a00316760f", id: "clip0_959_12452" }, S("rect", { key: "6632bab4f2645c9a5dc9f3f000a7b537d5b09f22", width: 24, height: 24, fill: "white" }))));
  }
  static get style() {
    return $t;
  }
}, [2, "google-logo-icon", {
  width: [1],
  height: [1]
}]);
function ut() {
  if (typeof customElements > "u")
    return;
  ["google-logo-icon"].forEach((t) => {
    switch (t) {
      case "google-logo-icon":
        customElements.get(t) || customElements.define(t, ft);
        break;
    }
  });
}
export {
  ft as G,
  ut as d
};
