// @nylas/web-elements/dist/collection/components/design-system/tooltip-component/toolitp-component.js
import { h } from "@stencil/core";
export class TooltipComponent {
    constructor() {
        this.position = 'bottom';
    }
    showTooltip(event) {
        event.target.nextElementSibling.style.visibility = 'visible';
        event.target.nextElementSibling.style.opacity = '1';
    }
    hideTooltip(event) {
        event.target.nextElementSibling.style.visibility = 'hidden';
        event.target.nextElementSibling.style.opacity = '0';
    }
    render() {
        return (h("div", { key: '2ca61ec97333e6f3c5504b5123747d82037f8650', class: "tooltip-container", part: "tc__container" }, h("span", { key: '4133e0e9a968139a8101789c5291f824a87b5c9d', class: "info-icon", tabindex: "0", onFocus: this.showTooltip.bind(this), onBlur: this.hideTooltip.bind(this) }, h("slot", { key: '6b07811a2de2bb05b8133c4d2992eb36ae280f8e', name: "tooltip-icon" }, "i")), h("span", { key: '99de5567aaa0998d1d29042f4b86fac5c0689dc6', class: `tooltip-content ${this.position}`, part: "tc__content" }, h("slot", { key: '608f52f2bd9bfef59c8dd32fd6f5f15bf044bb12', name: "tooltip-content" }, "This is a help text."))));
    }
    static get is() { return "tooltip-component"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["tooltip-component.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["tooltip-component.css"]
        };
    }
    static get properties() {
        return {
            "position": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "'top' | 'bottom' | 'left' | 'right'",
                    "resolved": "\"bottom\" | \"left\" | \"right\" | \"top\"",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The position of the tooltip."
                },
                "attribute": "position",
                "reflect": false,
                "defaultValue": "'bottom'"
            }
        };
    }
    static get elementRef() { return "el"; }
}
