// @nylas/web-elements/dist/cdn/trash/trash.es.js
const L = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, B, _, ot = !1, w = !1, W = !1, v = !1, Z = null, D = !1;
const b = (t, e = "") => () => {
}, St = "slot-fb{display:contents}slot-fb[hidden]{display:none}", I = "http://www.w3.org/1999/xlink", Q = {}, mt = "http://www.w3.org/2000/svg", kt = "http://www.w3.org/1999/xhtml", Lt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function Tt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const F = (t, e, ...n) => {
  let s = null, l = null, r = null, o = !1, c = !1;
  const i = [], u = ($) => {
    for (let d = 0; d < $.length; d++)
      s = $[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof t != "function" && !z(s)) && (s = String(s)), o && c ? i[i.length - 1].$text$ += s : i.push(o ? U(null, s) : s), c = o);
  };
  if (u(n), e) {
    e.key && (l = e.key), e.name && (r = e.name);
    {
      const $ = e.className || e.class;
      $ && (e.class = typeof $ != "object" ? $ : Object.keys($).filter((d) => $[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, Ct);
  const f = U(t, null);
  return f.$attrs$ = e, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = r, f;
}, U = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, bt = {}, xt = (t) => t && t.$tag$ === bt, Ct = {
  forEach: (t, e) => t.map(X).forEach(e),
  map: (t, e) => t.map(X).map(e).map(Et)
}, X = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), Et = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), F(t.vtag, n, ...t.vchildren || []);
  }
  const e = U(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, _t = (t) => ee.map((e) => e(t)).find((e) => !!e), At = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, G = /* @__PURE__ */ new WeakMap(), Rt = (t, e, n) => {
  let s = P.get(t);
  oe && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, P.set(t, s);
}, Ot = (t, e, n) => {
  var s;
  const l = it(e, n), r = P.get(l);
  if (t = t.nodeType === 11 ? t : S, r)
    if (typeof r == "string") {
      t = t.head || t;
      let o = G.get(t), c;
      if (o || G.set(t, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Tt(S);
          i != null && c.setAttribute("nonce", i), t.insertBefore(c, t.querySelector("link"));
        }
        e.$flags$ & 4 && (c.innerHTML += St), o && o.add(l);
      }
    } else t.adoptedStyleSheets.includes(r) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, r]);
  return l;
}, Ht = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, l = b("attachStyles", e.$tagName$), r = Ot(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), l();
}, it = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), K = (t, e, n, s, l, r) => {
  if (n !== s) {
    let o = et(t, e), c = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = J(n), f = J(s);
      i.remove(...u.filter(($) => $ && !f.includes($))), i.add(...f.filter(($) => $ && !u.includes($)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(j, c) ? e = c.slice(2) : e = c[2] + e.slice(3), n || s) {
          const i = e.endsWith(rt);
          e = e.replace(Ft, ""), n && p.rel(t, e, n, i), s && p.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const f = s ?? "";
              e === "list" ? o = !1 : (n == null || t[e] != f) && (t[e] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (e = c, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(I, e) : t.removeAttribute(e)) : (!o || r & 4 || l) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(I, e, s) : t.setAttribute(e, s));
      }
  }
}, wt = /\s/, J = (t) => t ? t.split(wt) : [], rt = "Capture", Ft = new RegExp(rt + "$"), ct = (t, e, n, s) => {
  const l = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, r = t && t.$attrs$ || Q, o = e.$attrs$ || Q;
  for (s in r)
    s in o || K(l, s, r[s], void 0, n, e.$flags$);
  for (s in o)
    K(l, s, r[s], o[s], n, e.$flags$);
}, M = (t, e, n, s) => {
  const l = e.$children$[n];
  let r = 0, o, c, i;
  if (ot || (W = !0, l.$tag$ === "slot" && (x && s.classList.add(x + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? mt : kt, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), ct(null, l, v), Lt(x) && o["s-si"] !== x && o.classList.add(o["s-si"] = x), l.$children$)
      for (r = 0; r < l.$children$.length; ++r)
        c = M(t, l, r, o), c && o.appendChild(c);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = _, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = B, o["s-sn"] = l.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === l.$tag$ && t.$elm$ && A(t.$elm$, !1)), o;
}, A = (t, e) => {
  p.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== _ && l["s-ol"] && (ft(l).insertBefore(l, q(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), e && A(l, e);
  }
  p.$flags$ &= -2;
}, $t = (t, e, n, s, l, r) => {
  let o = t["s-cr"] && t["s-cr"].parentNode || t, c;
  for (o.shadowRoot && o.tagName === _ && (o = o.shadowRoot); l <= r; ++l)
    s[l] && (c = M(null, n, l, t), c && (s[l].$elm$ = c, o.insertBefore(c, q(e))));
}, at = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const l = t[s];
    if (l) {
      const r = l.$elm$;
      ht(l), r && (w = !0, r["s-ol"] ? r["s-ol"].remove() : A(r, !0), r.remove());
    }
  }
}, Ut = (t, e, n, s, l = !1) => {
  let r = 0, o = 0, c = 0, i = 0, u = e.length - 1, f = e[0], $ = e[u], d = s.length - 1, a = s[0], h = s[d], y, g;
  for (; r <= u && o <= d; )
    if (f == null)
      f = e[++r];
    else if ($ == null)
      $ = e[--u];
    else if (a == null)
      a = s[++o];
    else if (h == null)
      h = s[--d];
    else if (H(f, a, l))
      C(f, a, l), f = e[++r], a = s[++o];
    else if (H($, h, l))
      C($, h, l), $ = e[--u], h = s[--d];
    else if (H(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(f.$elm$.parentNode, !1), C(f, h, l), t.insertBefore(f.$elm$, $.$elm$.nextSibling), f = e[++r], h = s[--d];
    else if (H($, a, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A($.$elm$.parentNode, !1), C($, a, l), t.insertBefore($.$elm$, f.$elm$), $ = e[--u], a = s[++o];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === a.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = e[c], g.$tag$ !== a.$tag$ ? y = M(e && e[o], n, c, t) : (C(g, a, l), e[c] = void 0, y = g.$elm$), a = s[++o]) : (y = M(e && e[o], n, o, t), a = s[++o]), y && ft(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? $t(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && at(e, r, u);
}, H = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, ft = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, C = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, l = t.$children$, r = e.$children$, o = e.$tag$, c = e.$text$;
  let i;
  c === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || ct(t, e, v), l !== null && r !== null ? Ut(s, l, e, r, n) : r !== null ? (t.$text$ !== null && (s.textContent = ""), $t(s, null, e, r, 0, r.length - 1)) : l !== null && at(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : t.$text$ !== c && (s.data = c);
}, dt = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of e)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      dt(n);
    }
}, m = [], ut = (t) => {
  let e, n, s;
  for (const l of t.childNodes) {
    if (l["s-sr"] && (e = l["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const r = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== l["s-hn"] && !L.experimentalSlotFixes)
          if (Y(e, r)) {
            let o = m.find((c) => c.$nodeToRelocate$ === e);
            w = !0, e["s-sn"] = e["s-sn"] || r, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (e["s-sh"] = l["s-hn"], m.push({
              $slotRefNode$: l,
              $nodeToRelocate$: e
            })), e["s-sr"] && m.map((c) => {
              Y(c.$nodeToRelocate$, e["s-sn"]) && (o = m.find((i) => i.$nodeToRelocate$ === e), o && !c.$slotRefNode$ && (c.$slotRefNode$ = o.$slotRefNode$));
            });
          } else m.some((o) => o.$nodeToRelocate$ === e) || m.push({
            $nodeToRelocate$: e
          });
    }
    l.nodeType === 1 && ut(l);
  }
}, Y = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ht = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ht);
}, Mt = (t, e, n = !1) => {
  var s, l, r, o, c;
  const i = t.$hostElement$, u = t.$cmpMeta$, f = t.$vnode$ || U(null, null), $ = xt(e) ? e : F(null, null, e);
  if (_ = i.tagName, u.$attrsToReflect$ && ($.$attrs$ = $.$attrs$ || {}, u.$attrsToReflect$.map(([d, a]) => $.$attrs$[a] = i[d])), n && $.$attrs$)
    for (const d of Object.keys($.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && ($.$attrs$[d] = i[d]);
  $.$tag$ = null, $.$flags$ |= 4, t.$vnode$ = $, $.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], B = i["s-cr"], ot = (u.$flags$ & 1) !== 0, w = !1, C(f, $, n);
  {
    if (p.$flags$ |= 1, W) {
      ut($.$elm$);
      for (const d of m) {
        const a = d.$nodeToRelocate$;
        if (!a["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = a, a.parentNode.insertBefore(a["s-ol"] = h, a);
        }
      }
      for (const d of m) {
        const a = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = a["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let k = (l = T["s-nr"]) !== null && l !== void 0 ? l : null;
              if (k && k["s-sn"] === a["s-sn"] && y === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                g = k;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== a.parentNode || a.nextSibling !== g) && a !== g && (!a["s-hn"] && a["s-ol"] && (a["s-hn"] = a["s-ol"].parentNode.nodeName), y.insertBefore(a, g), a.nodeType === 1 && (a.hidden = (r = a["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          a.nodeType === 1 && (n && (a["s-ih"] = (o = a.hidden) !== null && o !== void 0 ? o : !1), a.hidden = !0);
      }
    }
    w && dt($.$elm$), p.$flags$ &= -2, m.length = 0;
  }
  if (L.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of $.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  B = void 0;
}, Pt = (t, e) => {
}, pt = (t, e) => (t.$flags$ |= 16, Pt(t, t.$ancestorComponent$), ce(() => jt(t, e))), jt = (t, e) => {
  const n = t.$hostElement$, s = b("scheduleUpdate", t.$cmpMeta$.$tagName$), l = n;
  let r;
  return e ? r = E(l, "componentWillLoad") : r = E(l, "componentWillUpdate"), r = V(r, () => E(l, "componentWillRender")), s(), V(r, () => Dt(t, l, e));
}, V = (t, e) => Bt(t) ? t.then(e) : e(), Bt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Dt = async (t, e, n) => {
  const s = t.$hostElement$, l = b("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ht(t);
  const r = b("render", t.$cmpMeta$.$tagName$);
  Nt(t, e, s, n), r(), l(), Wt(t);
}, Nt = (t, e, n, s) => {
  try {
    Z = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (L.hasRenderFn || L.reflect) && (L.vdomRender || L.reflect) && (L.hydrateServerSide || Mt(t, e, s));
  } catch (i) {
    O(i, t.$hostElement$);
  }
  return Z = null, null;
}, Wt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = b("postUpdate", e), l = n;
  t.$ancestorComponent$, E(l, "componentDidRender"), t.$flags$ & 64 ? (E(l, "componentDidUpdate"), s()) : (t.$flags$ |= 64, E(l, "componentDidLoad"), s());
}, E = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      O(s);
    }
}, zt = (t, e) => R(t).$instanceValues$.get(e), qt = (t, e, n, s) => {
  const l = R(t), r = t, o = l.$instanceValues$.get(e), c = l.$flags$, i = r;
  n = At(n, s.$members$[e][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && c & 128) {
        const $ = s.$watchers$[e];
        $ && $.map((d) => {
          try {
            i[d](n, o, e);
          } catch (a) {
            O(a, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, e) === !1)
          return;
        pt(l, !1);
      }
    }
  }
}, Zt = (t, e, n) => {
  var s;
  const l = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const r = Object.entries(e.$members$);
    r.map(([o, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(l, o, {
        get() {
          return zt(this, o);
        },
        set(i) {
          qt(this, o, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(c, i, u) {
        p.jmp(() => {
          var f;
          const $ = o.get(c);
          if (this.hasOwnProperty($))
            u = this[$], delete this[$];
          else {
            if (l.hasOwnProperty($) && typeof this[$] == "number" && this[$] == u)
              return;
            if ($ == null) {
              const d = R(this), a = d == null ? void 0 : d.$flags$;
              if (a && !(a & 8) && a & 128 && u !== i) {
                const y = this, g = (f = e.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[$] = u === null && typeof this[$] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return o.set(f, c), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return t;
}, It = async (t, e, n, s) => {
  let l;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, l = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[e.$modeName$ = _t(t)]);
    const c = it(n, e.$modeName$);
    if (!P.has(c)) {
      const i = b("registerStyles", n.$tagName$);
      Rt(c, o, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, pt(e, !0);
}, tt = (t) => {
}, Qt = (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t), n = e.$cmpMeta$, s = b("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (gt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Xt(t), n.$members$ && Object.entries(n.$members$).map(([l, [r]]) => {
      if (r & 31 && t.hasOwnProperty(l)) {
        const o = t[l];
        delete t[l], t[l] = o;
      }
    }), It(t, e, n)), s();
  }
}, Xt = (t) => {
  const e = t["s-cr"] = S.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Gt = async (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Kt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [], !st && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = t.prototype.connectedCallback, l = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      te(this, n);
    },
    connectedCallback() {
      Qt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Gt(this), l && l.call(this);
    },
    __attachShadow() {
      st ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), t.is = n.$tagName$, Zt(t, n);
}, gt = (t, e, n, s) => {
  n && n.map(([l, r, o]) => {
    const c = Yt(t, l), i = Jt(e, o), u = Vt(l);
    p.ael(c, r, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => p.rel(c, r, i, u));
  });
}, Jt = (t, e) => (n) => {
  try {
    L.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    O(s);
  }
}, Yt = (t, e) => e & 4 ? S : e & 8 ? j : e & 16 ? S.body : t, Vt = (t) => ne ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, yt = /* @__PURE__ */ new WeakMap(), R = (t) => yt.get(t), te = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return gt(t, n, e.$listeners$), yt.set(t, n);
}, et = (t, e) => e in t, O = (t, e) => (0, console.error)(t, e), P = /* @__PURE__ */ new Map(), ee = [], j = typeof window < "u" ? window : {}, S = j.document || { head: {} }, se = j.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, st = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), ne = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), le = (t) => Promise.resolve(t), oe = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), nt = [], vt = [], ie = (t, e) => (n) => {
  t.push(n), D || (D = !0, e && p.$flags$ & 4 ? re(N) : p.raf(N));
}, lt = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      O(n);
    }
  t.length = 0;
}, N = () => {
  lt(nt), lt(vt), (D = nt.length > 0) && p.raf(N);
}, re = (t) => le().then(t), ce = /* @__PURE__ */ ie(vt, !0), $e = ".sc-trash-icon-h{display:flex}", ae = $e, fe = Kt(class extends se {
  constructor() {
    super(), this.__registerHost(), this.width = "20", this.height = "21";
  }
  render() {
    return F("svg", { key: "8ae039082a2e896dafbb6e2cf02741cac38760bc", width: this.width, height: this.height, viewBox: "0 0 20 21", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, F("path", { key: "79e7dc705693fc18a431027efcfdc66b31dedf00", d: "M10 0.25C11.733 0.25 13.1492 1.60645 13.2449 3.31558L13.25 3.5H18.5C18.9142 3.5 19.25 3.83579 19.25 4.25C19.25 4.6297 18.9678 4.94349 18.6018 4.99315L18.5 5H17.704L16.4239 18.0192C16.2912 19.3683 15.1984 20.41 13.8626 20.4945L13.6871 20.5H6.31293C4.95734 20.5 3.81365 19.5145 3.59883 18.1934L3.57614 18.0192L2.295 5H1.5C1.1203 5 0.806509 4.71785 0.756847 4.35177L0.75 4.25C0.75 3.8703 1.03215 3.55651 1.39823 3.50685L1.5 3.5H6.75C6.75 1.70507 8.20507 0.25 10 0.25ZM16.197 5H3.802L5.06893 17.8724C5.12768 18.4696 5.60033 18.9343 6.18585 18.9936L6.31293 19H13.6871C14.2872 19 14.7959 18.5751 14.9123 17.9982L14.9311 17.8724L16.197 5ZM11.75 7.75C12.1297 7.75 12.4435 8.03215 12.4932 8.39823L12.5 8.5V15.5C12.5 15.9142 12.1642 16.25 11.75 16.25C11.3703 16.25 11.0565 15.9678 11.0068 15.6018L11 15.5V8.5C11 8.08579 11.3358 7.75 11.75 7.75ZM8.25 7.75C8.6297 7.75 8.94349 8.03215 8.99315 8.39823L9 8.5V15.5C9 15.9142 8.66421 16.25 8.25 16.25C7.8703 16.25 7.55651 15.9678 7.50685 15.6018L7.5 15.5V8.5C7.5 8.08579 7.83579 7.75 8.25 7.75ZM10 1.75C9.08183 1.75 8.32881 2.45711 8.2558 3.35647L8.25 3.5H11.75C11.75 2.5335 10.9665 1.75 10 1.75Z", fill: "currentColor" }));
  }
  static get style() {
    return ae;
  }
}, [2, "trash-icon", {
  width: [1],
  height: [1]
}]);
function ue() {
  if (typeof customElements > "u")
    return;
  ["trash-icon"].forEach((e) => {
    switch (e) {
      case "trash-icon":
        customElements.get(e) || customElements.define(e, fe);
        break;
    }
  });
}
export {
  fe as T,
  ue as d
};
