// @nylas/web-elements/dist/nylas-web-elements/multi-select-dropdown.entry.js
import { r as registerInstance, a as createEvent, h, i as Host, e as getElement } from './index-b5cb18eb.js';
import { a as debug } from './utils-14cfef23.js';

const multiSelectDropdownCss = ":host{display:block;width:inherit;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.dropdown{display:inline-block;width:100%;position:relative}.dropdown .dropdown-label{display:flex;align-items:center;gap:0.25rem;color:var(--nylas-base-800)}.dropdown .dropdown-label p{margin:0}.dropdown .dropdown-label span.required{color:var(--nylas-error);padding:0 0.25rem}.dropdown span.error{color:var(--nylas-error);font-size:14px}.dropbtn{width:inherit;height:48px;color:black;padding:0.5rem;font-size:16px;cursor:pointer;display:flex;gap:0.5rem;justify-content:space-between;align-items:center;background:transparent;border:1px solid var(--nylas-base-200);border-radius:var(--nylas-border-radius-2x)}.dropbtn.focus{background:transparent}.dropbtn.error{border:1px solid var(--nylas-error)}.dropbtn:hover,.dropbtn:active{border:1px solid var(--nylas-primary)}.dropbtn:active{outline:2px solid var(--nylas-primary)}.dropbtn span.open{transform:rotate(90deg)}.dropbtn span.closed{transform:rotate(270deg)}.dropdown-content{display:block;margin-top:0.5rem;background-color:var(--nylas-base-0);width:100%;max-height:336px;overflow:auto;z-index:1;border-radius:4px;position:absolute;top:72px;box-shadow:0px 4px 6px -2px rgba(0, 0, 0, 0.0509803922);box-shadow:0px 10px 15px -3px rgba(0, 0, 0, 0.1019607843)}.dropdown-content ul{padding:0;list-style-type:none;color:var(--nylas-base-900);max-height:336px}.dropdown-content ul li{padding:16px, 12px, 16px, 12px;color:black;padding:12px 16px;text-decoration:none;display:block;font-family:inherit;font-size:14px;font-weight:400;line-height:20px;letter-spacing:0px;text-align:left;cursor:pointer}.dropdown-content ul li:hover,.dropdown-content ul li:focus{background-color:var(--nylas-base-100)}.dropdown-content ul li label{display:flex;align-items:center;gap:0.5rem}.dropdown-content ul li label input{margin:0}.selected-options{display:flex;flex-wrap:wrap;gap:0.5rem;padding:0.5rem 0;margin-top:0.25rem;background:var(--nylas-base-0)}.selected-options .selected-option{display:flex;align-items:center;gap:0.5rem;padding:4px 8px;border-radius:var(--nylas-border-radius-2x);background:var(--nylas-base-100);color:var(--nylas-base-800);font-size:16px;font-weight:500;line-height:1.5rem;letter-spacing:0.5px}.selected-options .selected-option button{background:transparent;border:none;cursor:pointer;padding:0}.selected-options .selected-option button:hover{color:var(--nylas-primary)}";

const MultiSelectDropdown = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.selectedOptionsChanged = createEvent(this, "selectedOptionsChanged", 7);
        this.name = undefined;
        this.label = undefined;
        this.options = [];
        this.error = '';
        this.required = false;
        this.availableOptions = this.options;
        this.isOpen = false;
        this.ariaActivedescendant = '';
        this.shouldFocusFirstOption = false;
    }
    areOptionsEqual(arr1, arr2) {
        if (arr1.length !== arr2.length)
            return false;
        const sorted1 = [...arr1].sort((a, b) => a.label.localeCompare(b.label) || a.value.localeCompare(b.value));
        const sorted2 = [...arr2].sort((a, b) => a.label.localeCompare(b.label) || a.value.localeCompare(b.value));
        return sorted1.every((opt1, index) => {
            const opt2 = sorted2[index];
            return opt1.label === opt2.label && opt1.value === opt2.value;
        });
    }
    optionsChangedHandler(newValue) {
        if (!this.areOptionsEqual(newValue, this.availableOptions)) {
            this.availableOptions = newValue;
        }
    }
    componentDidLoad() {
        debug('multi-select-dropdown', 'componentDidLoad');
        if (this.options) {
            this.availableOptions = this.options;
        }
    }
    componentDidRender() {
        debug('multi-select-dropdown', 'componentDidRender');
        if (this.isOpen && this.shouldFocusFirstOption) {
            this.ariaActivedescendant = this.availableOptions[0]?.value;
            this.focusOption(0);
            this.shouldFocusFirstOption = false;
        }
    }
    handleOutsideClick(event) {
        const path = event.composedPath();
        const isClickInside = path.includes(this.el);
        if (!isClickInside && this.isOpen) {
            this.isOpen = false;
        }
    }
    handleBookingFormSubmitted(event) {
        const selectedOptions = this.availableOptions.filter(o => o.selected).map(o => o.value);
        if (this.required && (!selectedOptions || selectedOptions.length <= 0)) {
            this.error = 'Please select at least one option';
        }
        if (this.error) {
            event.preventDefault();
        }
    }
    selectOption(option) {
        this.availableOptions = this.availableOptions.map(o => {
            if (o.value === option.value) {
                o.selected = option.selected ? false : true;
                if (o.selected) {
                    this.error = '';
                }
            }
            return o;
        });
        const selectedOptions = this.availableOptions.filter(o => o.selected).map(o => o.value);
        this.selectedOptionsChanged.emit({
            value: selectedOptions,
            name: this.name,
        });
    }
    toggleDropdown() {
        this.isOpen = !this.isOpen;
        if (this.isOpen) {
            this.shouldFocusFirstOption = true;
        }
        else {
            this.ariaActivedescendant = '';
        }
    }
    handleSelectButtonKeyDown(event) {
        switch (event.key) {
            case 'ArrowDown':
            case 'Enter':
                event.preventDefault();
                if (!this.isOpen) {
                    this.toggleDropdown();
                }
                break;
            case 'Escape':
                this.isOpen = false;
                break;
        }
    }
    handleListboxKeydown(e) {
        const items = this.availableOptions;
        const currentIndex = items.findIndex(item => item.value === this.ariaActivedescendant);
        switch (e.key) {
            case 'ArrowDown':
            case 'Tab':
                if (!e.shiftKey) {
                    e.preventDefault();
                    const nextIndex = currentIndex + 1 < items.length ? currentIndex + 1 : 0;
                    this.ariaActivedescendant = items[nextIndex].value;
                    this.focusOption(nextIndex);
                }
                else {
                    e.preventDefault();
                    const prevIndex = currentIndex - 1 >= 0 ? currentIndex - 1 : items.length - 1;
                    this.ariaActivedescendant = items[prevIndex].value;
                    this.focusOption(prevIndex);
                }
                break;
            case 'ArrowUp':
                e.preventDefault();
                const prevIndex = currentIndex - 1 >= 0 ? currentIndex - 1 : items.length - 1;
                this.ariaActivedescendant = items[prevIndex].value;
                this.focusOption(prevIndex);
                break;
            case 'Enter':
                e.preventDefault();
                if (this.ariaActivedescendant) {
                    this.selectOption(items[currentIndex]);
                }
                break;
            case 'Escape':
                this.isOpen = false;
                break;
        }
    }
    focusOption(index) {
        const option = this.availableOptions[index];
        if (!option)
            return;
        const elementId = option.value;
        const element = this.el.shadowRoot?.getElementById(elementId);
        if (element) {
            element.focus();
            element.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
        }
    }
    getSelectedOptions() {
        return this.availableOptions.filter(option => option.selected);
    }
    renderOption(option) {
        return (h("li", { key: option.value, id: option.value, role: "option", tabindex: "0", "aria-selected": option.selected ? 'true' : 'false', onClick: e => {
                e.stopImmediatePropagation();
                this.selectOption(option);
            }, class: { selected: !!option.selected } }, h("label", { htmlFor: option.value }, h("input", { "aria-hidden": "true", id: option.value, type: "checkbox", checked: option.selected }), h("span", null, option.label))));
    }
    render() {
        return (h(Host, { key: 'f4652f92d5814ddc59b0af4f430b8c6bf8c53056' }, h("div", { key: 'ef1db0ae49a7c75cb9cfc909ac83b65d172d242b', class: "dropdown", part: "msd_dropdown" }, h("label", { key: '14c4bfa2dfebf9bb1c84f31926c0cb5113867218', class: "dropdown-label", part: "msd_dropdown-label" }, h("p", { key: '588aad0241a5afd894f11ab4826ec1c2e159e51e' }, h("span", { key: 'b99c592d1895a51075ea46e2f5ac94bed4d7d8d7', class: "label" }, this.label), this.required && h("span", { class: "required" }, "*")), h("slot", { key: '2161c9a28066e0cf26289cb2fb20493ef7be51da', name: "label-icon", "aria-hidden": "true" })), h("button", { key: '3a39cfacf9faaae6177df65e49e61ec5167be1b1', name: this.name, part: `msd_dropdown-button ${this.error ? 'msd_dropdown-button--error' : ''}`, class: { dropbtn: true, open: this.isOpen, error: !!this.error }, onClick: () => this.toggleDropdown(), "aria-haspopup": "listbox", "aria-expanded": this.isOpen ? 'true' : 'false', "aria-label": this.name, onKeyDown: e => this.handleSelectButtonKeyDown(e) }, h("slot", { key: '5d09d93a9565e2f12c8d3f8f2a391832b576e5ee', name: "select-icon", "aria-hidden": "true" }), h("span", { key: '7f15d2a1644a44448cef246c4480354d4fa2c3d0', class: "selected-option", part: "msd_dropdown-button-selected-label" }, this.getSelectedOptions().length > 1 ? `Multiple options selected` : this.availableOptions.filter(o => o.selected)[0]?.label ?? this.availableOptions[0]?.label), h("span", { key: 'af988d0cb66e710b0a23e72392d4cc84d36a6e17', class: this.isOpen ? 'open' : 'closed', "aria-hidden": "true" }, h("chevron-icon", { key: '64efffb0e7ab44bc44b210eea413d946cbad4dc5', width: "16", height: "16" }))), this.error ? (h("span", { class: "error", part: "msd_dropdown_error" }, this.error)) : null, this.isOpen ? null : (h("div", { class: 'selected-options' }, this.getSelectedOptions().map(option => (h("span", { class: "selected-option" }, option.label, h("button", { key: option.label, onClick: () => this.selectOption(option) }, h("close-icon", null))))))), this.isOpen ? (h("div", { class: "dropdown-content", part: "msd_dropdown-content" }, h("ul", { tabindex: "-1", role: "listbox", "aria-label": this.name, "aria-multiselectable": true, "aria-activedescendant": this.ariaActivedescendant, onKeyDown: e => this.handleListboxKeydown(e) }, this.availableOptions.map(option => this.renderOption(option))))) : null)));
    }
    get el() { return getElement(this); }
    static get watchers() { return {
        "options": ["optionsChangedHandler"]
    }; }
};
MultiSelectDropdown.style = multiSelectDropdownCss;

export { MultiSelectDropdown as multi_select_dropdown };
