// @nylas/web-elements/dist/collection/components/scheduler-editor/nylas-confirmation-email/nylas-confirmation-email.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { debug, sanitize } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
import i18next from "../../../utils/i18n";
export class NylasConfirmationEmail {
    constructor() {
        this.selectedConfiguration = undefined;
        this.confirmationEmailTemplate = {};
        this.participants = [];
        this.name = 'confirmation-email-template';
        this.isOpen = false;
        this.selectedLanguage = undefined;
        this.confirmationEmail = undefined;
        this.isConfirmationEmailOpen = this.isOpen;
        this.participantsState = this.participants;
    }
    connectedCallback() {
        debug('nylas-confirmation-email', 'connectedCallback');
    }
    disconnectedCallback() {
        debug('nylas-confirmation-email', 'disconnectedCallback');
    }
    componentWillLoad() {
        debug('nylas-confirmation-email', 'componentWillLoad');
        this.host.setAttribute('name', this.name);
    }
    componentDidLoad() {
        debug('nylas-confirmation-email', 'componentDidLoad');
        if (this.selectedConfiguration) {
            this.selectedConfigurationChangedHandler(this.selectedConfiguration);
        }
        else {
            this.confirmationEmail = this.confirmationEmailTemplate;
            if (!this.confirmationEmail?.show_nylas_branding) {
                this.confirmationEmail = { ...this.confirmationEmail, show_nylas_branding: true };
            }
            this.isConfirmationEmailOpen = this.isOpen;
            this.participantsState = this.participants;
        }
    }
    elementNameChangedHandler(newValue) {
        debug('nylas-confirmation-email', 'elementNameChangedHandler', newValue);
        this.host.setAttribute('name', newValue);
    }
    selectedConfigurationChangedHandler(newValue) {
        debug('nylas-confirmation-email', 'selectedConfigurationChangedHandler', newValue);
        this.confirmationEmail = newValue?.scheduler?.email_template || this.confirmationEmailTemplate;
        if (typeof this.confirmationEmail?.show_nylas_branding === 'undefined') {
            this.confirmationEmail = { ...this.confirmationEmail, show_nylas_branding: true };
        }
        this.isConfirmationEmailOpen = !newValue?.event_booking?.reminders?.length;
        this.participantsState = newValue?.participants || this.participants;
    }
    selectedLanguageChangedHandler(newVal, oldVal) {
        if (newVal === oldVal)
            return;
        i18next.changeLanguage(newVal);
    }
    async nylasFormInputChangeHandler(event) {
        const { name, value } = event.detail;
        const type = event.detail?.type;
        if (name === 'confirmation-email-title') {
            this.confirmationEmail = { ...this.confirmationEmail, booking_confirmed: { ...(this.confirmationEmail?.booking_confirmed || {}), title: value } };
            this.internals.setFormValue(JSON.stringify(this.confirmationEmail), this.name);
            this.valueChanged.emit({ value: JSON.stringify(this.confirmationEmail), name: this.name });
        }
        else if (name === 'confirmation-email-body' && type === 'multi_line_text') {
            this.handleEmailBodyChange(event);
        }
    }
    valueChangedHandler(event) {
        debug('nylas-confirmation-email', 'valueChangedHandler', event.detail);
        const { value, name } = event.detail;
        if (name === 'confirmation-email-logo') {
            this.confirmationEmail = { ...this.confirmationEmail, logo: value };
            this.valueChanged.emit({ value: JSON.stringify(this.confirmationEmail), name: this.name });
            if (typeof this.internals.setValidity === 'function' && typeof this.internals.setFormValue === 'function') {
                this.internals.setFormValue(JSON.stringify(this.confirmationEmail), this.name);
                this.internals.setValidity({});
            }
        }
    }
    async nylasFormInputImageUrlInvalidHandler(event) {
        const { value, name } = event.detail;
        if (name === 'confirmation-email-logo' && typeof this.internals.setValidity === 'function') {
            this.internals.setValidity({ customError: true }, value || 'Invalid image URL.');
        }
    }
    nyalsFormSwitchToggledHandler(event) {
        debug('nylas-confirmation-email', 'nyalsFormSwitchToggledHandler', event.detail);
        if (event.detail.name === 'confirmation-email-show-branding') {
            this.confirmationEmail = { ...this.confirmationEmail, show_nylas_branding: event.detail.checked };
            this.internals.setFormValue(JSON.stringify(this.confirmationEmail), this.name);
            this.valueChanged.emit({ value: JSON.stringify(this.confirmationEmail), name: this.name });
        }
    }
    updateConfirmationFormValue() {
        debug('nylas-confirmation-email', 'updateConfirmationFormValue');
        this.internals.setFormValue(JSON.stringify(this.confirmationEmail), this.name);
        this.valueChanged.emit({ value: JSON.stringify(this.confirmationEmail), name: this.name });
    }
    toggleConfirmationEmail() {
        this.isConfirmationEmailOpen = !this.isConfirmationEmailOpen;
    }
    handleEmailBodyChange(event) {
        const value = sanitize(event.detail.value);
        this.confirmationEmail = { ...this.confirmationEmail, booking_confirmed: { ...(this.confirmationEmail?.booking_confirmed || {}), body: value } };
        this.internals.setFormValue(JSON.stringify(this.confirmationEmail), this.name);
        this.valueChanged.emit({ value: JSON.stringify(this.confirmationEmail), name: this.name });
    }
    render() {
        const organizer = this.participantsState.find(p => p.is_organizer);
        const organizerName = organizer?.name || organizer?.email;
        return (h(Host, { key: 'f3d6c5e775d26df9e6693f6e2ebb66cbf50afb50' }, h("div", { key: 'eeb7e06b2fc8a43b137eb8ad0aa23aea7c057050', class: "nylas-confirmation-email", part: "nce" }, h("div", { key: '4d318a3ee7bfe014c4fe28f4295e706ccdb55c54', class: "header", part: "nce__header" }, h("div", { key: '3f727906b1c0146922464e8e34c7ef7c87de5def' }, h("h3", { key: 'e003e19ba7713de2bc4dbd549786d8223d62d25b' }, this.confirmationEmail?.booking_confirmed?.title ||
            i18next.t('nylasConfirmationEmail.headerTitle', { organizerName: `${organizerName ? ` with ${organizerName}` : ''}` })), h("p", { key: 'f6fd1e8a4b80ae811b2b1ea1e265337dfed5c0f1' }, i18next.t('nylasConfirmationEmail.headerSubTitle'))), this.isConfirmationEmailOpen ? (h("div", { class: "confirmation-email-toggle", part: "nce__confirmation-email-toggle--container" }, h("span", { class: `chevron ${this.isConfirmationEmailOpen ? 'open' : 'closed'} `, onClick: () => this.toggleConfirmationEmail() }, h("chevron-icon", { width: "24", height: "24" })))) : (h("button-component", { variant: 'basic', clickHandler: event => {
                event.preventDefault();
                this.toggleConfirmationEmail();
            } }, h("edit-icon", { width: "16", height: "16" }), i18next.t('nylasConfirmationEmail.editButtonText')))), this.isConfirmationEmailOpen ? (h("div", { class: "nylas-confirmation-email__body", part: "nce__body" }, h("div", { class: "nylas-confirmation-email__section" }, h("div", { class: "nylas-confirmation-email__row" }, h("label", null, i18next.t('nylasConfirmationEmail.confirmationEmail.label'), h("span", { class: "label-icon" }, h("tooltip-component", null, h("info-icon", { slot: "tooltip-icon" }), h("span", { slot: "tooltip-content" }, " ", i18next.t('nylasConfirmationEmail.confirmationEmail.tooltip.desc'))))), h("span", { class: "selected-value" }, i18next.t('nylasConfirmationEmail.immediatelyUponBooking'))), h("div", { class: "nylas-confirmation-email__row" }, h("label", null, i18next.t('nylasConfirmationEmail.nylasBranding.label'), h("span", { class: "label-icon" }, h("tooltip-component", null, h("info-icon", { slot: "tooltip-icon" }), h("span", { slot: "tooltip-content" }, i18next.t('nylasConfirmationEmail.nylasBranding.tooltip.desc'))))), h("toggle-switch", { exportparts: "ts_label: nce__toggle-label, ts_input: nce_toggle-input, ts_slider: nce_toggle-slider", name: `confirmation-email-show-branding`, checked: this.confirmationEmail?.show_nylas_branding })), h("div", { class: "nylas-confirmation-email__row" }, h("div", { class: "subsection" }, h("div", { class: "input-container" }, h("label", null, i18next.t('nylasConfirmationEmail.companyLogoUrl.label'), h("span", { class: "label-icon" }, h("tooltip-component", null, h("info-icon", { slot: "tooltip-icon" }), h("span", { slot: "tooltip-content" }, i18next.t('nylasConfirmationEmail.companyLogoUrl.tooltip.desc'))))), this.confirmationEmail && (h("input-image-url", { name: "confirmation-email-logo", exportparts: "iiu__logo-input-textfield: nce__input-image-url", imageUrl: this.confirmationEmail?.logo ?? '' }))))), h("div", { class: "nylas-confirmation-email__row full-width-col" }, h("div", { class: "subsection" }, h("h3", null, i18next.t('nylasConfirmationEmail.emailMessage.header')), h("div", { class: "input-container" }, h("label", { htmlFor: "title" }, i18next.t('nylasConfirmationEmail.emailMessage.label'), h("span", { class: "label-icon" }, h("tooltip-component", null, h("info-icon", { slot: "tooltip-icon" }), h("span", { slot: "tooltip-content" }, i18next.t('nylasConfirmationEmail.emailMessage.tooltip.desc'))))), h("input-component", { name: "confirmation-email-title", maxLength: 60, placeholder: i18next.t('nylasConfirmationEmail.headerTitle', { organizerName: `${organizerName ? ` with ${organizerName}` : ''}` }), defaultValue: this.confirmationEmail?.booking_confirmed?.title ?? '' })), h("div", { class: "input-container" }, h("textarea-component", { label: i18next.t('nylasConfirmationEmail.emailMessage.additionalInfoLabel'), name: "confirmation-email-body", id: "confirmation-email-body", placeholder: i18next.t('nylasConfirmationEmail.emailMessage.additionalInfoPlaceholder'), tooltip: i18next.t('nylasConfirmationEmail.emailMessage.additionalInfoTooltip'), maxLength: 500, defaultValue: this.confirmationEmail?.booking_confirmed?.body ?? '' }))))))) : (h("div", { class: "nylas-confirmation-email__summary", part: "nce__summary" }, h("p", null, h("span", { class: "summary-icon" }, h("person-clipboard-icon", null)), i18next.t('nylasConfirmationEmail.allHosts')), h("p", null, h("span", { class: "summary-icon" }, h("clock-icon", null)), i18next.t('nylasConfirmationEmail.immediatelyUponBooking')))))));
    }
    static get is() { return "nylas-confirmation-email"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-confirmation-email.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-confirmation-email.css"]
        };
    }
    static get properties() {
        return {
            "selectedConfiguration": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "Configuration",
                    "resolved": "Configuration | undefined",
                    "references": {
                        "Configuration": {
                            "location": "import",
                            "path": "@nylas/core",
                            "id": "../nylas-js-core/dist/index.d.ts::Configuration"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [{
                            "name": "standalone",
                            "text": "The selected config"
                        }],
                    "text": ""
                }
            },
            "confirmationEmailTemplate": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "EmailTemplate",
                    "resolved": "EmailTemplate",
                    "references": {
                        "EmailTemplate": {
                            "location": "import",
                            "path": "@nylas/core",
                            "id": "../nylas-js-core/dist/index.d.ts::EmailTemplate"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "standalone",
                            "text": "The confirmation email template data to display"
                        }],
                    "text": ""
                },
                "defaultValue": "{} as EmailTemplate"
            },
            "participants": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "Participant[]",
                    "resolved": "Participant[]",
                    "references": {
                        "Participant": {
                            "location": "import",
                            "path": "@nylas/core",
                            "id": "../nylas-js-core/dist/index.d.ts::Participant"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "standalone",
                            "text": "List of participants"
                        }],
                    "text": ""
                },
                "defaultValue": "[]"
            },
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [{
                            "name": "standalone",
                            "text": "The name of the component"
                        }],
                    "text": ""
                },
                "attribute": "name",
                "reflect": false,
                "defaultValue": "'confirmation-email-template'"
            },
            "isOpen": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Is the confirmation email card open"
                },
                "attribute": "is-open",
                "reflect": false,
                "defaultValue": "false"
            },
            "selectedLanguage": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "LANGUAGE_CODE",
                    "resolved": "LANGUAGE_CODE.de | LANGUAGE_CODE.en | LANGUAGE_CODE.es | LANGUAGE_CODE.fr | LANGUAGE_CODE.ja | LANGUAGE_CODE.nl | LANGUAGE_CODE.sv | LANGUAGE_CODE.zh | undefined",
                    "references": {
                        "LANGUAGE_CODE": {
                            "location": "import",
                            "path": "@/common/constants",
                            "id": "src/common/constants.ts::LANGUAGE_CODE"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The selected language."
                },
                "attribute": "selected-language",
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "confirmationEmail": {},
            "isConfirmationEmailOpen": {},
            "participantsState": {}
        };
    }
    static get events() {
        return [{
                "method": "valueChanged",
                "name": "valueChanged",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the email reminders change."
                },
                "complexType": {
                    "original": "{\n    value: string;\n    name: string;\n    valueChanged?: (event: CustomEvent<{ value: string; name: string }>) => void;\n  }",
                    "resolved": "{ value: string; name: string; valueChanged?: ((event: CustomEvent<{ value: string; name: string; }>) => void) | undefined; }",
                    "references": {
                        "CustomEvent": {
                            "location": "global",
                            "id": "global::CustomEvent"
                        }
                    }
                }
            }];
    }
    static get elementRef() { return "host"; }
    static get watchers() {
        return [{
                "propName": "name",
                "methodName": "elementNameChangedHandler"
            }, {
                "propName": "selectedConfiguration",
                "methodName": "selectedConfigurationChangedHandler"
            }, {
                "propName": "selectedLanguage",
                "methodName": "selectedLanguageChangedHandler"
            }];
    }
    static get listeners() {
        return [{
                "name": "nylasFormInputChanged",
                "method": "nylasFormInputChangeHandler",
                "target": undefined,
                "capture": false,
                "passive": false
            }, {
                "name": "valueChanged",
                "method": "valueChangedHandler",
                "target": undefined,
                "capture": false,
                "passive": false
            }, {
                "name": "nylasFormInputImageUrlInvalid",
                "method": "nylasFormInputImageUrlInvalidHandler",
                "target": undefined,
                "capture": false,
                "passive": false
            }, {
                "name": "nylasFormSwitchToggled",
                "method": "nyalsFormSwitchToggledHandler",
                "target": undefined,
                "capture": false,
                "passive": false
            }];
    }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-confirmation-email',
        stateToProps: new Map([
            ['schedulerConfig.selectedConfiguration', 'selectedConfiguration'],
            ['schedulerConfig.selectedLanguage', 'selectedLanguage'],
        ]),
        eventToProps: {
            valueChanged: async (event, _nylasSchedulerConfigConnector) => {
                const { valueChanged } = event.detail;
                if (valueChanged) {
                    valueChanged(event);
                }
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasConfirmationEmail.prototype, "render", null);
