// @nylas/react/dist/nylas-if-state.entry-fdc20d01.js
import { r as h, d as o, h as l, H as u } from "./index-cd93b584.js";
import { R as m } from "./register-component-a169494c-62768a00.js";
import "react";
import "react-dom";
const p = ":host{display:block}div{height:inherit}@keyframes fadeIn{0%{opacity:0}100%{opacity:1}}@keyframes fadeOut{0%{opacity:1}100%{opacity:0}}.fade-in{animation:fadeIn 1.5s}.fade-out{animation:fadeOut 1.5s}@keyframes slideIn{0%{transform:translateY(100%)}100%{transform:translateY(0)}}@keyframes slideOut{0%{transform:translateY(0)}100%{transform:translateY(100%)}}.slide-in{animation:slideIn 1.5s}.slide-out{animation:slideOut 1.5s}@keyframes swipeIn{0%{transform:translateX(100%)}100%{transform:translateX(0)}}@keyframes swipeOut{0%{transform:translateX(0)}100%{transform:translateX(100%)}}.swipe-in{animation:swipeIn 1.5s}.swipe-out{animation:swipeOut 1.5s}", y = p;
var S = function(e, s, n, a) {
  var t = arguments.length, i = t < 3 ? s : a === null ? a = Object.getOwnPropertyDescriptor(s, n) : a, r;
  if (typeof Reflect == "object" && typeof Reflect.decorate == "function")
    i = Reflect.decorate(e, s, n, a);
  else
    for (var f = e.length - 1; f >= 0; f--)
      (r = e[f]) && (i = (t < 3 ? r(i) : t > 3 ? r(s, n, i) : r(s, n)) || i);
  return t > 3 && i && Object.defineProperty(s, n, i), i;
}, d = function(e, s) {
  if (typeof Reflect == "object" && typeof Reflect.metadata == "function")
    return Reflect.metadata(e, s);
};
const c = class {
  constructor(e) {
    h(this, e), this.state = void 0, this.isStateSet = void 0, this.isBoolean = void 0, this.getStore = void 0, this.transition = "none", this.show = !1, this.classes = {}, this.initTransition = this.initTransition.bind(this);
  }
  connectedCallback() {
  }
  disconnectedCallback() {
  }
  async componentWillLoad() {
    o("[nylas-if-state] Component will load"), this.watchStore(!0);
  }
  async componentDidLoad() {
    o("[nylas-if-state] Component did load");
  }
  shouldRender() {
    const [e, s] = this.state ? this.state.split(".") : ["", ""], n = this.getStore;
    if (typeof n > "u")
      return o(`[nylas-if-state] Store name: ${e} | State name: ${s}`, { getStore: n }), !1;
    const a = n(e);
    if (typeof a > "u")
      return o(`[nylas-if-state] Store name: ${e} | State name: ${s}`, { store: a }), !1;
    const t = a.state[s];
    return typeof this.isStateSet == "boolean" ? this.isStateSet === !1 && (typeof t > "u" || t === null || t.length === 0) || this.isStateSet === !0 && typeof t < "u" && t !== null : typeof this.isBoolean == "boolean" ? this.isBoolean === !0 && typeof t < "u" && t === !0 || this.isBoolean === !1 && typeof t < "u" && t === !1 : !1;
  }
  onGetStoreChange() {
    this.watchStore(!0);
  }
  watchStore(e = !1) {
    const [s, n] = this.state ? this.state.split(".") : ["", ""], a = this.getStore;
    if (typeof a > "u")
      return o(`[nylas-if-state] Store name: ${s} | State name: ${n}`, { getStore: a }), null;
    const t = a(s);
    if (typeof t > "u")
      return o(`[nylas-if-state] Store name: ${s} | State name: ${n}`, { store: t }), null;
    e === !0 && this.initTransition(), t.onChange(n, this.initTransition);
  }
  initTransition() {
    const e = this.shouldRender();
    switch (this.transition) {
      case "fade":
      case "slide":
      case "swipe":
        this.classes = {
          [`${this.transition}-in`]: e,
          [`${this.transition}-out`]: !e
        }, setTimeout(() => {
          this.show = e;
        }, 150);
        break;
      default:
        this.show = this.shouldRender();
        break;
    }
  }
  render() {
    return l(u, { key: "36d3cd4c91d34f8716528990469a2b015b4e3f72" }, l("div", { key: "2f598550f94fcb8e92e783f93c384703a6229776", class: this.classes }, this.show && l("slot", null)));
  }
  static get watchers() {
    return {
      getStore: ["onGetStoreChange"]
    };
  }
};
S([
  m({
    name: "nylas-if-state",
    getStoresToProp: "getStore",
    fireRegisterEvent: !0
  }),
  d("design:type", Function),
  d("design:paramtypes", []),
  d("design:returntype", void 0)
], c.prototype, "render", null);
c.style = y;
export {
  c as nylas_if_state
};
