// @nylas/react/dist/nylas-form-card.entry-8842785d.js
import { r as i, d as o, K as y, h as n, H as m, g as p } from "./index-cd93b584.js";
import { R as u } from "./register-component-a169494c-62768a00.js";
import "react";
import "react-dom";
const h = ':host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:"Inter", sans-serif}.nylas-form-card{width:100%;border-radius:8px;border:1px solid var(--nylas-base-200)}.nylas-form-card__header{padding:1rem;border-bottom:1px solid var(--nylas-base-200)}', v = h;
var _ = function(e, a, t, r) {
  var l = arguments.length, s = l < 3 ? a : r === null ? r = Object.getOwnPropertyDescriptor(a, t) : r, c;
  if (typeof Reflect == "object" && typeof Reflect.decorate == "function")
    s = Reflect.decorate(e, a, t, r);
  else
    for (var d = e.length - 1; d >= 0; d--)
      (c = e[d]) && (s = (l < 3 ? c(s) : l > 3 ? c(a, t, s) : c(a, t)) || s);
  return l > 3 && s && Object.defineProperty(a, t, s), s;
}, f = function(e, a) {
  if (typeof Reflect == "object" && typeof Reflect.metadata == "function")
    return Reflect.metadata(e, a);
};
const b = class {
  constructor(e) {
    i(this, e), e.$hostElement$["s-ei"] ? this.internals = e.$hostElement$["s-ei"] : (this.internals = e.$hostElement$.attachInternals(), e.$hostElement$["s-ei"] = this.internals);
  }
  connectedCallback() {
    o("nylas-event-card", "connectedCallback");
  }
  componentWillLoad() {
    o("nylas-event-card", "componentWillLoad");
  }
  componentDidLoad() {
    o("nylas-event-card", "componentDidLoad");
  }
  disconnectedCallback() {
    o("nylas-event-card", "disconnectedCallback"), y(["header-title", "header-subtitle", "content"], this.host);
  }
  render() {
    return n(m, { key: "9c490244ceaa1c08fae75cad5f796e2b45ea2538" }, n("div", { key: "696b86c8697f4062afcd8a5862c97bf3bf5492e0", class: "nylas-form-card", part: "nfc" }, n("div", { key: "af1f2eb827a74586fe44b476f1d59f9f8785e980", class: "nylas-form-card__header", part: "nfc__header" }, n("slot", { key: "968b720f04d9e64df5708fd1c4d94a9998aa72fc", name: "header-title" }), n("slot", { key: "4df6bddacc17d303f3be42584be48e50bf7fb0f0", name: "header-subtitle" })), n("div", { key: "890fbc0f1628e40a2b92201bb3e4ebc116ed2f9e", class: "nylas-form-card__content" }, n("slot", { key: "861b981bb7a420151fc375cc68dcdcd58ffbb8c2", name: "content" }))));
  }
  static get formAssociated() {
    return !0;
  }
  get host() {
    return p(this);
  }
};
_([
  u({
    name: "nylas-form-card",
    eventToProps: {},
    fireRegisterEvent: !0
  }),
  f("design:type", Function),
  f("design:paramtypes", []),
  f("design:returntype", void 0)
], b.prototype, "render", null);
b.style = v;
export {
  b as nylas_form_card
};
