// @nylas/react/dist/react.js
import { Z as e, _ as l, $ as i, a0 as c, a1 as t, a2 as r, a3 as b, a4 as I, a5 as y, a6 as N, a7 as d, a8 as C, a9 as m, aa as p, ab as g, ac as k, ad as u, ae as P, af as f, ag as E, ah as h, ai as v, aj as S, ak as B, al as F, am as T, an as L, ao as w, ap as D, aq as A, ar as R, as as M, at as x, au as G, av as W, aw as z, ax as O, ay as U, az as q, aA as V, aB as X, aC as Z, aD as j, aE as H, aF as J, aG as K, aH as Q, aI as Y, aJ as _, aK as $, aL as aa, aM as sa, aN as oa, aO as na, aP as ea, aQ as la, aR as ia, aS as ca, aT as ta, aU as ra, aV as ba, aW as Ia, aX as ya, aY as Na, aZ as da, a_ as Ca, a$ as ma, b0 as pa, b1 as ga, b2 as ka, b3 as ua, b4 as Pa, b5 as fa, b6 as Ea, b7 as ha, b8 as va, N as Sa, b9 as Ba, ba as Fa, bb as Ta, bc as La, bd as wa, be as Da, bf as Aa, bg as Ra, bh as Ma, bi as xa, bj as Ga, bk as Wa, bl as za, bm as Oa, bn as Ua, bo as qa, bp as Va, bq as Xa, br as Za, bs as ja, bt as Ha, bu as Ja, bv as Ka, bw as Qa, bx as Ya, by as _a, bz as $a, bA as as, bB as ss, bC as os, bD as ns, bE as es, bF as ls, bG as is, bH as cs, bI as ts, bJ as rs, bK as bs, bL as Is, bM as ys, bN as Ns, bO as ds, bP as Cs, bQ as ms, bR as ps, bS as gs, bT as ks, bU as us, bV as Ps, bW as fs, bX as Es, W as hs, X as vs, Y as Ss, V as Bs } from "./index-cd93b584.js";
import "react";
import "react-dom";
export {
  e as AddCircleIcon,
  l as ArchiveIcon,
  i as ArrowIcon,
  c as BoldIcon,
  t as ButtonComponent,
  r as CalendarAgendaFillIcon,
  b as CalendarAgendaIcon,
  I as CalendarCancelIcon,
  y as CalendarCheckIcon,
  N as CalendarIcon,
  d as CalendarInfoIcon,
  C as CalendarPatternsIcon,
  m as CheckboxComponent,
  p as CheckmarkCircleIcon,
  g as CheckmarkIcon,
  k as ChevronIcon,
  u as ClockIcon,
  P as CloseIcon,
  f as CopyIcon,
  E as DeleteIcon,
  h as DocumentRefreshIcon,
  v as DragableIcon,
  S as EditIcon,
  B as EnvelopeFillIcon,
  F as EnvelopeIcon,
  T as EyeIcon,
  L as FeedbackIcon,
  w as FlowIcon,
  D as FolderIcon,
  A as ForwardIcon,
  R as GlobeIcon,
  M as GoogleLogoIcon,
  x as GoogleMeetIcon,
  G as InboxIcon,
  W as InfoIcon,
  z as InputColorPicker,
  O as InputComponent,
  U as InputDropdown,
  q as InputImageUrl,
  V as ItalicIcon,
  X as LoadingIcon,
  Z as LocationIcon,
  j as LocationOffIcon,
  H as MicrosoftLogoIcon,
  J as MicrosoftTeamsIcon,
  K as MultiSelectDropdown,
  Q as NylasAdditionalParticipants,
  Y as NylasAvailabilityPicker,
  _ as NylasBookedEventCard,
  $ as NylasBookingCalendarPicker,
  aa as NylasBookingConfirmationRedirect,
  sa as NylasBookingConfirmationType,
  oa as NylasBookingForm,
  na as NylasBookingFormConfig,
  ea as NylasBufferTime,
  la as NylasCalendarPicker,
  ia as NylasCancelBookingForm,
  ca as NylasCancellationPolicy,
  ta as NylasCancelledEventCard,
  ra as NylasConfirmationEmail,
  ba as NylasConfirmedEventCard,
  Ia as NylasConnectedCalendars,
  ya as NylasCustomBookingFlow,
  Na as NylasCustomEventSlug,
  da as NylasCustomizeBookingSettings,
  Ca as NylasDateComponent,
  ma as NylasDatePicker,
  pa as NylasEditorTabs,
  ga as NylasEventDescription,
  ka as NylasEventDuration,
  ua as NylasEventInfo,
  Pa as NylasEventLimits,
  fa as NylasEventLocation,
  Ea as NylasEventTitle,
  ha as NylasFeedbackForm,
  va as NylasFormCard,
  Sa as NylasIdentityRequestWrapper,
  Ba as NylasIfState,
  Fa as NylasLimitFutureBookings,
  Ta as NylasListConfigurations,
  La as NylasLocaleSwitch,
  wa as NylasLogin,
  Da as NylasLogo,
  Aa as NylasMinBookingNotice,
  Ra as NylasMinCancellationNotice,
  Ma as NylasNotification,
  xa as NylasOrganizerConfirmationCard,
  Ga as NylasPageName,
  Wa as NylasPageStyling,
  za as NylasParticipantBookingCalendars,
  Oa as NylasParticipantsCustomAvailability,
  Ua as NylasProvider,
  qa as NylasReminderEmails,
  Va as NylasReminderTime,
  Xa as NylasSchedulerEditor,
  Za as NylasScheduling,
  ja as NylasSchedulingMethod,
  Ha as NylasSelectedEventCard,
  Ja as NylasTimeWindowPicker,
  Ka as NylasTimeslotInterval,
  Qa as NylasTimeslotPicker,
  Ya as PaintbrushFillIcon,
  _a as PaintbrushIcon,
  $a as PeopleIcon,
  as as PersonClipboardIcon,
  ss as PersonIcon,
  os as PlayIcon,
  ns as PlusIcon,
  es as RadioButtonGroup,
  ls as RefreshIcon,
  is as ReplyAllIcon,
  cs as ReplyIcon,
  ts as SearchIcon,
  rs as SelectDropdown,
  bs as SentIcon,
  Is as SpamIcon,
  ys as StarIcon,
  Ns as StopIcon,
  ds as TextareaComponent,
  Cs as TimePeriodSelector,
  ms as ToggleSwitch,
  ps as TooltipComponent,
  gs as TranslateIcon,
  ks as TrashFillIcon,
  us as TrashIcon,
  Ps as UnderlineIcon,
  fs as WarningIcon,
  Es as ZoomIcon,
  hs as applyPolyfills,
  vs as defineCustomElements,
  Ss as setNonce,
  Bs as useNylasMailboxElement
};
