// @nylas/web-elements/dist/collection/components/scheduler-editor/nylas-form-card/nylas-form-card.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
export class NylasFormCard {
    connectedCallback() {
        debug('nylas-event-card', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-card', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-card', 'componentDidLoad');
    }
    disconnectedCallback() {
        debug('nylas-event-card', 'disconnectedCallback');
        const slots = ['header-title', 'header-subtitle', 'content'];
        checkForMissingSlots(slots, this.host);
    }
    render() {
        return (h(Host, { key: '9c490244ceaa1c08fae75cad5f796e2b45ea2538' }, h("div", { key: '696b86c8697f4062afcd8a5862c97bf3bf5492e0', class: "nylas-form-card", part: "nfc" }, h("div", { key: 'af1f2eb827a74586fe44b476f1d59f9f8785e980', class: "nylas-form-card__header", part: "nfc__header" }, h("slot", { key: '968b720f04d9e64df5708fd1c4d94a9998aa72fc', name: "header-title" }), h("slot", { key: '4df6bddacc17d303f3be42584be48e50bf7fb0f0', name: "header-subtitle" })), h("div", { key: '890fbc0f1628e40a2b92201bb3e4ebc116ed2f9e', class: "nylas-form-card__content" }, h("slot", { key: '861b981bb7a420151fc375cc68dcdcd58ffbb8c2', name: "content" })))));
    }
    static get is() { return "nylas-form-card"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-form-card.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-form-card.css"]
        };
    }
    static get elementRef() { return "host"; }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-form-card',
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasFormCard.prototype, "render", null);
