// @nylas/web-elements/dist/cdn/nylas-logo/nylas-logo.es.js
const L = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, B, A, oe = !1, w = !1, W = !1, v = !1, q = null, D = !1;
const T = (e, t = "") => () => {
}, Se = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, be = "http://www.w3.org/2000/svg", ke = "http://www.w3.org/1999/xhtml", Le = (e) => e != null, z = (e) => (e = typeof e, e === "object" || e === "function");
function me(e) {
  var t, n, s;
  return (s = (n = (t = e.head) === null || t === void 0 ? void 0 : t.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const p = (e, t, ...n) => {
  let s = null, l = null, c = null, o = !1, r = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((o = typeof e != "function" && !z(s)) && (s = String(s)), o && r ? i[i.length - 1].$text$ += s : i.push(o ? F(null, s) : s), r = o);
  };
  if (u(n), t) {
    t.key && (l = t.key), t.name && (c = t.name);
    {
      const a = t.className || t.class;
      a && (t.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof e == "function")
    return e(t === null ? {} : t, i, Ee);
  const f = F(e, null);
  return f.$attrs$ = t, i.length > 0 && (f.$children$ = i), f.$key$ = l, f.$name$ = c, f;
}, F = (e, t) => {
  const n = {
    $flags$: 0,
    $tag$: e,
    $text$: t,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Te = {}, xe = (e) => e && e.$tag$ === Te, Ee = {
  forEach: (e, t) => e.map(G).forEach(t),
  map: (e, t) => e.map(G).map(t).map(_e)
}, G = (e) => ({
  vattrs: e.$attrs$,
  vchildren: e.$children$,
  vkey: e.$key$,
  vname: e.$name$,
  vtag: e.$tag$,
  vtext: e.$text$
}), _e = (e) => {
  if (typeof e.vtag == "function") {
    const n = Object.assign({}, e.vattrs);
    return e.vkey && (n.key = e.vkey), e.vname && (n.name = e.vname), p(e.vtag, n, ...e.vchildren || []);
  }
  const t = F(e.vtag, e.vtext);
  return t.$attrs$ = e.vattrs, t.$children$ = e.vchildren, t.$key$ = e.vkey, t.$name$ = e.vname, t;
}, Ae = (e) => st.map((t) => t(e)).find((t) => !!t), Re = (e, t) => e != null && !z(e) ? t & 4 ? e === "false" ? !1 : e === "" || !!e : t & 2 ? parseFloat(e) : t & 1 ? String(e) : e : e, K = /* @__PURE__ */ new WeakMap(), Oe = (e, t, n) => {
  let s = U.get(e);
  it && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = t : s.replaceSync(t)) : s = t, U.set(e, s);
}, He = (e, t, n) => {
  var s;
  const l = ie(t, n), c = U.get(l);
  if (e = e.nodeType === 11 ? e : S, c)
    if (typeof c == "string") {
      e = e.head || e;
      let o = K.get(e), r;
      if (o || K.set(e, o = /* @__PURE__ */ new Set()), !o.has(l)) {
        {
          r = S.createElement("style"), r.innerHTML = c;
          const i = (s = g.$nonce$) !== null && s !== void 0 ? s : me(S);
          i != null && r.setAttribute("nonce", i), e.insertBefore(r, e.querySelector("link"));
        }
        t.$flags$ & 4 && (r.innerHTML += Se), o && o.add(l);
      }
    } else e.adoptedStyleSheets.includes(c) || (e.adoptedStyleSheets = [...e.adoptedStyleSheets, c]);
  return l;
}, Me = (e) => {
  const t = e.$cmpMeta$, n = e.$hostElement$, s = t.$flags$, l = T("attachStyles", t.$tagName$), c = He(n.shadowRoot ? n.shadowRoot : n.getRootNode(), t, e.$modeName$);
  s & 10 && (n["s-sc"] = c, n.classList.add(c + "-h"), s & 2 && n.classList.add(c + "-s")), l();
}, ie = (e, t) => "sc-" + (t && e.$flags$ & 32 ? e.$tagName$ + "-" + t : e.$tagName$), I = (e, t, n, s, l, c) => {
  if (n !== s) {
    let o = te(e, t), r = t.toLowerCase();
    if (t === "class") {
      const i = e.classList, u = J(n), f = J(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (t === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? e.style.removeProperty(i) : e.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? e.style.setProperty(i, s[i]) : e.style[i] = s[i]);
    } else if (t !== "key")
      if (t === "ref")
        s && s(e);
      else if (!e.__lookupSetter__(t) && t[0] === "o" && t[1] === "n") {
        if (t[2] === "-" ? t = t.slice(3) : te(j, r) ? t = r.slice(2) : t = r[2] + t.slice(3), n || s) {
          const i = t.endsWith(ce);
          t = t.replace(Fe, ""), n && g.rel(e, t, n, i), s && g.ael(e, t, s, i);
        }
      } else {
        const i = z(s);
        if ((o || i && s !== null) && !l)
          try {
            if (e.tagName.includes("-"))
              e[t] = s;
            else {
              const f = s ?? "";
              t === "list" ? o = !1 : (n == null || e[t] != f) && (e[t] = f);
            }
          } catch {
          }
        let u = !1;
        r !== (r = r.replace(/^xlink\:?/, "")) && (t = r, u = !0), s == null || s === !1 ? (s !== !1 || e.getAttribute(t) === "") && (u ? e.removeAttributeNS(Q, t) : e.removeAttribute(t)) : (!o || c & 4 || l) && !i && (s = s === !0 ? "" : s, u ? e.setAttributeNS(Q, t, s) : e.setAttribute(t, s));
      }
  }
}, we = /\s/, J = (e) => e ? e.split(we) : [], ce = "Capture", Fe = new RegExp(ce + "$"), re = (e, t, n, s) => {
  const l = t.$elm$.nodeType === 11 && t.$elm$.host ? t.$elm$.host : t.$elm$, c = e && e.$attrs$ || X, o = t.$attrs$ || X;
  for (s in c)
    s in o || I(l, s, c[s], void 0, n, t.$flags$);
  for (s in o)
    I(l, s, c[s], o[s], n, t.$flags$);
}, P = (e, t, n, s) => {
  const l = t.$children$[n];
  let c = 0, o, r, i;
  if (oe || (W = !0, l.$tag$ === "slot" && (x && s.classList.add(x + "-s"), l.$flags$ |= l.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), l.$text$ !== null)
    o = l.$elm$ = S.createTextNode(l.$text$);
  else if (l.$flags$ & 1)
    o = l.$elm$ = S.createTextNode("");
  else {
    if (v || (v = l.$tag$ === "svg"), o = l.$elm$ = S.createElementNS(v ? be : ke, l.$flags$ & 2 ? "slot-fb" : l.$tag$), v && l.$tag$ === "foreignObject" && (v = !1), re(null, l, v), Le(x) && o["s-si"] !== x && o.classList.add(o["s-si"] = x), l.$children$)
      for (c = 0; c < l.$children$.length; ++c)
        r = P(e, l, c, o), r && o.appendChild(r);
    l.$tag$ === "svg" ? v = !1 : o.tagName === "foreignObject" && (v = !0);
  }
  return o["s-hn"] = A, l.$flags$ & 3 && (o["s-sr"] = !0, o["s-cr"] = B, o["s-sn"] = l.$name$ || "", i = e && e.$children$ && e.$children$[n], i && i.$tag$ === l.$tag$ && e.$elm$ && R(e.$elm$, !1)), o;
}, R = (e, t) => {
  g.$flags$ |= 1;
  const n = e.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const l = n[s];
    l["s-hn"] !== A && l["s-ol"] && (fe(l).insertBefore(l, Z(l)), l["s-ol"].remove(), l["s-ol"] = void 0, l["s-sh"] = void 0, W = !0), t && R(l, t);
  }
  g.$flags$ &= -2;
}, ae = (e, t, n, s, l, c) => {
  let o = e["s-cr"] && e["s-cr"].parentNode || e, r;
  for (o.shadowRoot && o.tagName === A && (o = o.shadowRoot); l <= c; ++l)
    s[l] && (r = P(null, n, l, e), r && (s[l].$elm$ = r, o.insertBefore(r, Z(t))));
}, $e = (e, t, n) => {
  for (let s = t; s <= n; ++s) {
    const l = e[s];
    if (l) {
      const c = l.$elm$;
      he(l), c && (w = !0, c["s-ol"] ? c["s-ol"].remove() : R(c, !0), c.remove());
    }
  }
}, Pe = (e, t, n, s, l = !1) => {
  let c = 0, o = 0, r = 0, i = 0, u = t.length - 1, f = t[0], a = t[u], d = s.length - 1, $ = s[0], h = s[d], C, y;
  for (; c <= u && o <= d; )
    if (f == null)
      f = t[++c];
    else if (a == null)
      a = t[--u];
    else if ($ == null)
      $ = s[++o];
    else if (h == null)
      h = s[--d];
    else if (M(f, $, l))
      E(f, $, l), f = t[++c], $ = s[++o];
    else if (M(a, h, l))
      E(a, h, l), a = t[--u], h = s[--d];
    else if (M(f, h, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && R(f.$elm$.parentNode, !1), E(f, h, l), e.insertBefore(f.$elm$, a.$elm$.nextSibling), f = t[++c], h = s[--d];
    else if (M(a, $, l))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && R(a.$elm$.parentNode, !1), E(a, $, l), e.insertBefore(a.$elm$, f.$elm$), a = t[--u], $ = s[++o];
    else {
      for (r = -1, i = c; i <= u; ++i)
        if (t[i] && t[i].$key$ !== null && t[i].$key$ === $.$key$) {
          r = i;
          break;
        }
      r >= 0 ? (y = t[r], y.$tag$ !== $.$tag$ ? C = P(t && t[o], n, r, e) : (E(y, $, l), t[r] = void 0, C = y.$elm$), $ = s[++o]) : (C = P(t && t[o], n, o, e), $ = s[++o]), C && fe(f.$elm$).insertBefore(C, Z(f.$elm$));
    }
  c > u ? ae(e, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, o, d) : o > d && $e(t, c, u);
}, M = (e, t, n = !1) => e.$tag$ === t.$tag$ ? e.$tag$ === "slot" ? e.$name$ === t.$name$ : n ? !0 : e.$key$ === t.$key$ : !1, Z = (e) => e && e["s-ol"] || e, fe = (e) => (e["s-ol"] ? e["s-ol"] : e).parentNode, E = (e, t, n = !1) => {
  const s = t.$elm$ = e.$elm$, l = e.$children$, c = t.$children$, o = t.$tag$, r = t.$text$;
  let i;
  r === null ? (v = o === "svg" ? !0 : o === "foreignObject" ? !1 : v, o === "slot" || re(e, t, v), l !== null && c !== null ? Pe(s, l, t, c, n) : c !== null ? (e.$text$ !== null && (s.textContent = ""), ae(s, null, t, c, 0, c.length - 1)) : l !== null && $e(l, 0, l.length - 1), v && o === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = r : e.$text$ !== r && (s.data = r);
}, de = (e) => {
  const t = e.childNodes;
  for (const n of t)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const l of t)
          if (l !== n) {
            if (l["s-hn"] !== n["s-hn"] || s !== "") {
              if (l.nodeType === 1 && (s === l.getAttribute("slot") || s === l["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (l.nodeType === 1 || l.nodeType === 3 && l.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      de(n);
    }
}, b = [], ue = (e) => {
  let t, n, s;
  for (const l of e.childNodes) {
    if (l["s-sr"] && (t = l["s-cr"]) && t.parentNode) {
      n = t.parentNode.childNodes;
      const c = l["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (t = n[s], !t["s-cn"] && !t["s-nr"] && t["s-hn"] !== l["s-hn"] && !L.experimentalSlotFixes)
          if (Y(t, c)) {
            let o = b.find((r) => r.$nodeToRelocate$ === t);
            w = !0, t["s-sn"] = t["s-sn"] || c, o ? (o.$nodeToRelocate$["s-sh"] = l["s-hn"], o.$slotRefNode$ = l) : (t["s-sh"] = l["s-hn"], b.push({
              $slotRefNode$: l,
              $nodeToRelocate$: t
            })), t["s-sr"] && b.map((r) => {
              Y(r.$nodeToRelocate$, t["s-sn"]) && (o = b.find((i) => i.$nodeToRelocate$ === t), o && !r.$slotRefNode$ && (r.$slotRefNode$ = o.$slotRefNode$));
            });
          } else b.some((o) => o.$nodeToRelocate$ === t) || b.push({
            $nodeToRelocate$: t
          });
    }
    l.nodeType === 1 && ue(l);
  }
}, Y = (e, t) => e.nodeType === 1 ? e.getAttribute("slot") === null && t === "" || e.getAttribute("slot") === t : e["s-sn"] === t ? !0 : t === "", he = (e) => {
  e.$attrs$ && e.$attrs$.ref && e.$attrs$.ref(null), e.$children$ && e.$children$.map(he);
}, Ue = (e, t, n = !1) => {
  var s, l, c, o, r;
  const i = e.$hostElement$, u = e.$cmpMeta$, f = e.$vnode$ || F(null, null), a = xe(t) ? t : p(null, null, t);
  if (A = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, e.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], B = i["s-cr"], oe = (u.$flags$ & 1) !== 0, w = !1, E(f, a, n);
  {
    if (g.$flags$ |= 1, W) {
      ue(a.$elm$);
      for (const d of b) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of b) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const C = h.parentNode;
          let y = h.nextSibling;
          {
            let m = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; m; ) {
              let k = (l = m["s-nr"]) !== null && l !== void 0 ? l : null;
              if (k && k["s-sn"] === $["s-sn"] && C === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                y = k;
                break;
              }
              m = m.previousSibling;
            }
          }
          (!y && C !== $.parentNode || $.nextSibling !== y) && $ !== y && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), C.insertBefore($, y), $.nodeType === 1 && ($.hidden = (c = $["s-ih"]) !== null && c !== void 0 ? c : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (o = $.hidden) !== null && o !== void 0 ? o : !1), $.hidden = !0);
      }
    }
    w && de(a.$elm$), g.$flags$ &= -2, b.length = 0;
  }
  if (L.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== A && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (r = d.hidden) !== null && r !== void 0 ? r : !1), d.hidden = !0);
  B = void 0;
}, je = (e, t) => {
}, pe = (e, t) => (e.$flags$ |= 16, je(e, e.$ancestorComponent$), at(() => Be(e, t))), Be = (e, t) => {
  const n = e.$hostElement$, s = T("scheduleUpdate", e.$cmpMeta$.$tagName$), l = n;
  let c;
  return t ? c = _(l, "componentWillLoad") : c = _(l, "componentWillUpdate"), c = V(c, () => _(l, "componentWillRender")), s(), V(c, () => Ne(e, l, t));
}, V = (e, t) => De(e) ? e.then(t) : t(), De = (e) => e instanceof Promise || e && e.then && typeof e.then == "function", Ne = async (e, t, n) => {
  const s = e.$hostElement$, l = T("update", e.$cmpMeta$.$tagName$);
  s["s-rc"], n && Me(e);
  const c = T("render", e.$cmpMeta$.$tagName$);
  We(e, t, s, n), c(), l(), ze(e);
}, We = (e, t, n, s) => {
  try {
    q = t, t = t.render && t.render(), e.$flags$ &= -17, e.$flags$ |= 2, (L.hasRenderFn || L.reflect) && (L.vdomRender || L.reflect) && (L.hydrateServerSide || Ue(e, t, s));
  } catch (i) {
    H(i, e.$hostElement$);
  }
  return q = null, null;
}, ze = (e) => {
  const t = e.$cmpMeta$.$tagName$, n = e.$hostElement$, s = T("postUpdate", t), l = n;
  e.$ancestorComponent$, _(l, "componentDidRender"), e.$flags$ & 64 ? (_(l, "componentDidUpdate"), s()) : (e.$flags$ |= 64, _(l, "componentDidLoad"), s());
}, _ = (e, t, n) => {
  if (e && e[t])
    try {
      return e[t](n);
    } catch (s) {
      H(s);
    }
}, Ze = (e, t) => O(e).$instanceValues$.get(t), qe = (e, t, n, s) => {
  const l = O(e), c = e, o = l.$instanceValues$.get(t), r = l.$flags$, i = c;
  n = Re(n, s.$members$[t][0]);
  const u = Number.isNaN(o) && Number.isNaN(n);
  if (n !== o && !u) {
    l.$instanceValues$.set(t, n);
    {
      if (s.$watchers$ && r & 128) {
        const a = s.$watchers$[t];
        a && a.map((d) => {
          try {
            i[d](n, o, t);
          } catch ($) {
            H($, c);
          }
        });
      }
      if ((r & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, o, t) === !1)
          return;
        pe(l, !1);
      }
    }
  }
}, Qe = (e, t, n) => {
  var s;
  const l = e.prototype;
  if (t.$members$) {
    e.watchers && (t.$watchers$ = e.watchers);
    const c = Object.entries(t.$members$);
    c.map(([o, [r]]) => {
      (r & 31 || r & 32) && Object.defineProperty(l, o, {
        get() {
          return Ze(this, o);
        },
        set(i) {
          qe(this, o, i, t);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const o = /* @__PURE__ */ new Map();
      l.attributeChangedCallback = function(r, i, u) {
        g.jmp(() => {
          var f;
          const a = o.get(r);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (l.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = O(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const C = this, y = (f = t.$watchers$) === null || f === void 0 ? void 0 : f[r];
                y == null || y.forEach((m) => {
                  C[m] != null && C[m].call(C, u, i, r);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, e.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = t.$watchers$) !== null && s !== void 0 ? s : {}),
        ...c.filter(
          ([r, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([r, i]) => {
          var u;
          const f = i[1] || r;
          return o.set(f, r), i[0] & 512 && ((u = t.$attrsToReflect$) === null || u === void 0 || u.push([r, f])), f;
        })
      ]));
    }
  }
  return e;
}, Xe = async (e, t, n, s) => {
  let l;
  if (!(t.$flags$ & 32) && (t.$flags$ |= 32, l = e.constructor, customElements.whenDefined(n.$tagName$).then(() => t.$flags$ |= 128), l.style)) {
    let o = l.style;
    typeof o != "string" && (o = o[t.$modeName$ = Ae(e)]);
    const r = ie(n, t.$modeName$);
    if (!U.has(r)) {
      const i = T("registerStyles", n.$tagName$);
      Oe(r, o, !!(n.$flags$ & 1)), i();
    }
  }
  t.$ancestorComponent$, pe(t, !0);
}, ee = (e) => {
}, Ge = (e) => {
  if (!(g.$flags$ & 1)) {
    const t = O(e), n = t.$cmpMeta$, s = T("connectedCallback", n.$tagName$);
    t.$flags$ & 1 ? (ge(e, t, n.$listeners$), t != null && t.$lazyInstance$ ? ee(t.$lazyInstance$) : t != null && t.$onReadyPromise$ && t.$onReadyPromise$.then(() => ee(t.$lazyInstance$))) : (t.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Ke(e), n.$members$ && Object.entries(n.$members$).map(([l, [c]]) => {
      if (c & 31 && e.hasOwnProperty(l)) {
        const o = e[l];
        delete e[l], e[l] = o;
      }
    }), Xe(e, t, n)), s();
  }
}, Ke = (e) => {
  const t = e["s-cr"] = S.createComment("");
  t["s-cn"] = !0, e.insertBefore(t, e.firstChild);
}, Ie = async (e) => {
  if (!(g.$flags$ & 1)) {
    const t = O(e);
    t.$rmListeners$ && (t.$rmListeners$.map((n) => n()), t.$rmListeners$ = void 0);
  }
}, Je = (e, t) => {
  const n = {
    $flags$: t[0],
    $tagName$: t[1]
  };
  n.$members$ = t[2], n.$listeners$ = t[3], n.$watchers$ = e.$watchers$, n.$attrsToReflect$ = [], !se && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = e.prototype.connectedCallback, l = e.prototype.disconnectedCallback;
  return Object.assign(e.prototype, {
    __registerHost() {
      tt(this, n);
    },
    connectedCallback() {
      Ge(this), s && s.call(this);
    },
    disconnectedCallback() {
      Ie(this), l && l.call(this);
    },
    __attachShadow() {
      se ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), e.is = n.$tagName$, Qe(e, n);
}, ge = (e, t, n, s) => {
  n && n.map(([l, c, o]) => {
    const r = Ve(e, l), i = Ye(t, o), u = et(l);
    g.ael(r, c, i, u), (t.$rmListeners$ = t.$rmListeners$ || []).push(() => g.rel(r, c, i, u));
  });
}, Ye = (e, t) => (n) => {
  try {
    L.lazyLoad || e.$hostElement$[t](n);
  } catch (s) {
    H(s);
  }
}, Ve = (e, t) => t & 4 ? S : t & 8 ? j : t & 16 ? S.body : e, et = (e) => lt ? {
  passive: (e & 1) !== 0,
  capture: (e & 2) !== 0
} : (e & 2) !== 0, ye = /* @__PURE__ */ new WeakMap(), O = (e) => ye.get(e), tt = (e, t) => {
  const n = {
    $flags$: 0,
    $hostElement$: e,
    $cmpMeta$: t,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return ge(e, n, t.$listeners$), ye.set(e, n);
}, te = (e, t) => t in e, H = (e, t) => (0, console.error)(e, t), U = /* @__PURE__ */ new Map(), st = [], j = typeof window < "u" ? window : {}, S = j.document || { head: {} }, nt = j.HTMLElement || class {
}, g = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (e) => e(),
  raf: (e) => requestAnimationFrame(e),
  ael: (e, t, n, s) => e.addEventListener(t, n, s),
  rel: (e, t, n, s) => e.removeEventListener(t, n, s),
  ce: (e, t) => new CustomEvent(e, t)
}, se = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), lt = /* @__PURE__ */ (() => {
  let e = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        e = !0;
      }
    }));
  } catch {
  }
  return e;
})(), ot = (e) => Promise.resolve(e), it = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), ne = [], Ce = [], ct = (e, t) => (n) => {
  e.push(n), D || (D = !0, t && g.$flags$ & 4 ? rt(N) : g.raf(N));
}, le = (e) => {
  for (let t = 0; t < e.length; t++)
    try {
      e[t](performance.now());
    } catch (n) {
      H(n);
    }
  e.length = 0;
}, N = () => {
  le(ne), le(Ce), (D = ne.length > 0) && g.raf(N);
}, rt = (e) => ot().then(e), at = /* @__PURE__ */ ct(Ce, !0), $t = ".sc-nylas-logo-h{display:flex}", ft = $t, ve = Je(class extends nt {
  constructor() {
    super(), this.__registerHost(), this.width = "69", this.height = "24";
  }
  render() {
    return p("svg", { key: "99d3ef972eb18b7beebcd264c3863762f20db756", width: this.width, height: this.height, viewBox: "0 0 69 20", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, p("g", { key: "855b23e638e91b1142b7fb47c17a4763d1e0302f", "clip-path": "url(#clip0_455_3543)" }, p("path", { key: "da34ed8ca958f7424e91b1414c4d4e339b7a703c", d: "M15.8092 3.03248C14.6497 1.86996 13.1076 1.23047 11.4646 1.23047C9.82156 1.23047 8.27753 1.86996 7.11999 3.03248L0.835347 9.33381C0.686846 9.4827 0.60498 9.68123 0.60498 9.89312C0.60498 10.105 0.686846 10.3035 0.835347 10.4524C0.983848 10.6013 1.18185 10.6834 1.39318 10.6834C1.60451 10.6834 1.80251 10.6013 1.95101 10.4524L8.23374 4.15111C10.0139 2.36627 12.9134 2.36627 14.6935 4.15111C16.4736 5.93785 16.4756 8.84322 14.6935 10.6281L8.4108 16.9275C8.10428 17.2348 8.10428 17.7368 8.4108 18.0461C8.5593 18.195 8.75731 18.2771 8.96863 18.2771C9.17996 18.2771 9.37796 18.195 9.52646 18.0461L15.8092 11.7467C16.9687 10.5841 17.6064 9.03793 17.6064 7.39053C17.6064 5.74314 16.9687 4.19501 15.8092 3.03439V3.03248Z", fill: "currentColor" }), p("path", { key: "bd3f95628660594d607d45a000a9c745db80fa74", d: "M14.9296 7.39038C14.9296 6.46074 14.5698 5.58837 13.9148 4.93361C12.5631 3.57828 10.3641 3.57828 9.01239 4.93361L0.663961 13.3023C0.51546 13.4512 0.433594 13.6497 0.433594 13.8616C0.433594 14.0735 0.51546 14.272 0.663961 14.4209C0.812462 14.5698 1.01046 14.6519 1.22179 14.6519C1.43312 14.6519 1.63112 14.5698 1.77962 14.4209L10.0367 6.1515L10.05 6.13623C10.0747 6.10759 10.1014 6.07705 10.128 6.05033C10.1585 6.01978 10.1928 5.99115 10.2251 5.96252L10.2442 5.94534C10.9886 5.31349 12.1119 5.3593 12.8011 6.05033C13.4903 6.74135 13.536 7.78171 12.9705 8.53192L12.9648 8.53955C12.9058 8.61591 12.8544 8.67699 12.8011 8.73044C12.7497 8.78198 12.6906 8.83352 12.6069 8.90033L4.45264 17.0991C4.30414 17.248 4.22227 17.4465 4.22227 17.6584C4.22227 17.8703 4.30414 18.0689 4.45264 18.2177C4.60114 18.3666 4.79914 18.4487 5.01047 18.4487C5.2218 18.4487 5.4198 18.3666 5.5683 18.2177L13.9167 9.84716C14.5698 9.1924 14.9315 8.31812 14.9315 7.39038H14.9296Z", fill: "currentColor" }), p("path", { key: "6dcc6994d434f6d92f6f8977a262072c5442425b", d: "M10.6821 9.11238C10.4727 9.01694 10.2861 8.88904 10.1281 8.7306C9.97008 8.57216 9.84252 8.38508 9.74733 8.1751L9.69021 8.05103L0.467916 17.2978C0.287049 17.4792 0.174721 17.7082 0.149971 17.9411C0.125221 18.1778 0.19376 18.3897 0.344165 18.5405C0.471724 18.6684 0.65259 18.739 0.850592 18.739C1.11142 18.739 1.37796 18.6207 1.58358 18.4145L10.8059 9.16774L10.6821 9.11047V9.11238Z", fill: "currentColor" }), p("path", { key: "16c56405f07d879815be6befad1770575c800d7b", d: "M68.401 10.6185C68.2126 10.2692 67.9593 9.97334 67.6433 9.73854C67.3235 9.49993 66.956 9.31094 66.5486 9.17541C66.1316 9.03606 65.7033 8.91771 65.2749 8.81844C64.7951 8.70582 64.3039 8.57983 63.8165 8.44811C63.3196 8.31258 62.9065 8.14269 62.5905 7.94225C62.4743 7.88689 62.3391 7.76663 62.3391 7.50129C62.3391 7.39821 62.3639 7.29704 62.4153 7.19587L62.4248 7.18059C62.5352 7.02216 62.7142 6.8828 62.975 6.75109C63.234 6.62128 63.609 6.55829 64.1231 6.55829C64.5286 6.55829 64.877 6.61365 65.1569 6.72436C65.4348 6.83508 65.6652 6.96871 65.8442 7.12524C66.0212 7.27986 66.1659 7.45357 66.2782 7.63873C66.3563 7.76854 66.4267 7.89262 66.4858 8.00525L68.2754 6.84081C68.1669 6.59456 68.0146 6.33495 67.8204 6.0677C67.6033 5.768 67.3254 5.4893 66.9922 5.23732C66.659 4.98534 66.2516 4.77345 65.7813 4.60929C65.3111 4.44512 64.7532 4.36304 64.1231 4.36304C63.3844 4.36304 62.7599 4.46421 62.2706 4.66083C61.7813 4.85935 61.3815 5.09988 61.0845 5.37667C60.7875 5.65346 60.5628 5.95125 60.4143 6.2605C60.2639 6.57547 60.1611 6.85799 60.1078 7.10233C60.0945 7.18059 60.0812 7.27031 60.0678 7.36767C60.0545 7.4593 60.0469 7.5452 60.0469 7.62537C60.0469 8.13314 60.1745 8.56074 60.4239 8.89671C60.679 9.24222 61.0045 9.53429 61.391 9.76336C61.7813 9.99624 62.2116 10.1852 62.6704 10.3227C63.1388 10.462 63.5862 10.5823 64.0012 10.6796C64.4067 10.777 64.7932 10.8782 65.1531 10.9774C65.5224 11.0786 65.8289 11.2122 66.0631 11.3745C66.2801 11.5425 66.3887 11.7448 66.3887 11.9777C66.3887 12.2278 66.2916 12.4263 66.0993 12.5695C65.9279 12.6954 65.7261 12.7985 65.4958 12.8711C65.273 12.9436 65.0407 12.9913 64.8066 13.0142C64.58 13.0352 64.3915 13.0467 64.2468 13.0467C63.8032 13.0467 63.4224 13.0028 63.1197 12.915C62.8151 12.8272 62.5638 12.7241 62.3734 12.6057C62.1811 12.4874 62.0288 12.3614 61.9222 12.2316C61.8213 12.1094 61.747 12.0082 61.7014 11.9319L61.4957 11.6303L59.708 12.7928C59.8508 13.0161 60.0298 13.2643 60.2392 13.5296C60.4772 13.8293 60.7837 14.1138 61.153 14.3734C61.5224 14.633 61.966 14.8525 62.4724 15.0243C62.9788 15.1961 63.5748 15.282 64.2468 15.282C64.797 15.282 65.2901 15.2324 65.7166 15.137C66.1412 15.0396 66.5162 14.9117 66.8285 14.7552C67.1388 14.6005 67.4015 14.4192 67.6109 14.2149C67.8223 14.0107 67.9974 13.8026 68.1326 13.6003C68.4563 13.1135 68.6409 12.5504 68.6828 11.9262C68.6828 11.4069 68.5876 10.9641 68.401 10.6166V10.6185Z", fill: "currentColor" }), p("path", { key: "3de8d99ca731b16e8828eca2959e9673830f8e33", d: "M26.3585 4.27905C23.7901 4.27905 21.6997 6.37504 21.6997 8.95017V15.0701H23.9291V8.95017C23.9291 7.6082 25.0181 6.5163 26.3565 6.5163C27.695 6.5163 28.784 7.6082 28.784 8.95017V15.0701H31.0134V8.95017C31.0134 6.37504 28.9229 4.27905 26.3546 4.27905H26.3585Z", fill: "currentColor" }), p("path", { key: "d03b5475a651095cc139a63bc5edeb510ffa803d", d: "M53.4522 4.27905C50.4251 4.27905 47.9634 6.74728 47.9634 9.78245C47.9634 12.8176 50.427 15.2859 53.4522 15.2859C54.5545 15.2859 55.6188 14.9594 56.5288 14.3409L56.7116 14.2169V15.0682H58.941V9.78245C58.941 6.74728 56.4793 4.27905 53.4522 4.27905ZM53.4522 13.0505C51.655 13.0505 50.1928 11.5845 50.1928 9.78245C50.1928 7.98044 51.655 6.51439 53.4522 6.51439C55.2494 6.51439 56.7116 7.98044 56.7116 9.78245C56.7116 11.5845 55.2494 13.0505 53.4522 13.0505Z", fill: "currentColor" }), p("path", { key: "97c2129d3c1de6065d31419fdbf9003a898c2ce8", d: "M45.5795 10.6166V0.356201H43.3501V10.6166C43.3501 13.1517 45.3758 15.2228 47.8908 15.2858V13.0486C46.6057 12.9875 45.5795 11.9204 45.5795 10.6166Z", fill: "currentColor" }), p("path", { key: "4d3686bc5f9c1ce78816523cd32a13c9d7220468", d: "M39.5428 4.4967V10.63C39.5428 11.6761 38.8669 12.621 37.9016 12.9284C37.1249 13.1746 36.3157 13.0486 35.6817 12.5829C35.0592 12.1247 34.6879 11.3917 34.6879 10.6186V4.4967H32.4585V10.5995C32.4585 12.7165 33.8388 14.57 35.8131 15.1084C37.0354 15.4424 38.2995 15.2973 39.3695 14.7037L39.5428 14.6063V14.9747C39.5428 16.2766 38.5166 17.3437 37.2315 17.4067V19.6439C39.7465 19.5809 41.7722 17.5117 41.7722 14.9747V4.4967H39.5428Z", fill: "currentColor" })), p("defs", { key: "9900a72ad101d4d2cb635fad18af7e491487f290" }, p("clipPath", { key: "9afa44c94e88b8929ac91ebe6d6258d9dc92342f", id: "clip0_455_3543" }, p("rect", { key: "bd688e672d4113e964c6fe2e79d50757b5415353", width: "68.7123", height: "19.2877", fill: "white", transform: "translate(0.144043 0.356201)" }))));
  }
  static get style() {
    return ft;
  }
}, [2, "nylas-logo", {
  width: [1],
  height: [1]
}]);
function dt() {
  if (typeof customElements > "u")
    return;
  ["nylas-logo"].forEach((t) => {
    switch (t) {
      case "nylas-logo":
        customElements.get(t) || customElements.define(t, ve);
        break;
    }
  });
}
const ht = ve, pt = dt;
export {
  ht as NylasLogo,
  pt as defineCustomElement
};
