// @nylas/web-elements/dist/cdn/tooltip-component/tooltip-component.es.js
const k = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let E, H, R, lt = !1, w = !1, W = !1, m = !1, X = null, M = !1;
const L = (t, e = "") => () => {
}, vt = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", Y = {}, St = "http://www.w3.org/2000/svg", kt = "http://www.w3.org/1999/xhtml", Tt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function xt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const x = (t, e, ...n) => {
  let s = null, o = null, r = null, l = !1, a = !1;
  const i = [], u = (c) => {
    for (let d = 0; d < c.length; d++)
      s = c[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof t != "function" && !z(s)) && (s = String(s)), l && a ? i[i.length - 1].$text$ += s : i.push(l ? P(null, s) : s), a = l);
  };
  if (u(n), e) {
    e.key && (o = e.key), e.name && (r = e.name);
    {
      const c = e.className || e.class;
      c && (e.class = typeof c != "object" ? c : Object.keys(c).filter((d) => c[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, _t);
  const $ = P(t, null);
  return $.$attrs$ = e, i.length > 0 && ($.$children$ = i), $.$key$ = o, $.$name$ = r, $;
}, P = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, Lt = {}, Et = (t) => t && t.$tag$ === Lt, _t = {
  forEach: (t, e) => t.map(G).forEach(e),
  map: (t, e) => t.map(G).map(e).map(At)
}, G = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), At = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), x(t.vtag, n, ...t.vchildren || []);
  }
  const e = P(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, Rt = (t) => se.map((e) => e(t)).find((e) => !!e), Ct = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, I = /* @__PURE__ */ new WeakMap(), Ot = (t, e, n) => {
  let s = B.get(t);
  ie && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, B.set(t, s);
}, Ft = (t, e, n) => {
  var s;
  const o = it(e, n), r = B.get(o);
  if (t = t.nodeType === 11 ? t : b, r)
    if (typeof r == "string") {
      t = t.head || t;
      let l = I.get(t), a;
      if (l || I.set(t, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          a = b.createElement("style"), a.innerHTML = r;
          const i = (s = h.$nonce$) !== null && s !== void 0 ? s : xt(b);
          i != null && a.setAttribute("nonce", i), t.insertBefore(a, t.querySelector("link"));
        }
        e.$flags$ & 4 && (a.innerHTML += vt), l && l.add(o);
      }
    } else t.adoptedStyleSheets.includes(r) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, r]);
  return o;
}, Ut = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, o = L("attachStyles", e.$tagName$), r = Ft(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), o();
}, it = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), K = (t, e, n, s, o, r) => {
  if (n !== s) {
    let l = et(t, e), a = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = J(n), $ = J(s);
      i.remove(...u.filter((c) => c && !$.includes(c))), i.add(...$.filter((c) => c && !u.includes(c)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(D, a) ? e = a.slice(2) : e = a[2] + e.slice(3), n || s) {
          const i = e.endsWith(rt);
          e = e.replace(Pt, ""), n && h.rel(t, e, n, i), s && h.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((l || i && s !== null) && !o)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const $ = s ?? "";
              e === "list" ? l = !1 : (n == null || t[e] != $) && (t[e] = $);
            }
          } catch {
          }
        let u = !1;
        a !== (a = a.replace(/^xlink\:?/, "")) && (e = a, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!l || r & 4 || o) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, wt = /\s/, J = (t) => t ? t.split(wt) : [], rt = "Capture", Pt = new RegExp(rt + "$"), at = (t, e, n, s) => {
  const o = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, r = t && t.$attrs$ || Y, l = e.$attrs$ || Y;
  for (s in r)
    s in l || K(o, s, r[s], void 0, n, e.$flags$);
  for (s in l)
    K(o, s, r[s], l[s], n, e.$flags$);
}, j = (t, e, n, s) => {
  const o = e.$children$[n];
  let r = 0, l, a, i;
  if (lt || (W = !0, o.$tag$ === "slot" && (E && s.classList.add(E + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = b.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = b.createTextNode("");
  else {
    if (m || (m = o.$tag$ === "svg"), l = o.$elm$ = b.createElementNS(m ? St : kt, o.$flags$ & 2 ? "slot-fb" : o.$tag$), m && o.$tag$ === "foreignObject" && (m = !1), at(null, o, m), Tt(E) && l["s-si"] !== E && l.classList.add(l["s-si"] = E), o.$children$)
      for (r = 0; r < o.$children$.length; ++r)
        a = j(t, o, r, l), a && l.appendChild(a);
    o.$tag$ === "svg" ? m = !1 : l.tagName === "foreignObject" && (m = !0);
  }
  return l["s-hn"] = R, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = H, l["s-sn"] = o.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === o.$tag$ && t.$elm$ && C(t.$elm$, !1)), l;
}, C = (t, e) => {
  h.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== R && o["s-ol"] && ($t(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), e && C(o, e);
  }
  h.$flags$ &= -2;
}, ct = (t, e, n, s, o, r) => {
  let l = t["s-cr"] && t["s-cr"].parentNode || t, a;
  for (l.shadowRoot && l.tagName === R && (l = l.shadowRoot); o <= r; ++o)
    s[o] && (a = j(null, n, o, t), a && (s[o].$elm$ = a, l.insertBefore(a, q(e))));
}, ft = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const o = t[s];
    if (o) {
      const r = o.$elm$;
      pt(o), r && (w = !0, r["s-ol"] ? r["s-ol"].remove() : C(r, !0), r.remove());
    }
  }
}, jt = (t, e, n, s, o = !1) => {
  let r = 0, l = 0, a = 0, i = 0, u = e.length - 1, $ = e[0], c = e[u], d = s.length - 1, f = s[0], p = s[d], y, g;
  for (; r <= u && l <= d; )
    if ($ == null)
      $ = e[++r];
    else if (c == null)
      c = e[--u];
    else if (f == null)
      f = s[++l];
    else if (p == null)
      p = s[--d];
    else if (U($, f, o))
      _($, f, o), $ = e[++r], f = s[++l];
    else if (U(c, p, o))
      _(c, p, o), c = e[--u], p = s[--d];
    else if (U($, p, o))
      ($.$tag$ === "slot" || p.$tag$ === "slot") && C($.$elm$.parentNode, !1), _($, p, o), t.insertBefore($.$elm$, c.$elm$.nextSibling), $ = e[++r], p = s[--d];
    else if (U(c, f, o))
      ($.$tag$ === "slot" || p.$tag$ === "slot") && C(c.$elm$.parentNode, !1), _(c, f, o), t.insertBefore(c.$elm$, $.$elm$), c = e[--u], f = s[++l];
    else {
      for (a = -1, i = r; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === f.$key$) {
          a = i;
          break;
        }
      a >= 0 ? (g = e[a], g.$tag$ !== f.$tag$ ? y = j(e && e[l], n, a, t) : (_(g, f, o), e[a] = void 0, y = g.$elm$), f = s[++l]) : (y = j(e && e[l], n, l, t), f = s[++l]), y && $t($.$elm$).insertBefore(y, q($.$elm$));
    }
  r > u ? ct(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, l, d) : l > d && ft(e, r, u);
}, U = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, $t = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, _ = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, o = t.$children$, r = e.$children$, l = e.$tag$, a = e.$text$;
  let i;
  a === null ? (m = l === "svg" ? !0 : l === "foreignObject" ? !1 : m, l === "slot" || at(t, e, m), o !== null && r !== null ? jt(s, o, e, r, n) : r !== null ? (t.$text$ !== null && (s.textContent = ""), ct(s, null, e, r, 0, r.length - 1)) : o !== null && ft(o, 0, o.length - 1), m && l === "svg" && (m = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = a : t.$text$ !== a && (s.data = a);
}, dt = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of e)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      dt(n);
    }
}, v = [], ut = (t) => {
  let e, n, s;
  for (const o of t.childNodes) {
    if (o["s-sr"] && (e = o["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const r = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== o["s-hn"] && !k.experimentalSlotFixes)
          if (Z(e, r)) {
            let l = v.find((a) => a.$nodeToRelocate$ === e);
            w = !0, e["s-sn"] = e["s-sn"] || r, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (e["s-sh"] = o["s-hn"], v.push({
              $slotRefNode$: o,
              $nodeToRelocate$: e
            })), e["s-sr"] && v.map((a) => {
              Z(a.$nodeToRelocate$, e["s-sn"]) && (l = v.find((i) => i.$nodeToRelocate$ === e), l && !a.$slotRefNode$ && (a.$slotRefNode$ = l.$slotRefNode$));
            });
          } else v.some((l) => l.$nodeToRelocate$ === e) || v.push({
            $nodeToRelocate$: e
          });
    }
    o.nodeType === 1 && ut(o);
  }
}, Z = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", pt = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(pt);
}, Bt = (t, e, n = !1) => {
  var s, o, r, l, a;
  const i = t.$hostElement$, u = t.$cmpMeta$, $ = t.$vnode$ || P(null, null), c = Et(e) ? e : x(null, null, e);
  if (R = i.tagName, u.$attrsToReflect$ && (c.$attrs$ = c.$attrs$ || {}, u.$attrsToReflect$.map(([d, f]) => c.$attrs$[f] = i[d])), n && c.$attrs$)
    for (const d of Object.keys(c.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (c.$attrs$[d] = i[d]);
  c.$tag$ = null, c.$flags$ |= 4, t.$vnode$ = c, c.$elm$ = $.$elm$ = i.shadowRoot || i, E = i["s-sc"], H = i["s-cr"], lt = (u.$flags$ & 1) !== 0, w = !1, _($, c, n);
  {
    if (h.$flags$ |= 1, W) {
      ut(c.$elm$);
      for (const d of v) {
        const f = d.$nodeToRelocate$;
        if (!f["s-ol"]) {
          const p = b.createTextNode("");
          p["s-nr"] = f, f.parentNode.insertBefore(f["s-ol"] = p, f);
        }
      }
      for (const d of v) {
        const f = d.$nodeToRelocate$, p = d.$slotRefNode$;
        if (p) {
          const y = p.parentNode;
          let g = p.nextSibling;
          {
            let T = (s = f["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let S = (o = T["s-nr"]) !== null && o !== void 0 ? o : null;
              if (S && S["s-sn"] === f["s-sn"] && y === S.parentNode && (S = S.nextSibling, !S || !S["s-nr"])) {
                g = S;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== f.parentNode || f.nextSibling !== g) && f !== g && (!f["s-hn"] && f["s-ol"] && (f["s-hn"] = f["s-ol"].parentNode.nodeName), y.insertBefore(f, g), f.nodeType === 1 && (f.hidden = (r = f["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          f.nodeType === 1 && (n && (f["s-ih"] = (l = f.hidden) !== null && l !== void 0 ? l : !1), f.hidden = !0);
      }
    }
    w && dt(c.$elm$), h.$flags$ &= -2, v.length = 0;
  }
  if (k.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of c.$elm$.childNodes)
      d["s-hn"] !== R && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (a = d.hidden) !== null && a !== void 0 ? a : !1), d.hidden = !0);
  H = void 0;
}, Dt = (t, e) => {
}, ht = (t, e) => (t.$flags$ |= 16, Dt(t, t.$ancestorComponent$), ce(() => Ht(t, e))), Ht = (t, e) => {
  const n = t.$hostElement$, s = L("scheduleUpdate", t.$cmpMeta$.$tagName$), o = n;
  let r;
  return e ? r = A(o, "componentWillLoad") : r = A(o, "componentWillUpdate"), r = V(r, () => A(o, "componentWillRender")), s(), V(r, () => Nt(t, o, e));
}, V = (t, e) => Mt(t) ? t.then(e) : e(), Mt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Nt = async (t, e, n) => {
  const s = t.$hostElement$, o = L("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ut(t);
  const r = L("render", t.$cmpMeta$.$tagName$);
  Wt(t, e, s, n), r(), o(), zt(t);
}, Wt = (t, e, n, s) => {
  try {
    X = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (k.hasRenderFn || k.reflect) && (k.vdomRender || k.reflect) && (k.hydrateServerSide || Bt(t, e, s));
  } catch (i) {
    F(i, t.$hostElement$);
  }
  return X = null, null;
}, zt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = L("postUpdate", e), o = n;
  t.$ancestorComponent$, A(o, "componentDidRender"), t.$flags$ & 64 ? (A(o, "componentDidUpdate"), s()) : (t.$flags$ |= 64, A(o, "componentDidLoad"), s());
}, A = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      F(s);
    }
}, qt = (t, e) => O(t).$instanceValues$.get(e), Xt = (t, e, n, s) => {
  const o = O(t), r = t, l = o.$instanceValues$.get(e), a = o.$flags$, i = r;
  n = Ct(n, s.$members$[e][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && a & 128) {
        const c = s.$watchers$[e];
        c && c.map((d) => {
          try {
            i[d](n, l, e);
          } catch (f) {
            F(f, r);
          }
        });
      }
      if ((a & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, l, e) === !1)
          return;
        ht(o, !1);
      }
    }
  }
}, Qt = (t, e, n) => {
  var s;
  const o = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const r = Object.entries(e.$members$);
    r.map(([l, [a]]) => {
      (a & 31 || a & 32) && Object.defineProperty(o, l, {
        get() {
          return qt(this, l);
        },
        set(i) {
          Xt(this, l, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(a, i, u) {
        h.jmp(() => {
          var $;
          const c = l.get(a);
          if (this.hasOwnProperty(c))
            u = this[c], delete this[c];
          else {
            if (o.hasOwnProperty(c) && typeof this[c] == "number" && this[c] == u)
              return;
            if (c == null) {
              const d = O(this), f = d == null ? void 0 : d.$flags$;
              if (f && !(f & 8) && f & 128 && u !== i) {
                const y = this, g = ($ = e.$watchers$) === null || $ === void 0 ? void 0 : $[a];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, a);
                });
              }
              return;
            }
          }
          this[c] = u === null && typeof this[c] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([a, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([a, i]) => {
          var u;
          const $ = i[1] || a;
          return l.set($, a), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([a, $])), $;
        })
      ]));
    }
  }
  return t;
}, Yt = async (t, e, n, s) => {
  let o;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, o = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[e.$modeName$ = Rt(t)]);
    const a = it(n, e.$modeName$);
    if (!B.has(a)) {
      const i = L("registerStyles", n.$tagName$);
      Ot(a, l, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, ht(e, !0);
}, tt = (t) => {
}, Gt = (t) => {
  if (!(h.$flags$ & 1)) {
    const e = O(t), n = e.$cmpMeta$, s = L("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (gt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && It(t), n.$members$ && Object.entries(n.$members$).map(([o, [r]]) => {
      if (r & 31 && t.hasOwnProperty(o)) {
        const l = t[o];
        delete t[o], t[o] = l;
      }
    }), Yt(t, e, n)), s();
  }
}, It = (t) => {
  const e = t["s-cr"] = b.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Kt = async (t) => {
  if (!(h.$flags$ & 1)) {
    const e = O(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Jt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [], !st && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = t.prototype.connectedCallback, o = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      ee(this, n);
    },
    connectedCallback() {
      Gt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Kt(this), o && o.call(this);
    },
    __attachShadow() {
      st ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), t.is = n.$tagName$, Qt(t, n);
}, gt = (t, e, n, s) => {
  n && n.map(([o, r, l]) => {
    const a = Vt(t, o), i = Zt(e, l), u = te(o);
    h.ael(a, r, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => h.rel(a, r, i, u));
  });
}, Zt = (t, e) => (n) => {
  try {
    k.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    F(s);
  }
}, Vt = (t, e) => e & 4 ? b : e & 8 ? D : e & 16 ? b.body : t, te = (t) => oe ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, yt = /* @__PURE__ */ new WeakMap(), O = (t) => yt.get(t), ee = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return gt(t, n, e.$listeners$), yt.set(t, n);
}, et = (t, e) => e in t, F = (t, e) => (0, console.error)(t, e), B = /* @__PURE__ */ new Map(), se = [], D = typeof window < "u" ? window : {}, b = D.document || { head: {} }, ne = D.HTMLElement || class {
}, h = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, st = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), oe = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    b.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), le = (t) => Promise.resolve(t), ie = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), nt = [], mt = [], re = (t, e) => (n) => {
  t.push(n), M || (M = !0, e && h.$flags$ & 4 ? ae(N) : h.raf(N));
}, ot = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      F(n);
    }
  t.length = 0;
}, N = () => {
  ot(nt), ot(mt), (M = nt.length > 0) && h.raf(N);
}, ae = (t) => le().then(t), ce = /* @__PURE__ */ re(mt, !0), fe = '.tooltip-container{position:relative;display:flex;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:"Inter", sans-serif}.tooltip-content{visibility:hidden;background-color:var(--nylas-base-800);color:var(--nylas-base-0);text-align:left;border-radius:6px;padding:5px 0;position:absolute;z-index:1;opacity:0;transition:opacity 0.3s;padding:0.5rem 1rem;max-width:200px;width:max-content}.tooltip-content.top{bottom:0;left:0;margin-bottom:1.25rem;transform:translateX(-50%)}.tooltip-content.right{top:50%;left:0%;margin-left:1.25rem;transform:translateY(-50%)}.tooltip-content.bottom{top:0;left:0;margin-top:1.25rem;transform:translateX(-50%)}.tooltip-content.left{top:50%;right:0%;margin-right:1.25rem;transform:translateY(-50%)}.tooltip-container:hover .tooltip-content,.tooltip-container:focus-within .tooltip-content{visibility:visible;opacity:1}.tooltip-content::after{content:"";position:absolute;top:-0.75rem;left:50%;transform:rotate(180deg);border-width:6px;border-style:solid;border-color:black transparent transparent transparent}.tooltip-content.top::after{top:100%;left:50%;transform:rotate(0deg)}.tooltip-content.bottom::after{top:-0.75rem;left:50%;transform:rotate(180deg)}.tooltip-content.left::after{top:47.5%;left:100%;transform:rotate(270deg)}.tooltip-content.right::after{top:47.5%;left:-0.75rem;transform:rotate(90deg)}', $e = fe, bt = Jt(class extends ne {
  constructor() {
    super(), this.__registerHost(), this.__attachShadow(), this.position = "bottom";
  }
  showTooltip(e) {
    e.target.nextElementSibling.style.visibility = "visible", e.target.nextElementSibling.style.opacity = "1";
  }
  hideTooltip(e) {
    e.target.nextElementSibling.style.visibility = "hidden", e.target.nextElementSibling.style.opacity = "0";
  }
  render() {
    return x("div", { key: "2ca61ec97333e6f3c5504b5123747d82037f8650", class: "tooltip-container", part: "tc__container" }, x("span", { key: "4133e0e9a968139a8101789c5291f824a87b5c9d", class: "info-icon", tabindex: "0", onFocus: this.showTooltip.bind(this), onBlur: this.hideTooltip.bind(this) }, x("slot", { key: "6b07811a2de2bb05b8133c4d2992eb36ae280f8e", name: "tooltip-icon" }, "i")), x("span", { key: "99de5567aaa0998d1d29042f4b86fac5c0689dc6", class: `tooltip-content ${this.position}`, part: "tc__content" }, x("slot", { key: "608f52f2bd9bfef59c8dd32fd6f5f15bf044bb12", name: "tooltip-content" }, "This is a help text.")));
  }
  get el() {
    return this;
  }
  static get style() {
    return $e;
  }
}, [1, "tooltip-component", {
  position: [1]
}]);
function de() {
  if (typeof customElements > "u")
    return;
  ["tooltip-component"].forEach((e) => {
    switch (e) {
      case "tooltip-component":
        customElements.get(e) || customElements.define(e, bt);
        break;
    }
  });
}
const pe = bt, he = de;
export {
  pe as TooltipComponent,
  he as defineCustomElement
};
