// @nylas/web-elements/dist/cdn/location-off/location-off.es.js
const k = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let x, D, _, lt = !1, U = !1, W = !1, v = !1, I = null, H = !1;
const b = (t, e = "") => () => {
}, St = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, Lt = "http://www.w3.org/2000/svg", mt = "http://www.w3.org/1999/xhtml", kt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function Tt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const w = (t, e, ...n) => {
  let s = null, o = null, r = null, l = !1, c = !1;
  const i = [], u = ($) => {
    for (let d = 0; d < $.length; d++)
      s = $[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof t != "function" && !z(s)) && (s = String(s)), l && c ? i[i.length - 1].$text$ += s : i.push(l ? P(null, s) : s), c = l);
  };
  if (u(n), e) {
    e.key && (o = e.key), e.name && (r = e.name);
    {
      const $ = e.className || e.class;
      $ && (e.class = typeof $ != "object" ? $ : Object.keys($).filter((d) => $[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, Ct);
  const f = P(t, null);
  return f.$attrs$ = e, i.length > 0 && (f.$children$ = i), f.$key$ = o, f.$name$ = r, f;
}, P = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, bt = {}, xt = (t) => t && t.$tag$ === bt, Ct = {
  forEach: (t, e) => t.map(Z).forEach(e),
  map: (t, e) => t.map(Z).map(e).map(Et)
}, Z = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), Et = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), w(t.vtag, n, ...t.vchildren || []);
  }
  const e = P(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, _t = (t) => ee.map((e) => e(t)).find((e) => !!e), At = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, G = /* @__PURE__ */ new WeakMap(), Rt = (t, e, n) => {
  let s = M.get(t);
  le && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, M.set(t, s);
}, Ot = (t, e, n) => {
  var s;
  const o = it(e, n), r = M.get(o);
  if (t = t.nodeType === 11 ? t : S, r)
    if (typeof r == "string") {
      t = t.head || t;
      let l = G.get(t), c;
      if (l || G.set(t, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          c = S.createElement("style"), c.innerHTML = r;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Tt(S);
          i != null && c.setAttribute("nonce", i), t.insertBefore(c, t.querySelector("link"));
        }
        e.$flags$ & 4 && (c.innerHTML += St), l && l.add(o);
      }
    } else t.adoptedStyleSheets.includes(r) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, r]);
  return o;
}, Ft = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, o = b("attachStyles", e.$tagName$), r = Ot(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = r, n.classList.add(r + "-h"), s & 2 && n.classList.add(r + "-s")), o();
}, it = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), K = (t, e, n, s, o, r) => {
  if (n !== s) {
    let l = et(t, e), c = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = J(n), f = J(s);
      i.remove(...u.filter(($) => $ && !f.includes($))), i.add(...f.filter(($) => $ && !u.includes($)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(B, c) ? e = c.slice(2) : e = c[2] + e.slice(3), n || s) {
          const i = e.endsWith(rt);
          e = e.replace(wt, ""), n && p.rel(t, e, n, i), s && p.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((l || i && s !== null) && !o)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const f = s ?? "";
              e === "list" ? l = !1 : (n == null || t[e] != f) && (t[e] = f);
            }
          } catch {
          }
        let u = !1;
        c !== (c = c.replace(/^xlink\:?/, "")) && (e = c, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!l || r & 4 || o) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, Ut = /\s/, J = (t) => t ? t.split(Ut) : [], rt = "Capture", wt = new RegExp(rt + "$"), ct = (t, e, n, s) => {
  const o = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, r = t && t.$attrs$ || X, l = e.$attrs$ || X;
  for (s in r)
    s in l || K(o, s, r[s], void 0, n, e.$flags$);
  for (s in l)
    K(o, s, r[s], l[s], n, e.$flags$);
}, j = (t, e, n, s) => {
  const o = e.$children$[n];
  let r = 0, l, c, i;
  if (lt || (W = !0, o.$tag$ === "slot" && (x && s.classList.add(x + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = S.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = S.createTextNode("");
  else {
    if (v || (v = o.$tag$ === "svg"), l = o.$elm$ = S.createElementNS(v ? Lt : mt, o.$flags$ & 2 ? "slot-fb" : o.$tag$), v && o.$tag$ === "foreignObject" && (v = !1), ct(null, o, v), kt(x) && l["s-si"] !== x && l.classList.add(l["s-si"] = x), o.$children$)
      for (r = 0; r < o.$children$.length; ++r)
        c = j(t, o, r, l), c && l.appendChild(c);
    o.$tag$ === "svg" ? v = !1 : l.tagName === "foreignObject" && (v = !0);
  }
  return l["s-hn"] = _, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = D, l["s-sn"] = o.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === o.$tag$ && t.$elm$ && A(t.$elm$, !1)), l;
}, A = (t, e) => {
  p.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== _ && o["s-ol"] && (ft(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), e && A(o, e);
  }
  p.$flags$ &= -2;
}, $t = (t, e, n, s, o, r) => {
  let l = t["s-cr"] && t["s-cr"].parentNode || t, c;
  for (l.shadowRoot && l.tagName === _ && (l = l.shadowRoot); o <= r; ++o)
    s[o] && (c = j(null, n, o, t), c && (s[o].$elm$ = c, l.insertBefore(c, q(e))));
}, at = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const o = t[s];
    if (o) {
      const r = o.$elm$;
      ht(o), r && (U = !0, r["s-ol"] ? r["s-ol"].remove() : A(r, !0), r.remove());
    }
  }
}, Pt = (t, e, n, s, o = !1) => {
  let r = 0, l = 0, c = 0, i = 0, u = e.length - 1, f = e[0], $ = e[u], d = s.length - 1, a = s[0], h = s[d], y, g;
  for (; r <= u && l <= d; )
    if (f == null)
      f = e[++r];
    else if ($ == null)
      $ = e[--u];
    else if (a == null)
      a = s[++l];
    else if (h == null)
      h = s[--d];
    else if (F(f, a, o))
      C(f, a, o), f = e[++r], a = s[++l];
    else if (F($, h, o))
      C($, h, o), $ = e[--u], h = s[--d];
    else if (F(f, h, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A(f.$elm$.parentNode, !1), C(f, h, o), t.insertBefore(f.$elm$, $.$elm$.nextSibling), f = e[++r], h = s[--d];
    else if (F($, a, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && A($.$elm$.parentNode, !1), C($, a, o), t.insertBefore($.$elm$, f.$elm$), $ = e[--u], a = s[++l];
    else {
      for (c = -1, i = r; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === a.$key$) {
          c = i;
          break;
        }
      c >= 0 ? (g = e[c], g.$tag$ !== a.$tag$ ? y = j(e && e[l], n, c, t) : (C(g, a, o), e[c] = void 0, y = g.$elm$), a = s[++l]) : (y = j(e && e[l], n, l, t), a = s[++l]), y && ft(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  r > u ? $t(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, l, d) : l > d && at(e, r, u);
}, F = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, ft = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, C = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, o = t.$children$, r = e.$children$, l = e.$tag$, c = e.$text$;
  let i;
  c === null ? (v = l === "svg" ? !0 : l === "foreignObject" ? !1 : v, l === "slot" || ct(t, e, v), o !== null && r !== null ? Pt(s, o, e, r, n) : r !== null ? (t.$text$ !== null && (s.textContent = ""), $t(s, null, e, r, 0, r.length - 1)) : o !== null && at(o, 0, o.length - 1), v && l === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = c : t.$text$ !== c && (s.data = c);
}, dt = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of e)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      dt(n);
    }
}, L = [], ut = (t) => {
  let e, n, s;
  for (const o of t.childNodes) {
    if (o["s-sr"] && (e = o["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const r = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== o["s-hn"] && !k.experimentalSlotFixes)
          if (Y(e, r)) {
            let l = L.find((c) => c.$nodeToRelocate$ === e);
            U = !0, e["s-sn"] = e["s-sn"] || r, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (e["s-sh"] = o["s-hn"], L.push({
              $slotRefNode$: o,
              $nodeToRelocate$: e
            })), e["s-sr"] && L.map((c) => {
              Y(c.$nodeToRelocate$, e["s-sn"]) && (l = L.find((i) => i.$nodeToRelocate$ === e), l && !c.$slotRefNode$ && (c.$slotRefNode$ = l.$slotRefNode$));
            });
          } else L.some((l) => l.$nodeToRelocate$ === e) || L.push({
            $nodeToRelocate$: e
          });
    }
    o.nodeType === 1 && ut(o);
  }
}, Y = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ht = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ht);
}, jt = (t, e, n = !1) => {
  var s, o, r, l, c;
  const i = t.$hostElement$, u = t.$cmpMeta$, f = t.$vnode$ || P(null, null), $ = xt(e) ? e : w(null, null, e);
  if (_ = i.tagName, u.$attrsToReflect$ && ($.$attrs$ = $.$attrs$ || {}, u.$attrsToReflect$.map(([d, a]) => $.$attrs$[a] = i[d])), n && $.$attrs$)
    for (const d of Object.keys($.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && ($.$attrs$[d] = i[d]);
  $.$tag$ = null, $.$flags$ |= 4, t.$vnode$ = $, $.$elm$ = f.$elm$ = i.shadowRoot || i, x = i["s-sc"], D = i["s-cr"], lt = (u.$flags$ & 1) !== 0, U = !1, C(f, $, n);
  {
    if (p.$flags$ |= 1, W) {
      ut($.$elm$);
      for (const d of L) {
        const a = d.$nodeToRelocate$;
        if (!a["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = a, a.parentNode.insertBefore(a["s-ol"] = h, a);
        }
      }
      for (const d of L) {
        const a = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = a["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let m = (o = T["s-nr"]) !== null && o !== void 0 ? o : null;
              if (m && m["s-sn"] === a["s-sn"] && y === m.parentNode && (m = m.nextSibling, !m || !m["s-nr"])) {
                g = m;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== a.parentNode || a.nextSibling !== g) && a !== g && (!a["s-hn"] && a["s-ol"] && (a["s-hn"] = a["s-ol"].parentNode.nodeName), y.insertBefore(a, g), a.nodeType === 1 && (a.hidden = (r = a["s-ih"]) !== null && r !== void 0 ? r : !1));
        } else
          a.nodeType === 1 && (n && (a["s-ih"] = (l = a.hidden) !== null && l !== void 0 ? l : !1), a.hidden = !0);
      }
    }
    U && dt($.$elm$), p.$flags$ &= -2, L.length = 0;
  }
  if (k.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of $.$elm$.childNodes)
      d["s-hn"] !== _ && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (c = d.hidden) !== null && c !== void 0 ? c : !1), d.hidden = !0);
  D = void 0;
}, Mt = (t, e) => {
}, pt = (t, e) => (t.$flags$ |= 16, Mt(t, t.$ancestorComponent$), ce(() => Bt(t, e))), Bt = (t, e) => {
  const n = t.$hostElement$, s = b("scheduleUpdate", t.$cmpMeta$.$tagName$), o = n;
  let r;
  return e ? r = E(o, "componentWillLoad") : r = E(o, "componentWillUpdate"), r = V(r, () => E(o, "componentWillRender")), s(), V(r, () => Ht(t, o, e));
}, V = (t, e) => Dt(t) ? t.then(e) : e(), Dt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Ht = async (t, e, n) => {
  const s = t.$hostElement$, o = b("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ft(t);
  const r = b("render", t.$cmpMeta$.$tagName$);
  Nt(t, e, s, n), r(), o(), Wt(t);
}, Nt = (t, e, n, s) => {
  try {
    I = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (k.hasRenderFn || k.reflect) && (k.vdomRender || k.reflect) && (k.hydrateServerSide || jt(t, e, s));
  } catch (i) {
    O(i, t.$hostElement$);
  }
  return I = null, null;
}, Wt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = b("postUpdate", e), o = n;
  t.$ancestorComponent$, E(o, "componentDidRender"), t.$flags$ & 64 ? (E(o, "componentDidUpdate"), s()) : (t.$flags$ |= 64, E(o, "componentDidLoad"), s());
}, E = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      O(s);
    }
}, zt = (t, e) => R(t).$instanceValues$.get(e), qt = (t, e, n, s) => {
  const o = R(t), r = t, l = o.$instanceValues$.get(e), c = o.$flags$, i = r;
  n = At(n, s.$members$[e][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && c & 128) {
        const $ = s.$watchers$[e];
        $ && $.map((d) => {
          try {
            i[d](n, l, e);
          } catch (a) {
            O(a, r);
          }
        });
      }
      if ((c & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, l, e) === !1)
          return;
        pt(o, !1);
      }
    }
  }
}, It = (t, e, n) => {
  var s;
  const o = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const r = Object.entries(e.$members$);
    r.map(([l, [c]]) => {
      (c & 31 || c & 32) && Object.defineProperty(o, l, {
        get() {
          return zt(this, l);
        },
        set(i) {
          qt(this, l, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(c, i, u) {
        p.jmp(() => {
          var f;
          const $ = l.get(c);
          if (this.hasOwnProperty($))
            u = this[$], delete this[$];
          else {
            if (o.hasOwnProperty($) && typeof this[$] == "number" && this[$] == u)
              return;
            if ($ == null) {
              const d = R(this), a = d == null ? void 0 : d.$flags$;
              if (a && !(a & 8) && a & 128 && u !== i) {
                const y = this, g = (f = e.$watchers$) === null || f === void 0 ? void 0 : f[c];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, c);
                });
              }
              return;
            }
          }
          this[$] = u === null && typeof this[$] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...r.filter(
          ([c, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([c, i]) => {
          var u;
          const f = i[1] || c;
          return l.set(f, c), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([c, f])), f;
        })
      ]));
    }
  }
  return t;
}, Qt = async (t, e, n, s) => {
  let o;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, o = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[e.$modeName$ = _t(t)]);
    const c = it(n, e.$modeName$);
    if (!M.has(c)) {
      const i = b("registerStyles", n.$tagName$);
      Rt(c, l, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, pt(e, !0);
}, tt = (t) => {
}, Xt = (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t), n = e.$cmpMeta$, s = b("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (gt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Zt(t), n.$members$ && Object.entries(n.$members$).map(([o, [r]]) => {
      if (r & 31 && t.hasOwnProperty(o)) {
        const l = t[o];
        delete t[o], t[o] = l;
      }
    }), Qt(t, e, n)), s();
  }
}, Zt = (t) => {
  const e = t["s-cr"] = S.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Gt = async (t) => {
  if (!(p.$flags$ & 1)) {
    const e = R(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Kt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [], !st && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = t.prototype.connectedCallback, o = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      te(this, n);
    },
    connectedCallback() {
      Xt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Gt(this), o && o.call(this);
    },
    __attachShadow() {
      st ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), t.is = n.$tagName$, It(t, n);
}, gt = (t, e, n, s) => {
  n && n.map(([o, r, l]) => {
    const c = Yt(t, o), i = Jt(e, l), u = Vt(o);
    p.ael(c, r, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => p.rel(c, r, i, u));
  });
}, Jt = (t, e) => (n) => {
  try {
    k.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    O(s);
  }
}, Yt = (t, e) => e & 4 ? S : e & 8 ? B : e & 16 ? S.body : t, Vt = (t) => ne ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, yt = /* @__PURE__ */ new WeakMap(), R = (t) => yt.get(t), te = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return gt(t, n, e.$listeners$), yt.set(t, n);
}, et = (t, e) => e in t, O = (t, e) => (0, console.error)(t, e), M = /* @__PURE__ */ new Map(), ee = [], B = typeof window < "u" ? window : {}, S = B.document || { head: {} }, se = B.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, st = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), ne = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), oe = (t) => Promise.resolve(t), le = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), nt = [], vt = [], ie = (t, e) => (n) => {
  t.push(n), H || (H = !0, e && p.$flags$ & 4 ? re(N) : p.raf(N));
}, ot = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      O(n);
    }
  t.length = 0;
}, N = () => {
  ot(nt), ot(vt), (H = nt.length > 0) && p.raf(N);
}, re = (t) => oe().then(t), ce = /* @__PURE__ */ ie(vt, !0), $e = ".sc-location-off-icon-h{display:flex}", ae = $e, fe = Kt(class extends se {
  constructor() {
    super(), this.__registerHost(), this.width = "16", this.height = "18";
  }
  render() {
    return w("svg", { key: "954a088889bb64d20861c53c37e449d5575c3bee", width: this.width, height: this.height, fill: "none", xmlns: "http://www.w3.org/2000/svg" }, w("path", { key: "1a4e4176eb9c90697f936001e12384823775dbfe", d: "M0.146447 0.808556C0.341709 0.613294 0.658291 0.613294 0.853553 0.808556L15.8536 15.8086C16.0488 16.0038 16.0488 16.3204 15.8536 16.5157C15.6583 16.7109 15.3417 16.7109 15.1464 16.5157L12.0938 13.463L11.8223 13.7312L9.38611 16.1012C8.65623 16.8076 7.52168 16.8468 6.74691 16.2188L6.61408 16.101L4.57128 14.1159L3.05025 12.6171C0.607421 10.1732 0.347525 6.37239 2.27056 3.63976L0.146447 1.51566C-0.0488155 1.3204 -0.0488155 1.00382 0.146447 0.808556ZM9.07768 10.4469C8.74346 10.576 8.38024 10.6468 8.0005 10.6468C6.34839 10.6468 5.00909 9.30689 5.00909 7.65406C5.00909 7.27461 5.07967 6.91166 5.20843 6.57763L2.98793 4.35713C1.49206 6.62038 1.68917 9.67377 3.57927 11.7288L3.75499 11.912L4.3263 12.4791L7.30713 15.3845L7.40012 15.4642C7.7543 15.732 8.24594 15.732 8.60016 15.4642L8.69315 15.3846L10.9237 13.216L11.3883 12.7575L9.07768 10.4469ZM9.85409 8.39486L10.5983 9.13904C10.8487 8.70143 10.9919 8.19447 10.9919 7.65406C10.9919 6.00122 9.65261 4.66134 8.0005 4.66134C7.45986 4.66134 6.95271 4.80482 6.51504 5.05581L7.25921 5.79998C7.48839 5.70819 7.73855 5.6577 8.0005 5.6577C9.10258 5.6577 9.99599 6.5515 9.99599 7.65406C9.99599 7.91581 9.94564 8.16581 9.85409 8.39486ZM4.35291 2.89368C6.70671 1.08827 10.0908 1.26314 12.245 3.4183C14.3986 5.57283 14.5739 8.95702 12.771 11.3118L13.4812 12.022C15.6702 9.27273 15.493 5.25762 12.9497 2.71326C10.4058 0.168159 6.39081 -0.00843871 3.64269 2.18347L4.35291 2.89368Z", fill: "currentColor" }));
  }
  static get style() {
    return ae;
  }
}, [2, "location-off-icon", {
  width: [1],
  height: [1]
}]);
function ue() {
  if (typeof customElements > "u")
    return;
  ["location-off-icon"].forEach((e) => {
    switch (e) {
      case "location-off-icon":
        customElements.get(e) || customElements.define(e, fe);
        break;
    }
  });
}
export {
  fe as L,
  ue as d
};
