// @nylas/web-elements/dist/collection/components/design-system/checkbox-component/checkbox-component.js
import { h } from "@stencil/core";
export class CheckboxComponent {
    constructor() {
        this.componentType = 'checkbox';
        this.name = 'checkbox';
        this.checked = undefined;
        this.label = '';
        this.required = false;
        this.isChecked = undefined;
        this.error = '';
    }
    checkedPropChangedHandler(newValue) {
        if (typeof newValue === 'boolean') {
            this.isChecked = newValue;
        }
    }
    componentWillLoad() {
        this.el.setAttribute('component-type', this.componentType);
    }
    componentDidLoad() {
        this.isChecked = this.checked || false;
    }
    handleBookingFormSubmitted(event) {
        if (this.required && !this.isChecked) {
            this.error = `${this.label} is required.`;
            return;
        }
        if (this.error) {
            event.preventDefault();
        }
    }
    toggleCheck() {
        this.error = '';
        this.isChecked = !this.isChecked;
        this.nylasFormCheckboxToggled.emit({
            checked: this.isChecked,
            name: this.name,
            label: this.label,
        });
    }
    render() {
        return (h("div", { key: '846414c76901571e7c2591f2f73a1e81a75da8cb', class: "checkbox-container" }, h("label", { key: '0d4f046e5e16b98e8bf456b1a839764ef8ca2149' }, h("input", { key: '53e0e070c154543aa4b66c90d7506938a926fbed', type: "checkbox", name: this.name, checked: this.isChecked, onChange: () => this.toggleCheck() }), h("span", { key: '651d933ddb459c8b3226babc82bf0fa05663a389', class: "checkmark" }), h("slot", { key: '2b59df9447d01fb0000d4dfcfa3d029d48a4400e', name: "label" }, h("span", { key: '644a2d26e208e41545566471ebba96c93629af0c', class: "label" }, this.label))), this.error && h("p", { class: "error" }, this.error)));
    }
    static get is() { return "checkbox-component"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["checkbox-component.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["checkbox-component.css"]
        };
    }
    static get properties() {
        return {
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The name of the checkbox. This is used to identify the checkbox when submitting a form."
                },
                "attribute": "name",
                "reflect": false,
                "defaultValue": "'checkbox'"
            },
            "checked": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean | undefined",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The default value of the checkbox. This is the value that is displayed when the checkbox is rendered."
                },
                "attribute": "checked",
                "reflect": false
            },
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The label of the checkbox. This is displayed next to the checkbox."
                },
                "attribute": "label",
                "reflect": false,
                "defaultValue": "''"
            },
            "required": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Whether the checkbox is required. If true, the checkbox must be checked when submitting a form.\nDefault is false. If the checkbox is required and not checked, an error message is displayed."
                },
                "attribute": "required",
                "reflect": false,
                "defaultValue": "false"
            }
        };
    }
    static get states() {
        return {
            "isChecked": {},
            "error": {}
        };
    }
    static get events() {
        return [{
                "method": "nylasFormCheckboxToggled",
                "name": "nylasFormCheckboxToggled",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the checkbox is toggled."
                },
                "complexType": {
                    "original": "{\n    checked: boolean;\n    name: string;\n    label: string;\n  }",
                    "resolved": "{ checked: boolean; name: string; label: string; }",
                    "references": {}
                }
            }];
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "checked",
                "methodName": "checkedPropChangedHandler"
            }];
    }
    static get listeners() {
        return [{
                "name": "bookingFormSubmitted",
                "method": "handleBookingFormSubmitted",
                "target": "document",
                "capture": false,
                "passive": false
            }];
    }
}
