// @nylas/web-elements/dist/collection/components/design-system/button-component/button-component.js
import { h } from "@stencil/core";
export class ButtonComponent {
    constructor() {
        this.variant = 'primary';
        this.disabled = false;
        this.type = 'button';
        this.tooltip = '';
        this.isLoading = false;
        this.clickHandler = undefined;
        this.mouseOverHandler = undefined;
        this.mouseOutHandler = undefined;
        this.focusHandler = undefined;
        this.blurHandler = undefined;
    }
    render() {
        const className = `button-${this.variant}` + (this.disabled ? ' disabled' : '');
        return (h("button", { key: 'c2ea933b85e63b6305793712f65cf335e7361866', class: className, disabled: this.disabled, type: "type", title: this.tooltip, onClick: this.clickHandler, onMouseOver: this.mouseOverHandler, onMouseOut: this.mouseOutHandler, onFocus: this.focusHandler, onBlur: this.blurHandler }, !!this.isLoading && h("loading-icon", { slot: "icon" }), h("slot", { key: 'faa43f2d964f51c5fd83e09cebc97b6eeb1fe5ab' })));
    }
    static get is() { return "button-component"; }
    static get encapsulation() { return "scoped"; }
    static get originalStyleUrls() {
        return {
            "$": ["button-component.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["button-component.css"]
        };
    }
    static get properties() {
        return {
            "variant": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "ButtonType",
                    "resolved": "\"basic\" | \"destructive\" | \"invisible\" | \"primary\"",
                    "references": {
                        "ButtonType": {
                            "location": "global",
                            "id": "global::ButtonType"
                        }
                    }
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "attribute": "variant",
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "disabled": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "attribute": "disabled",
                "reflect": false,
                "defaultValue": "false"
            },
            "type": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "attribute": "type",
                "reflect": false,
                "defaultValue": "'button'"
            },
            "tooltip": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "attribute": "tooltip",
                "reflect": false,
                "defaultValue": "''"
            },
            "isLoading": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "attribute": "is-loading",
                "reflect": false,
                "defaultValue": "false"
            },
            "clickHandler": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "(event: MouseEvent) => void",
                    "resolved": "((event: MouseEvent) => void) | undefined",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                }
            },
            "mouseOverHandler": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "(event: MouseEvent) => void",
                    "resolved": "((event: MouseEvent) => void) | undefined",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                }
            },
            "mouseOutHandler": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "(event: MouseEvent) => void",
                    "resolved": "((event: MouseEvent) => void) | undefined",
                    "references": {
                        "MouseEvent": {
                            "location": "global",
                            "id": "global::MouseEvent"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                }
            },
            "focusHandler": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "(event: FocusEvent) => void",
                    "resolved": "((event: FocusEvent) => void) | undefined",
                    "references": {
                        "FocusEvent": {
                            "location": "global",
                            "id": "global::FocusEvent"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                }
            },
            "blurHandler": {
                "type": "unknown",
                "mutable": false,
                "complexType": {
                    "original": "(event: FocusEvent) => void",
                    "resolved": "((event: FocusEvent) => void) | undefined",
                    "references": {
                        "FocusEvent": {
                            "location": "global",
                            "id": "global::FocusEvent"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": ""
                }
            }
        };
    }
}
