// @nylas/web-elements/dist/collection/components/design-system/input-color-picker/input-color-picker.js
import { h } from "@stencil/core";
export class InputColorPicker {
    constructor() {
        this.componentType = 'color-picker';
        this.name = undefined;
        this.defaultSelectedColor = undefined;
        this.selectedColor = this.defaultSelectedColor;
        this.isOpen = false;
        this.ariaActivedescendant = '';
        this.errorMessage = '';
    }
    componentWillLoad() {
        this.selectedColor = this.defaultSelectedColor;
        this.el.setAttribute('component-type', this.componentType);
    }
    handleFocus() {
        this.isOpen = true;
    }
    handleBlur() {
        this.isOpen = false;
    }
    handleInput(event) {
        this.selectedColor = event.target.value;
    }
    handleChange(event) {
        this.selectedColor = event.target.value;
        this.nylasFormDropdownChanged.emit({
            value: this.selectedColor,
            name: this.name,
            error: this.errorMessage,
        });
    }
    handleKeyDown(event) {
        if (event.code === 'Enter' || event.code === 'Space') {
            event.preventDefault();
            this.inputRef?.click();
        }
    }
    handleOutsideClick(event) {
        const path = event.composedPath();
        const isClickInside = path.includes(this.el);
        if (!isClickInside && this.isOpen) {
            this.isOpen = false;
        }
    }
    render() {
        return (h("div", { key: '821db0f91c515797a660b950a35360c3c6b04a31', class: "color-picker-container", part: "icp_color-picker", tabindex: "0", onKeyDown: e => this.handleKeyDown(e) }, h("input", { key: '415a71f72ed2450052720b31f31a2d289ce6bc4d', type: "color", ref: el => (this.inputRef = el), id: this.name, name: this.name, value: this.selectedColor, tabindex: "-1", onFocus: () => this.handleFocus(), onBlur: () => this.handleBlur(), onInput: e => this.handleInput(e), onChange: e => this.handleChange(e), onKeyDown: e => this.handleKeyDown(e) }), h("label", { key: '2e5b60172e2b64528377ec40f6a8f88633372186', part: "icp_color-picker-button", htmlFor: this.name, class: { labelBtn: true, open: this.isOpen, error: !!this.errorMessage }, "aria-haspopup": "listbox", "aria-expanded": this.isOpen ? 'true' : 'false', "aria-label": this.name }, h("span", { key: '6e6e6fd8b989e877004a2bb5b33342c8a27c492c', class: "selected-option", part: "icp_color-picker-button-selected-label" }, this.selectedColor || 'Select a color'), h("span", { key: '4deb0977faf3c2259bc8e0e2aa2720c59503ec12', class: {
                open: this.isOpen,
                closed: !this.isOpen,
                chevron: true,
            }, "aria-hidden": "true" }, h("chevron-icon", { key: 'da16cec17511ed70e1a4dcdb8f2f08a3b7682674', width: "16", height: "16" })), this.errorMessage && h("span", { class: "error help-text" }, this.errorMessage))));
    }
    static get is() { return "input-color-picker"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["input-color-picker.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["input-color-picker.css"]
        };
    }
    static get properties() {
        return {
            "name": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "The name of the dropdown"
                },
                "attribute": "name",
                "reflect": false
            },
            "defaultSelectedColor": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string | undefined",
                    "references": {}
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The default selected color"
                },
                "attribute": "default-selected-color",
                "reflect": false
            }
        };
    }
    static get states() {
        return {
            "selectedColor": {},
            "isOpen": {},
            "ariaActivedescendant": {},
            "errorMessage": {}
        };
    }
    static get events() {
        return [{
                "method": "nylasFormDropdownChanged",
                "name": "nylasFormDropdownChanged",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": "This event is fired when the selected option is changed"
                },
                "complexType": {
                    "original": "{\n    value: string;\n    name: string;\n    error?: string;\n    label?: string;\n  }",
                    "resolved": "{ value: string; name: string; error?: string | undefined; label?: string | undefined; }",
                    "references": {}
                }
            }];
    }
    static get elementRef() { return "el"; }
    static get listeners() {
        return [{
                "name": "click",
                "method": "handleOutsideClick",
                "target": "document",
                "capture": true,
                "passive": false
            }];
    }
}
