// @nylas/web-elements/dist/collection/components/scheduler-editor/nylas-event-info/nylas-event-info.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
import i18next from "../../../utils/i18n";
export class NylasEventInfo {
    constructor() {
        this.selectedLanguage = undefined;
    }
    selectedLanguageChangedHandler(newVal, oldVal) {
        if (newVal === oldVal)
            return;
        i18next.changeLanguage(newVal);
    }
    connectedCallback() {
        debug('nylas-event-info', 'connectedCallback');
    }
    componentWillLoad() {
        debug('nylas-event-info', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-event-info', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    disconnectedCallback() {
        debug('nylas-event-info', 'disconnectedCallback');
    }
    render() {
        return (h(Host, { key: 'bc38653b8c5e4073e489128f0a7311b6796ea08c' }, h("nylas-form-card", { key: 'eb8c3f03b81873f7073526c227f3fb65e3ff5def', exportparts: "nfc__header: nei__header" }, h("h1", { key: '18636006ebde96ee874bc0f4f6200b2c30985371', slot: "header-title", class: "nylas-event-info__title", part: "nei__title" }, i18next.t('nylasEventInfo.headerTitle')), h("div", { key: '150b7b72eb936c1e50cf9053224939dbe7fb893f', slot: "content", class: "nylas-event-info__content" }, h("slot", { key: '93b7b7b333009214666dd6e10612cd8a4e11292a', name: "inputs" })))));
    }
    static get is() { return "nylas-event-info"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-event-info.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-event-info.css"]
        };
    }
    static get properties() {
        return {
            "selectedLanguage": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "LANGUAGE_CODE",
                    "resolved": "LANGUAGE_CODE.de | LANGUAGE_CODE.en | LANGUAGE_CODE.es | LANGUAGE_CODE.fr | LANGUAGE_CODE.ja | LANGUAGE_CODE.nl | LANGUAGE_CODE.sv | LANGUAGE_CODE.zh | undefined",
                    "references": {
                        "LANGUAGE_CODE": {
                            "location": "import",
                            "path": "@/common/constants",
                            "id": "src/common/constants.ts::LANGUAGE_CODE"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The selected language."
                },
                "attribute": "selected-language",
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "selectedLanguage",
                "methodName": "selectedLanguageChangedHandler"
            }];
    }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-event-info',
        stateToProps: new Map([['schedulerConfig.selectedLanguage', 'selectedLanguage']]),
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasEventInfo.prototype, "render", null);
