// @nylas/web-elements/dist/collection/components/scheduler-editor/nylas-custom-booking-flow/nylas-custom-booking-flow.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
import { RegisterComponent } from "../../../common/register-component";
import { checkForMissingSlots, debug } from "../../../utils/utils";
import { Host, h } from "@stencil/core";
import i18next from "../../../utils/i18n";
export class NylasCustomBookingFlow {
    constructor() {
        this.selectedLanguage = undefined;
    }
    selectedLanguageChangedHandler(newVal, oldVal) {
        if (newVal === oldVal)
            return;
        i18next.changeLanguage(newVal);
    }
    connectedCallback() {
        debug('nylas-custom-booking-flow', 'connectedCallback');
    }
    disconnectedCallback() {
        debug('nylas-custom-booking-flow', 'disconnectedCallback');
    }
    componentWillLoad() {
        debug('nylas-custom-booking-flow', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-custom-booking-flow', 'componentDidLoad');
        checkForMissingSlots(['inputs'], this.el);
    }
    render() {
        return (h(Host, { key: '58cb7aa5e903ccfb94eec8f736184d0b3c5be870', part: "ncbflow" }, h("nylas-form-card", { key: '3f20d0782c85e3c2cad19e714fe824903017f6c9', exportparts: "nfc__header: ncbflow__header" }, h("h1", { key: '0ec0a40d24501a9be682f36c2205aabccaafcbc5', slot: "header-title", class: "nylas-custom-booking-flow__title", part: "ncbflow__title" }, i18next.t('nylasCustomBookingFlow.headerTitle')), h("div", { key: 'cb769c671be867a59e02c82983ef1bea3a4a71bc', slot: "content", class: "nylas-custom-booking-flow__content", part: "ncbflow__content" }, h("slot", { key: '78d0d4e24c7d3f5c404d2e8af1d0354da965d23c', name: "inputs" })))));
    }
    static get is() { return "nylas-custom-booking-flow"; }
    static get encapsulation() { return "shadow"; }
    static get formAssociated() { return true; }
    static get originalStyleUrls() {
        return {
            "$": ["nylas-custom-booking-flow.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["nylas-custom-booking-flow.css"]
        };
    }
    static get properties() {
        return {
            "selectedLanguage": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "LANGUAGE_CODE",
                    "resolved": "LANGUAGE_CODE.de | LANGUAGE_CODE.en | LANGUAGE_CODE.es | LANGUAGE_CODE.fr | LANGUAGE_CODE.ja | LANGUAGE_CODE.nl | LANGUAGE_CODE.sv | LANGUAGE_CODE.zh | undefined",
                    "references": {
                        "LANGUAGE_CODE": {
                            "location": "import",
                            "path": "@/common/constants",
                            "id": "src/common/constants.ts::LANGUAGE_CODE"
                        }
                    }
                },
                "required": false,
                "optional": true,
                "docs": {
                    "tags": [],
                    "text": "The selected language."
                },
                "attribute": "selected-language",
                "reflect": false
            }
        };
    }
    static get elementRef() { return "el"; }
    static get watchers() {
        return [{
                "propName": "selectedLanguage",
                "methodName": "selectedLanguageChangedHandler"
            }];
    }
    static get attachInternalsMemberName() { return "internals"; }
}
__decorate([
    RegisterComponent({
        name: 'nylas-custom-booking-flow',
        eventToProps: {},
        stateToProps: new Map([['schedulerConfig.selectedLanguage', 'selectedLanguage']]),
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasCustomBookingFlow.prototype, "render", null);
