// @nylas/web-elements/dist/cdn/microsoft-logo/microsoft-logo.es.js
const b = {
  allRenderFn: !1,
  cmpDidLoad: !0,
  cmpDidUnload: !1,
  cmpDidUpdate: !0,
  cmpDidRender: !0,
  cmpWillLoad: !0,
  cmpWillUpdate: !0,
  cmpWillRender: !0,
  connectedCallback: !0,
  disconnectedCallback: !0,
  element: !0,
  event: !0,
  hasRenderFn: !0,
  lifecycle: !0,
  hostListener: !0,
  hostListenerTargetWindow: !0,
  hostListenerTargetDocument: !0,
  hostListenerTargetBody: !0,
  hostListenerTargetParent: !1,
  hostListenerTarget: !0,
  member: !0,
  method: !0,
  mode: !0,
  observeAttribute: !0,
  prop: !0,
  propMutable: !0,
  reflect: !0,
  scoped: !0,
  shadowDom: !0,
  slot: !0,
  cssAnnotations: !0,
  state: !0,
  style: !0,
  formAssociated: !1,
  svg: !0,
  updatable: !0,
  vdomAttribute: !0,
  vdomXlink: !0,
  vdomClass: !0,
  vdomFunctional: !0,
  vdomKey: !0,
  vdomListener: !0,
  vdomRef: !0,
  vdomPropOrAttr: !0,
  vdomRender: !0,
  vdomStyle: !0,
  vdomText: !0,
  watchCallback: !0,
  taskQueue: !0,
  hotModuleReplacement: !1,
  isDebug: !1,
  isDev: !1,
  isTesting: !1,
  hydrateServerSide: !1,
  hydrateClientSide: !1,
  lifecycleDOMEvents: !1,
  lazyLoad: !1,
  profile: !1,
  slotRelocation: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  appendChildSlotFix: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  cloneNodeFix: !1,
  hydratedAttribute: !1,
  hydratedClass: !0,
  scriptDataOpts: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  scopedSlotTextContentFix: !1,
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  shadowDomShim: !1,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  slotChildNodesFix: !1,
  invisiblePrehydration: !0,
  propBoolean: !0,
  propNumber: !0,
  propString: !0,
  constructableCSS: !0,
  cmpShouldUpdate: !0,
  devTools: !1,
  shadowDelegatesFocus: !0,
  initializeNextTick: !1,
  asyncLoading: !1,
  asyncQueue: !1,
  transformTagName: !1,
  attachStyles: !0,
  // TODO(STENCIL-914): remove this option when `experimentalSlotFixes` is the default behavior
  experimentalSlotFixes: !1
};
let E, j, R, lt = !1, M = !1, W = !1, v = !1, I = null, D = !1;
const L = (t, e = "") => () => {
}, St = "slot-fb{display:contents}slot-fb[hidden]{display:none}", Q = "http://www.w3.org/1999/xlink", X = {}, mt = "http://www.w3.org/2000/svg", kt = "http://www.w3.org/1999/xhtml", bt = (t) => t != null, z = (t) => (t = typeof t, t === "object" || t === "function");
function Tt(t) {
  var e, n, s;
  return (s = (n = (e = t.head) === null || e === void 0 ? void 0 : e.querySelector('meta[name="csp-nonce"]')) === null || n === void 0 ? void 0 : n.getAttribute("content")) !== null && s !== void 0 ? s : void 0;
}
const x = (t, e, ...n) => {
  let s = null, o = null, c = null, l = !1, r = !1;
  const i = [], u = (a) => {
    for (let d = 0; d < a.length; d++)
      s = a[d], Array.isArray(s) ? u(s) : s != null && typeof s != "boolean" && ((l = typeof t != "function" && !z(s)) && (s = String(s)), l && r ? i[i.length - 1].$text$ += s : i.push(l ? w(null, s) : s), r = l);
  };
  if (u(n), e) {
    e.key && (o = e.key), e.name && (c = e.name);
    {
      const a = e.className || e.class;
      a && (e.class = typeof a != "object" ? a : Object.keys(a).filter((d) => a[d]).join(" "));
    }
  }
  if (typeof t == "function")
    return t(e === null ? {} : e, i, Et);
  const f = w(t, null);
  return f.$attrs$ = e, i.length > 0 && (f.$children$ = i), f.$key$ = o, f.$name$ = c, f;
}, w = (t, e) => {
  const n = {
    $flags$: 0,
    $tag$: t,
    $text$: e,
    $elm$: null,
    $children$: null
  };
  return n.$attrs$ = null, n.$key$ = null, n.$name$ = null, n;
}, xt = {}, Lt = (t) => t && t.$tag$ === xt, Et = {
  forEach: (t, e) => t.map(Z).forEach(e),
  map: (t, e) => t.map(Z).map(e).map(_t)
}, Z = (t) => ({
  vattrs: t.$attrs$,
  vchildren: t.$children$,
  vkey: t.$key$,
  vname: t.$name$,
  vtag: t.$tag$,
  vtext: t.$text$
}), _t = (t) => {
  if (typeof t.vtag == "function") {
    const n = Object.assign({}, t.vattrs);
    return t.vkey && (n.key = t.vkey), t.vname && (n.name = t.vname), x(t.vtag, n, ...t.vchildren || []);
  }
  const e = w(t.vtag, t.vtext);
  return e.$attrs$ = t.vattrs, e.$children$ = t.vchildren, e.$key$ = t.vkey, e.$name$ = t.vname, e;
}, At = (t) => ee.map((e) => e(t)).find((e) => !!e), Rt = (t, e) => t != null && !z(t) ? e & 4 ? t === "false" ? !1 : t === "" || !!t : e & 2 ? parseFloat(t) : e & 1 ? String(t) : t : t, G = /* @__PURE__ */ new WeakMap(), Ot = (t, e, n) => {
  let s = U.get(t);
  le && n ? (s = s || new CSSStyleSheet(), typeof s == "string" ? s = e : s.replaceSync(e)) : s = e, U.set(t, s);
}, Ft = (t, e, n) => {
  var s;
  const o = it(e, n), c = U.get(o);
  if (t = t.nodeType === 11 ? t : S, c)
    if (typeof c == "string") {
      t = t.head || t;
      let l = G.get(t), r;
      if (l || G.set(t, l = /* @__PURE__ */ new Set()), !l.has(o)) {
        {
          r = S.createElement("style"), r.innerHTML = c;
          const i = (s = p.$nonce$) !== null && s !== void 0 ? s : Tt(S);
          i != null && r.setAttribute("nonce", i), t.insertBefore(r, t.querySelector("link"));
        }
        e.$flags$ & 4 && (r.innerHTML += St), l && l.add(o);
      }
    } else t.adoptedStyleSheets.includes(c) || (t.adoptedStyleSheets = [...t.adoptedStyleSheets, c]);
  return o;
}, Ht = (t) => {
  const e = t.$cmpMeta$, n = t.$hostElement$, s = e.$flags$, o = L("attachStyles", e.$tagName$), c = Ft(n.shadowRoot ? n.shadowRoot : n.getRootNode(), e, t.$modeName$);
  s & 10 && (n["s-sc"] = c, n.classList.add(c + "-h"), s & 2 && n.classList.add(c + "-s")), o();
}, it = (t, e) => "sc-" + (e && t.$flags$ & 32 ? t.$tagName$ + "-" + e : t.$tagName$), K = (t, e, n, s, o, c) => {
  if (n !== s) {
    let l = et(t, e), r = e.toLowerCase();
    if (e === "class") {
      const i = t.classList, u = J(n), f = J(s);
      i.remove(...u.filter((a) => a && !f.includes(a))), i.add(...f.filter((a) => a && !u.includes(a)));
    } else if (e === "style") {
      for (const i in n)
        (!s || s[i] == null) && (i.includes("-") ? t.style.removeProperty(i) : t.style[i] = "");
      for (const i in s)
        (!n || s[i] !== n[i]) && (i.includes("-") ? t.style.setProperty(i, s[i]) : t.style[i] = s[i]);
    } else if (e !== "key")
      if (e === "ref")
        s && s(t);
      else if (!t.__lookupSetter__(e) && e[0] === "o" && e[1] === "n") {
        if (e[2] === "-" ? e = e.slice(3) : et(P, r) ? e = r.slice(2) : e = r[2] + e.slice(3), n || s) {
          const i = e.endsWith(ct);
          e = e.replace(Mt, ""), n && p.rel(t, e, n, i), s && p.ael(t, e, s, i);
        }
      } else {
        const i = z(s);
        if ((l || i && s !== null) && !o)
          try {
            if (t.tagName.includes("-"))
              t[e] = s;
            else {
              const f = s ?? "";
              e === "list" ? l = !1 : (n == null || t[e] != f) && (t[e] = f);
            }
          } catch {
          }
        let u = !1;
        r !== (r = r.replace(/^xlink\:?/, "")) && (e = r, u = !0), s == null || s === !1 ? (s !== !1 || t.getAttribute(e) === "") && (u ? t.removeAttributeNS(Q, e) : t.removeAttribute(e)) : (!l || c & 4 || o) && !i && (s = s === !0 ? "" : s, u ? t.setAttributeNS(Q, e, s) : t.setAttribute(e, s));
      }
  }
}, Ct = /\s/, J = (t) => t ? t.split(Ct) : [], ct = "Capture", Mt = new RegExp(ct + "$"), rt = (t, e, n, s) => {
  const o = e.$elm$.nodeType === 11 && e.$elm$.host ? e.$elm$.host : e.$elm$, c = t && t.$attrs$ || X, l = e.$attrs$ || X;
  for (s in c)
    s in l || K(o, s, c[s], void 0, n, e.$flags$);
  for (s in l)
    K(o, s, c[s], l[s], n, e.$flags$);
}, B = (t, e, n, s) => {
  const o = e.$children$[n];
  let c = 0, l, r, i;
  if (lt || (W = !0, o.$tag$ === "slot" && (E && s.classList.add(E + "-s"), o.$flags$ |= o.$children$ ? (
    // slot element has fallback content
    2
  ) : (
    // slot element does not have fallback content
    1
  ))), o.$text$ !== null)
    l = o.$elm$ = S.createTextNode(o.$text$);
  else if (o.$flags$ & 1)
    l = o.$elm$ = S.createTextNode("");
  else {
    if (v || (v = o.$tag$ === "svg"), l = o.$elm$ = S.createElementNS(v ? mt : kt, o.$flags$ & 2 ? "slot-fb" : o.$tag$), v && o.$tag$ === "foreignObject" && (v = !1), rt(null, o, v), bt(E) && l["s-si"] !== E && l.classList.add(l["s-si"] = E), o.$children$)
      for (c = 0; c < o.$children$.length; ++c)
        r = B(t, o, c, l), r && l.appendChild(r);
    o.$tag$ === "svg" ? v = !1 : l.tagName === "foreignObject" && (v = !0);
  }
  return l["s-hn"] = R, o.$flags$ & 3 && (l["s-sr"] = !0, l["s-cr"] = j, l["s-sn"] = o.$name$ || "", i = t && t.$children$ && t.$children$[n], i && i.$tag$ === o.$tag$ && t.$elm$ && O(t.$elm$, !1)), l;
}, O = (t, e) => {
  p.$flags$ |= 1;
  const n = t.childNodes;
  for (let s = n.length - 1; s >= 0; s--) {
    const o = n[s];
    o["s-hn"] !== R && o["s-ol"] && (ft(o).insertBefore(o, q(o)), o["s-ol"].remove(), o["s-ol"] = void 0, o["s-sh"] = void 0, W = !0), e && O(o, e);
  }
  p.$flags$ &= -2;
}, at = (t, e, n, s, o, c) => {
  let l = t["s-cr"] && t["s-cr"].parentNode || t, r;
  for (l.shadowRoot && l.tagName === R && (l = l.shadowRoot); o <= c; ++o)
    s[o] && (r = B(null, n, o, t), r && (s[o].$elm$ = r, l.insertBefore(r, q(e))));
}, $t = (t, e, n) => {
  for (let s = e; s <= n; ++s) {
    const o = t[s];
    if (o) {
      const c = o.$elm$;
      ht(o), c && (M = !0, c["s-ol"] ? c["s-ol"].remove() : O(c, !0), c.remove());
    }
  }
}, wt = (t, e, n, s, o = !1) => {
  let c = 0, l = 0, r = 0, i = 0, u = e.length - 1, f = e[0], a = e[u], d = s.length - 1, $ = s[0], h = s[d], y, g;
  for (; c <= u && l <= d; )
    if (f == null)
      f = e[++c];
    else if (a == null)
      a = e[--u];
    else if ($ == null)
      $ = s[++l];
    else if (h == null)
      h = s[--d];
    else if (C(f, $, o))
      _(f, $, o), f = e[++c], $ = s[++l];
    else if (C(a, h, o))
      _(a, h, o), a = e[--u], h = s[--d];
    else if (C(f, h, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && O(f.$elm$.parentNode, !1), _(f, h, o), t.insertBefore(f.$elm$, a.$elm$.nextSibling), f = e[++c], h = s[--d];
    else if (C(a, $, o))
      (f.$tag$ === "slot" || h.$tag$ === "slot") && O(a.$elm$.parentNode, !1), _(a, $, o), t.insertBefore(a.$elm$, f.$elm$), a = e[--u], $ = s[++l];
    else {
      for (r = -1, i = c; i <= u; ++i)
        if (e[i] && e[i].$key$ !== null && e[i].$key$ === $.$key$) {
          r = i;
          break;
        }
      r >= 0 ? (g = e[r], g.$tag$ !== $.$tag$ ? y = B(e && e[l], n, r, t) : (_(g, $, o), e[r] = void 0, y = g.$elm$), $ = s[++l]) : (y = B(e && e[l], n, l, t), $ = s[++l]), y && ft(f.$elm$).insertBefore(y, q(f.$elm$));
    }
  c > u ? at(t, s[d + 1] == null ? null : s[d + 1].$elm$, n, s, l, d) : l > d && $t(e, c, u);
}, C = (t, e, n = !1) => t.$tag$ === e.$tag$ ? t.$tag$ === "slot" ? t.$name$ === e.$name$ : n ? !0 : t.$key$ === e.$key$ : !1, q = (t) => t && t["s-ol"] || t, ft = (t) => (t["s-ol"] ? t["s-ol"] : t).parentNode, _ = (t, e, n = !1) => {
  const s = e.$elm$ = t.$elm$, o = t.$children$, c = e.$children$, l = e.$tag$, r = e.$text$;
  let i;
  r === null ? (v = l === "svg" ? !0 : l === "foreignObject" ? !1 : v, l === "slot" || rt(t, e, v), o !== null && c !== null ? wt(s, o, e, c, n) : c !== null ? (t.$text$ !== null && (s.textContent = ""), at(s, null, e, c, 0, c.length - 1)) : o !== null && $t(o, 0, o.length - 1), v && l === "svg" && (v = !1)) : (i = s["s-cr"]) ? i.parentNode.textContent = r : t.$text$ !== r && (s.data = r);
}, dt = (t) => {
  const e = t.childNodes;
  for (const n of e)
    if (n.nodeType === 1) {
      if (n["s-sr"]) {
        const s = n["s-sn"];
        n.hidden = !1;
        for (const o of e)
          if (o !== n) {
            if (o["s-hn"] !== n["s-hn"] || s !== "") {
              if (o.nodeType === 1 && (s === o.getAttribute("slot") || s === o["s-sn"])) {
                n.hidden = !0;
                break;
              }
            } else if (o.nodeType === 1 || o.nodeType === 3 && o.textContent.trim() !== "") {
              n.hidden = !0;
              break;
            }
          }
      }
      dt(n);
    }
}, m = [], ut = (t) => {
  let e, n, s;
  for (const o of t.childNodes) {
    if (o["s-sr"] && (e = o["s-cr"]) && e.parentNode) {
      n = e.parentNode.childNodes;
      const c = o["s-sn"];
      for (s = n.length - 1; s >= 0; s--)
        if (e = n[s], !e["s-cn"] && !e["s-nr"] && e["s-hn"] !== o["s-hn"] && !b.experimentalSlotFixes)
          if (Y(e, c)) {
            let l = m.find((r) => r.$nodeToRelocate$ === e);
            M = !0, e["s-sn"] = e["s-sn"] || c, l ? (l.$nodeToRelocate$["s-sh"] = o["s-hn"], l.$slotRefNode$ = o) : (e["s-sh"] = o["s-hn"], m.push({
              $slotRefNode$: o,
              $nodeToRelocate$: e
            })), e["s-sr"] && m.map((r) => {
              Y(r.$nodeToRelocate$, e["s-sn"]) && (l = m.find((i) => i.$nodeToRelocate$ === e), l && !r.$slotRefNode$ && (r.$slotRefNode$ = l.$slotRefNode$));
            });
          } else m.some((l) => l.$nodeToRelocate$ === e) || m.push({
            $nodeToRelocate$: e
          });
    }
    o.nodeType === 1 && ut(o);
  }
}, Y = (t, e) => t.nodeType === 1 ? t.getAttribute("slot") === null && e === "" || t.getAttribute("slot") === e : t["s-sn"] === e ? !0 : e === "", ht = (t) => {
  t.$attrs$ && t.$attrs$.ref && t.$attrs$.ref(null), t.$children$ && t.$children$.map(ht);
}, Bt = (t, e, n = !1) => {
  var s, o, c, l, r;
  const i = t.$hostElement$, u = t.$cmpMeta$, f = t.$vnode$ || w(null, null), a = Lt(e) ? e : x(null, null, e);
  if (R = i.tagName, u.$attrsToReflect$ && (a.$attrs$ = a.$attrs$ || {}, u.$attrsToReflect$.map(([d, $]) => a.$attrs$[$] = i[d])), n && a.$attrs$)
    for (const d of Object.keys(a.$attrs$))
      i.hasAttribute(d) && !["key", "ref", "style", "class"].includes(d) && (a.$attrs$[d] = i[d]);
  a.$tag$ = null, a.$flags$ |= 4, t.$vnode$ = a, a.$elm$ = f.$elm$ = i.shadowRoot || i, E = i["s-sc"], j = i["s-cr"], lt = (u.$flags$ & 1) !== 0, M = !1, _(f, a, n);
  {
    if (p.$flags$ |= 1, W) {
      ut(a.$elm$);
      for (const d of m) {
        const $ = d.$nodeToRelocate$;
        if (!$["s-ol"]) {
          const h = S.createTextNode("");
          h["s-nr"] = $, $.parentNode.insertBefore($["s-ol"] = h, $);
        }
      }
      for (const d of m) {
        const $ = d.$nodeToRelocate$, h = d.$slotRefNode$;
        if (h) {
          const y = h.parentNode;
          let g = h.nextSibling;
          {
            let T = (s = $["s-ol"]) === null || s === void 0 ? void 0 : s.previousSibling;
            for (; T; ) {
              let k = (o = T["s-nr"]) !== null && o !== void 0 ? o : null;
              if (k && k["s-sn"] === $["s-sn"] && y === k.parentNode && (k = k.nextSibling, !k || !k["s-nr"])) {
                g = k;
                break;
              }
              T = T.previousSibling;
            }
          }
          (!g && y !== $.parentNode || $.nextSibling !== g) && $ !== g && (!$["s-hn"] && $["s-ol"] && ($["s-hn"] = $["s-ol"].parentNode.nodeName), y.insertBefore($, g), $.nodeType === 1 && ($.hidden = (c = $["s-ih"]) !== null && c !== void 0 ? c : !1));
        } else
          $.nodeType === 1 && (n && ($["s-ih"] = (l = $.hidden) !== null && l !== void 0 ? l : !1), $.hidden = !0);
      }
    }
    M && dt(a.$elm$), p.$flags$ &= -2, m.length = 0;
  }
  if (b.experimentalScopedSlotChanges && u.$flags$ & 2)
    for (const d of a.$elm$.childNodes)
      d["s-hn"] !== R && !d["s-sh"] && (n && d["s-ih"] == null && (d["s-ih"] = (r = d.hidden) !== null && r !== void 0 ? r : !1), d.hidden = !0);
  j = void 0;
}, Ut = (t, e) => {
}, pt = (t, e) => (t.$flags$ |= 16, Ut(t, t.$ancestorComponent$), re(() => Pt(t, e))), Pt = (t, e) => {
  const n = t.$hostElement$, s = L("scheduleUpdate", t.$cmpMeta$.$tagName$), o = n;
  let c;
  return e ? c = A(o, "componentWillLoad") : c = A(o, "componentWillUpdate"), c = V(c, () => A(o, "componentWillRender")), s(), V(c, () => Dt(t, o, e));
}, V = (t, e) => jt(t) ? t.then(e) : e(), jt = (t) => t instanceof Promise || t && t.then && typeof t.then == "function", Dt = async (t, e, n) => {
  const s = t.$hostElement$, o = L("update", t.$cmpMeta$.$tagName$);
  s["s-rc"], n && Ht(t);
  const c = L("render", t.$cmpMeta$.$tagName$);
  Nt(t, e, s, n), c(), o(), Wt(t);
}, Nt = (t, e, n, s) => {
  try {
    I = e, e = e.render && e.render(), t.$flags$ &= -17, t.$flags$ |= 2, (b.hasRenderFn || b.reflect) && (b.vdomRender || b.reflect) && (b.hydrateServerSide || Bt(t, e, s));
  } catch (i) {
    H(i, t.$hostElement$);
  }
  return I = null, null;
}, Wt = (t) => {
  const e = t.$cmpMeta$.$tagName$, n = t.$hostElement$, s = L("postUpdate", e), o = n;
  t.$ancestorComponent$, A(o, "componentDidRender"), t.$flags$ & 64 ? (A(o, "componentDidUpdate"), s()) : (t.$flags$ |= 64, A(o, "componentDidLoad"), s());
}, A = (t, e, n) => {
  if (t && t[e])
    try {
      return t[e](n);
    } catch (s) {
      H(s);
    }
}, zt = (t, e) => F(t).$instanceValues$.get(e), qt = (t, e, n, s) => {
  const o = F(t), c = t, l = o.$instanceValues$.get(e), r = o.$flags$, i = c;
  n = Rt(n, s.$members$[e][0]);
  const u = Number.isNaN(l) && Number.isNaN(n);
  if (n !== l && !u) {
    o.$instanceValues$.set(e, n);
    {
      if (s.$watchers$ && r & 128) {
        const a = s.$watchers$[e];
        a && a.map((d) => {
          try {
            i[d](n, l, e);
          } catch ($) {
            H($, c);
          }
        });
      }
      if ((r & 18) === 2) {
        if (i.componentShouldUpdate && i.componentShouldUpdate(n, l, e) === !1)
          return;
        pt(o, !1);
      }
    }
  }
}, It = (t, e, n) => {
  var s;
  const o = t.prototype;
  if (e.$members$) {
    t.watchers && (e.$watchers$ = t.watchers);
    const c = Object.entries(e.$members$);
    c.map(([l, [r]]) => {
      (r & 31 || r & 32) && Object.defineProperty(o, l, {
        get() {
          return zt(this, l);
        },
        set(i) {
          qt(this, l, i, e);
        },
        configurable: !0,
        enumerable: !0
      });
    });
    {
      const l = /* @__PURE__ */ new Map();
      o.attributeChangedCallback = function(r, i, u) {
        p.jmp(() => {
          var f;
          const a = l.get(r);
          if (this.hasOwnProperty(a))
            u = this[a], delete this[a];
          else {
            if (o.hasOwnProperty(a) && typeof this[a] == "number" && this[a] == u)
              return;
            if (a == null) {
              const d = F(this), $ = d == null ? void 0 : d.$flags$;
              if ($ && !($ & 8) && $ & 128 && u !== i) {
                const y = this, g = (f = e.$watchers$) === null || f === void 0 ? void 0 : f[r];
                g == null || g.forEach((T) => {
                  y[T] != null && y[T].call(y, u, i, r);
                });
              }
              return;
            }
          }
          this[a] = u === null && typeof this[a] == "boolean" ? !1 : u;
        });
      }, t.observedAttributes = Array.from(/* @__PURE__ */ new Set([
        ...Object.keys((s = e.$watchers$) !== null && s !== void 0 ? s : {}),
        ...c.filter(
          ([r, i]) => i[0] & 15
          /* MEMBER_FLAGS.HasAttribute */
        ).map(([r, i]) => {
          var u;
          const f = i[1] || r;
          return l.set(f, r), i[0] & 512 && ((u = e.$attrsToReflect$) === null || u === void 0 || u.push([r, f])), f;
        })
      ]));
    }
  }
  return t;
}, Qt = async (t, e, n, s) => {
  let o;
  if (!(e.$flags$ & 32) && (e.$flags$ |= 32, o = t.constructor, customElements.whenDefined(n.$tagName$).then(() => e.$flags$ |= 128), o.style)) {
    let l = o.style;
    typeof l != "string" && (l = l[e.$modeName$ = At(t)]);
    const r = it(n, e.$modeName$);
    if (!U.has(r)) {
      const i = L("registerStyles", n.$tagName$);
      Ot(r, l, !!(n.$flags$ & 1)), i();
    }
  }
  e.$ancestorComponent$, pt(e, !0);
}, tt = (t) => {
}, Xt = (t) => {
  if (!(p.$flags$ & 1)) {
    const e = F(t), n = e.$cmpMeta$, s = L("connectedCallback", n.$tagName$);
    e.$flags$ & 1 ? (gt(t, e, n.$listeners$), e != null && e.$lazyInstance$ ? tt(e.$lazyInstance$) : e != null && e.$onReadyPromise$ && e.$onReadyPromise$.then(() => tt(e.$lazyInstance$))) : (e.$flags$ |= 1, // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
    n.$flags$ & 12 && Zt(t), n.$members$ && Object.entries(n.$members$).map(([o, [c]]) => {
      if (c & 31 && t.hasOwnProperty(o)) {
        const l = t[o];
        delete t[o], t[o] = l;
      }
    }), Qt(t, e, n)), s();
  }
}, Zt = (t) => {
  const e = t["s-cr"] = S.createComment("");
  e["s-cn"] = !0, t.insertBefore(e, t.firstChild);
}, Gt = async (t) => {
  if (!(p.$flags$ & 1)) {
    const e = F(t);
    e.$rmListeners$ && (e.$rmListeners$.map((n) => n()), e.$rmListeners$ = void 0);
  }
}, Kt = (t, e) => {
  const n = {
    $flags$: e[0],
    $tagName$: e[1]
  };
  n.$members$ = e[2], n.$listeners$ = e[3], n.$watchers$ = t.$watchers$, n.$attrsToReflect$ = [], !st && n.$flags$ & 1 && (n.$flags$ |= 8);
  const s = t.prototype.connectedCallback, o = t.prototype.disconnectedCallback;
  return Object.assign(t.prototype, {
    __registerHost() {
      te(this, n);
    },
    connectedCallback() {
      Xt(this), s && s.call(this);
    },
    disconnectedCallback() {
      Gt(this), o && o.call(this);
    },
    __attachShadow() {
      st ? this.attachShadow({
        mode: "open",
        delegatesFocus: !!(n.$flags$ & 16)
      }) : this.shadowRoot = this;
    }
  }), t.is = n.$tagName$, It(t, n);
}, gt = (t, e, n, s) => {
  n && n.map(([o, c, l]) => {
    const r = Yt(t, o), i = Jt(e, l), u = Vt(o);
    p.ael(r, c, i, u), (e.$rmListeners$ = e.$rmListeners$ || []).push(() => p.rel(r, c, i, u));
  });
}, Jt = (t, e) => (n) => {
  try {
    b.lazyLoad || t.$hostElement$[e](n);
  } catch (s) {
    H(s);
  }
}, Yt = (t, e) => e & 4 ? S : e & 8 ? P : e & 16 ? S.body : t, Vt = (t) => ne ? {
  passive: (t & 1) !== 0,
  capture: (t & 2) !== 0
} : (t & 2) !== 0, yt = /* @__PURE__ */ new WeakMap(), F = (t) => yt.get(t), te = (t, e) => {
  const n = {
    $flags$: 0,
    $hostElement$: t,
    $cmpMeta$: e,
    $instanceValues$: /* @__PURE__ */ new Map()
  };
  return gt(t, n, e.$listeners$), yt.set(t, n);
}, et = (t, e) => e in t, H = (t, e) => (0, console.error)(t, e), U = /* @__PURE__ */ new Map(), ee = [], P = typeof window < "u" ? window : {}, S = P.document || { head: {} }, se = P.HTMLElement || class {
}, p = {
  $flags$: 0,
  $resourcesUrl$: "",
  jmp: (t) => t(),
  raf: (t) => requestAnimationFrame(t),
  ael: (t, e, n, s) => t.addEventListener(e, n, s),
  rel: (t, e, n, s) => t.removeEventListener(e, n, s),
  ce: (t, e) => new CustomEvent(t, e)
}, st = (
  // TODO(STENCIL-854): Remove code related to legacy shadowDomShim field
  !0
), ne = /* @__PURE__ */ (() => {
  let t = !1;
  try {
    S.addEventListener("e", null, Object.defineProperty({}, "passive", {
      get() {
        t = !0;
      }
    }));
  } catch {
  }
  return t;
})(), oe = (t) => Promise.resolve(t), le = /* @__PURE__ */ (() => {
  try {
    return new CSSStyleSheet(), typeof new CSSStyleSheet().replaceSync == "function";
  } catch {
  }
  return !1;
})(), nt = [], vt = [], ie = (t, e) => (n) => {
  t.push(n), D || (D = !0, e && p.$flags$ & 4 ? ce(N) : p.raf(N));
}, ot = (t) => {
  for (let e = 0; e < t.length; e++)
    try {
      t[e](performance.now());
    } catch (n) {
      H(n);
    }
  t.length = 0;
}, N = () => {
  ot(nt), ot(vt), (D = nt.length > 0) && p.raf(N);
}, ce = (t) => oe().then(t), re = /* @__PURE__ */ ie(vt, !0), ae = ".sc-microsoft-logo-icon-h{display:flex}", $e = ae, fe = Kt(class extends se {
  constructor() {
    super(), this.__registerHost(), this.width = "22", this.height = "22";
  }
  render() {
    return x("svg", { key: "970afb49d36b25194c649b69ca0f136a588c20a4", width: this.width, height: this.height, viewBox: "0 0 22 22", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, x("path", { key: "b3f4f95ead19fff86c4026bcf226a31a22b9c09a", d: "M0.043457 0.043457H10.4782V10.4782H0.043457V0.043457Z", fill: "#F35325" }), x("path", { key: "425e0c099da7b5e3637665f2b9240aab9dd43589", d: "M11.5217 0.043457H21.9565V10.4782H11.5217V0.043457Z", fill: "#81BC06" }), x("path", { key: "2f531126f18dd92603d2caaa31dfaf8a3dac3c9f", d: "M0.043457 11.5217H10.4782V21.9565H0.043457V11.5217Z", fill: "#05A6F0" }), x("path", { key: "b500f059a84c5223d1688410bc8378e02fc87156", d: "M11.5217 11.5217H21.9565V21.9565H11.5217V11.5217Z", fill: "#FFBA08" }));
  }
  static get style() {
    return $e;
  }
}, [2, "microsoft-logo-icon", {
  width: [1],
  height: [1]
}]);
function ue() {
  if (typeof customElements > "u")
    return;
  ["microsoft-logo-icon"].forEach((e) => {
    switch (e) {
      case "microsoft-logo-icon":
        customElements.get(e) || customElements.define(e, fe);
        break;
    }
  });
}
export {
  fe as M,
  ue as d
};
