// @nylas/web-elements/dist/components/envelope-fill.js
import { proxyCustomElement, HTMLElement, h } from '@stencil/core/internal/client';

const iconCss = ".sc-envelope-fill-icon-h{display:flex}";

const EnvelopeFillIcon = proxyCustomElement(class EnvelopeFillIcon extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.width = '16';
        this.height = '13';
    }
    render() {
        return (h("svg", { key: '9e47e0f1192e2aac3283844917161fc3852fbf32', width: "16", height: "13", viewBox: "0 0 16 13", fill: "none", xmlns: "http://www.w3.org/2000/svg" }, h("path", { key: '92fb6a5f63b69bc46da5703702df3c77ab5ed2ba', d: "M16.0001 3.373V10.5C16.0001 11.8807 14.8808 13 13.5001 13H2.50008C1.11937 13 0 11.8807 0 10.5V3.373L7.74657 7.93097C7.90304 8.02301 8.09711 8.02301 8.25359 7.93097L16.0001 3.373ZM13.5001 0C14.7871 0 15.847 0.972604 15.9849 2.22293L8.00008 6.91991L0.0152595 2.22293C0.153116 0.972604 1.21302 0 2.50008 0H13.5001Z", fill: "currentColor" })));
    }
    static get style() { return iconCss; }
}, [2, "envelope-fill-icon", {
        "width": [1],
        "height": [1]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["envelope-fill-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "envelope-fill-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, EnvelopeFillIcon);
            }
            break;
    } });
}

export { EnvelopeFillIcon as E, defineCustomElement as d };
