// @nylas/web-elements/dist/components/nylas-participants-custom-availability2.js
import { proxyCustomElement, HTMLElement, createEvent, h, Host } from '@stencil/core/internal/client';
import { R as RegisterComponent } from './register-component.js';
import { a as debug, b as instance } from './utils.js';
import { D as DEFAULT_OPEN_HOURS } from './constants.js';
import { d as defineCustomElement$b } from './add-circle.js';
import { d as defineCustomElement$a } from './chevron.js';
import { d as defineCustomElement$9 } from './close.js';
import { d as defineCustomElement$8 } from './globe.js';
import { d as defineCustomElement$7 } from './info.js';
import { d as defineCustomElement$6 } from './nylas-availability-picker2.js';
import { d as defineCustomElement$5 } from './nylas-time-window-picker2.js';
import { d as defineCustomElement$4 } from './search.js';
import { d as defineCustomElement$3 } from './select-dropdown2.js';
import { d as defineCustomElement$2 } from './toggle-switch2.js';
import { d as defineCustomElement$1 } from './toolitp-component.js';

const nylasParticipantsCustomAvailabilityCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif;width:inherit}.nylas-participants-custom-availability{width:inherit;display:flex;flex-direction:column;margin:1rem;border-radius:var(--nylas-border-radius-2x);border:1px solid var(--nylas-base-200)}.nylas-participants-custom-availability .header{padding:1rem}.nylas-participants-custom-availability .header h3{margin:0;font-size:16px;font-weight:600;line-height:20px;color:var(--nylas-base-900);text-align:left}.nylas-participants-custom-availability .header p{margin:0.25rem 0 0 0;font-size:0.875rem;font-weight:400;line-height:20px;color:var(--nylas-base-600);text-align:left}.nylas-participants-custom-availability .content .participant-container{padding-top:8px;border-top:1px solid var(--nylas-base-200)}.nylas-participants-custom-availability .content .participant-container .participant-title{display:flex;justify-content:space-between;align-items:center;padding:4px 16px 12px}.nylas-participants-custom-availability .content .participant-container .participant-title p{margin:0}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle{display:flex;gap:24px;align-items:center}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron{display:flex;align-self:center;cursor:pointer}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron:hover,.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron:active{color:var(--nylas-primary)}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron.open{transform:rotate(90deg)}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron.closed{transform:rotate(270deg)}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron.disabled{cursor:not-allowed;color:var(--nylas-base-300)}.nylas-participants-custom-availability .content .participant-container .participant-title .participant-toggle .chevron.disabled:hover{color:var(--nylas-base-300)}.nylas-participants-custom-availability .content .participant-container nylas-availability-picker::part(nap){margin:0;border:none}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasParticipantsCustomAvailability = proxyCustomElement(class NylasParticipantsCustomAvailability extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.valueChanged = createEvent(this, "valueChanged", 7);
        this.internals = this.attachInternals();
        this.selectedConfiguration = undefined;
        this.name = 'participant-custom-availability';
        this.participants = this.selectedConfiguration?.participants || [];
        this.selectedLanguage = undefined;
        this.selectedParticipants = this.setParticipants(this.participants);
    }
    elementNameChangedHandler(newValue) {
        debug('nylas-participants-custom-availability', 'elementNameChangedHandler', newValue);
        this.host.setAttribute('name', newValue);
    }
    participantsChangedHandler(newValue, oldValue) {
        debug('nylas-participants-custom-availability', 'participantsChangedHandler', newValue, oldValue);
        this.selectedParticipants = this.setParticipants(newValue);
    }
    selectedConfigurationChangedHandler(newValue, oldValue) {
        debug('nylas-participants-custom-availability', 'selectedConfigurationChangedHandler', newValue, oldValue);
        if (newValue?.participants) {
            this.selectedParticipants = this.setParticipants(newValue?.participants);
        }
    }
    selectedLanguageChangedHandler(newVal, oldVal) {
        if (newVal === oldVal)
            return;
        instance.changeLanguage(newVal);
    }
    connectedCallback() {
        debug('nylas-participants-custom-availability', 'connectedCallback');
    }
    disconnectedCallback() {
        debug('nylas-participants-custom-availability', 'disconnectedCallback');
    }
    componentWillLoad() {
        debug('nylas-participants-custom-availability', 'componentWillLoad');
    }
    componentDidLoad() {
        debug('nylas-participants-custom-availability', 'componentDidLoad');
        if (this.selectedConfiguration) {
            this.selectedParticipants = this.setParticipants(this.selectedConfiguration?.participants);
        }
        else {
            this.selectedParticipants = this.setParticipants(this.participants);
        }
    }
    nylasFormSwitchToggledHandler(event) {
        debug('nylas-participants-custom-availability', 'nylasFormSwitchToggledHandler', event.detail);
        const existingParticipant = this.selectedConfiguration?.participants.find(participant => participant.email === event.detail.name);
        const existingOpenHours = existingParticipant?.availability?.open_hours;
        const openHours = existingOpenHours || this.selectedConfiguration?.availability?.availability_rules?.default_open_hours || [];
        const timeZone = existingOpenHours && existingOpenHours.length > 0
            ? existingParticipant?.timezone
            : this.selectedConfiguration?.availability?.availability_rules?.default_open_hours?.at(0)?.timezone ||
                this.selectedConfiguration?.event_booking?.timezone ||
                window.Intl.DateTimeFormat().resolvedOptions().timeZone;
        this.selectedParticipants[event.detail.name]['isOpen'] = event.detail.checked;
        this.selectedParticipants[event.detail.name]['setCustom'] = event.detail.checked;
        this.selectedParticipants[event.detail.name]['openHours'] = event.detail.checked ? (openHours?.length > 0 ? openHours : DEFAULT_OPEN_HOURS) : [];
        this.selectedParticipants[event.detail.name]['timezone'] = event.detail.checked ? timeZone : undefined;
        this.selectedParticipants = { ...this.selectedParticipants };
        this.updateFormValue();
    }
    handleValueChanged(event) {
        debug('[nylas-editor-tabs]', 'handleValueChanged', event);
        const { name, value } = event.detail;
        if (!name.startsWith('participant-')) {
            return;
        }
        if (this.isInternalsAvailable) {
            if (!this.participantFormRef.checkValidity()) {
                this.internals.setValidity({ customError: true }, 'Please fix the overlapping time ranges.');
                return;
            }
            else {
                this.internals.setValidity({});
            }
        }
        const jsonValue = JSON.parse(value);
        const key = name.split('-')[2];
        if (!this.selectedParticipants[key])
            return;
        this.selectedParticipants[key]['openHours'] = jsonValue.openHours;
        this.selectedParticipants = { ...this.selectedParticipants };
        this.updateFormValue();
    }
    updateFormValue() {
        const participantOpenHours = {};
        const participants = this.selectedConfiguration?.participants || this.participants;
        participants.forEach(participant => {
            participantOpenHours[participant.email] = this.selectedParticipants[participant.email].openHours;
        });
        this.isInternalsAvailable && this.internals.setFormValue(JSON.stringify(participantOpenHours), this.name);
        this.valueChanged.emit({ value: JSON.stringify(participantOpenHours), name: this.name });
    }
    setParticipants(participants) {
        const selectedParticipants = {};
        participants?.forEach(participant => {
            selectedParticipants[participant.email] = {
                isOpen: false,
                setCustom: Boolean(participant.availability?.open_hours),
                openHours: participant.availability?.open_hours || [],
                name: participant.name || participant.email,
                availability: participant.availability,
                timezone: participant.timezone,
            };
        });
        return selectedParticipants;
    }
    get isInternalsAvailable() {
        return this.internals !== undefined && typeof this.internals.setFormValue === 'function' && typeof this.internals.setValidity === 'function';
    }
    render() {
        return (h(Host, { key: 'ff6765c725b2c24a63e131ac7d4d52e0572c8f52' }, h("div", { key: 'dae18b83f3d342db5e3c058f9c782aa280811637', class: "nylas-participants-custom-availability", part: "npca" }, h("div", { key: 'a76dcaa11e0e2eb6b56aec6b58eb7c12d910e571', class: "header", part: "npca__header" }, h("h3", { key: 'd2c05942fca786c6ae06aa912dd81eb84607f27b' }, instance.t('nylasParticipantCustomAvailability.headerTitle')), h("p", { key: '3d7b2e15f6e5c9a528886c5e3b8702bfee33a36e' }, instance.t('nylasParticipantCustomAvailability.headerSubTitle'))), h("div", { key: '46520d59984a8dcea6e28fa72a0da64c6468b5ca', class: "content", part: "npca__content" }, h("form", { key: 'ce08bb8a25d1133ddfeb765c98088dbc3e3a2ec9', ref: el => (this.participantFormRef = el) }, Object.keys(this.selectedParticipants).map((key, index) => {
            const participant = this.selectedParticipants[key];
            const participantOpenHours = participant.availability?.open_hours || [];
            if (!participant || !participant.name || !participant.availability)
                return;
            return (h("div", { class: "participant-container", part: "npca__participant-container", key: `participant-conatiner-${index}` }, h("div", { class: "participant-title", part: "npca__participant-title" }, h("p", null, instance.t('nylasParticipantCustomAvailability.participantOpenHoursTitle', { participantName: participant.name })), h("div", { class: "participant-toggle", part: "npca__participant-toggle--container" }, h("toggle-switch", { exportparts: "ts_label: npca__toggle-label, ts_input: npca_toggle-input, ts_slider: npca_toggle-slider", name: key, checked: !!participant?.openHours && participant?.openHours.length > 0 }), h("span", { class: `chevron ${participant.isOpen ? 'open' : 'closed'} ${participant.setCustom ? '' : 'disabled'}`, onClick: () => {
                    if (!participant.setCustom)
                        return;
                    this.selectedParticipants[key].isOpen = !participant.isOpen;
                    this.selectedParticipants = { ...this.selectedParticipants };
                } }, h("chevron-icon", { width: "24", height: "24" })))), participant.isOpen && (h("nylas-availability-picker", { key: index, name: `participant-${index}-${key}`, openHours: participantOpenHours, defaultTimezone: participant.timezone, hideHeader: true }))));
        }))))));
    }
    static get formAssociated() { return true; }
    get host() { return this; }
    static get watchers() { return {
        "name": ["elementNameChangedHandler"],
        "participants": ["participantsChangedHandler"],
        "selectedConfiguration": ["selectedConfigurationChangedHandler"],
        "selectedLanguage": ["selectedLanguageChangedHandler"]
    }; }
    static get style() { return nylasParticipantsCustomAvailabilityCss; }
}, [65, "nylas-participants-custom-availability", {
        "selectedConfiguration": [16],
        "name": [1],
        "participants": [16],
        "selectedLanguage": [1, "selected-language"],
        "selectedParticipants": [32]
    }, [[0, "nylasFormSwitchToggled", "nylasFormSwitchToggledHandler"], [0, "valueChanged", "handleValueChanged"]], {
        "name": ["elementNameChangedHandler"],
        "participants": ["participantsChangedHandler"],
        "selectedConfiguration": ["selectedConfigurationChangedHandler"],
        "selectedLanguage": ["selectedLanguageChangedHandler"]
    }]);
__decorate([
    RegisterComponent({
        name: 'nylas-participants-custom-availability',
        stateToProps: new Map([
            ['schedulerConfig.selectedConfiguration', 'selectedConfiguration'],
            ['schedulerConfig.selectedLanguage', 'selectedLanguage'],
        ]),
        eventToProps: {},
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasParticipantsCustomAvailability.prototype, "render", null);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nylas-participants-custom-availability", "add-circle-icon", "chevron-icon", "close-icon", "globe-icon", "info-icon", "nylas-availability-picker", "nylas-time-window-picker", "search-icon", "select-dropdown", "toggle-switch", "tooltip-component"];
    components.forEach(tagName => { switch (tagName) {
        case "nylas-participants-custom-availability":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NylasParticipantsCustomAvailability);
            }
            break;
        case "add-circle-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "chevron-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "close-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "globe-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "info-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "nylas-availability-picker":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "nylas-time-window-picker":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "search-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "select-dropdown":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "toggle-switch":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "tooltip-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { NylasParticipantsCustomAvailability as N, defineCustomElement as d };
