// @nylas/web-elements/dist/components/nylas-feedback-form2.js
import { proxyCustomElement, HTMLElement, createEvent, h, Host } from '@stencil/core/internal/client';
import { R as RegisterComponent } from './register-component.js';
import { a as debug, b as instance } from './utils.js';
import { d as defineCustomElement$6 } from './button-component2.js';
import { d as defineCustomElement$5 } from './close.js';
import { d as defineCustomElement$4 } from './info.js';
import { d as defineCustomElement$3 } from './loading.js';
import { d as defineCustomElement$2 } from './textarea-component2.js';
import { d as defineCustomElement$1 } from './toolitp-component.js';

const nylasFeedbackFormCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.card-wrapper{background:var(--nylas-base-0);display:flex;flex-direction:column;font-family:var(--nylas-font-family);background-color:var(--nylas-base-0);color:var(--nylas-base-900);border-radius:var(--nylas-border-radius-3x);position:relative;padding:32px;width:562px;box-shadow:0px 1px 4px rgba(0, 0, 0, 0.1), 0px 3px 6px rgba(0, 0, 0, 0.06)}@media screen and (max-width: 768px){.card-wrapper{width:100%;border-radius:0px}}.header{overflow-wrap:anywhere;display:flex;flex-direction:column;font-size:16px;font-weight:400}.header h2{color:var(--nylas-base-900);font-size:24px;font-weight:600;text-align:left;margin:0}.description.title-desc{margin-bottom:2rem}.description.footer-desc{margin-top:2rem}.description.footer-desc a{color:var(--nylas-primary);text-decoration:none}.description p{margin:0;font-size:18px;color:var(--nylas-base-600)}.footer{margin-top:1rem;padding-top:1rem;display:flex;justify-content:flex-end;gap:0.5rem;box-sizing:border-box;background-color:var(--nylas-base-25);width:100%;border-radius:0 0 var(--nylas-border-radius-2x) var(--nylas-border-radius-2x)}.close-button{width:100%;display:flex;justify-content:flex-end}.close-button button-component button{padding:0;height:auto}.close-button button-component button:hover{background-color:transparent !important}button-component{--dot-color:var(--nylas-base-700)}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasFeedbackForm = proxyCustomElement(class NylasFeedbackForm extends HTMLElement {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.feedbackModalClosed = createEvent(this, "feedbackModalClosed", 7);
        this.feedbackSubmitted = createEvent(this, "feedbackSubmitted", 7);
        this.triggerValidation = createEvent(this, "triggerValidation", 7);
        this.handleConfirmBookingButtonClicked = () => {
            const validationEvent = this.triggerValidation.emit();
            if (validationEvent.defaultPrevented) {
                return;
            }
            this.feedbackSubmitted.emit({ feedback: this.feedbackMessage });
        };
        this.themeConfig = undefined;
        this.feedbackMessage = '';
    }
    connectedCallback() { }
    disconnectedCallback() { }
    async componentWillLoad() {
        debug(`[nylas-feedback-form] Component will load`);
    }
    async componentDidLoad() {
        debug(`[nylas-feedback-form] Component did load`);
        this.applyThemeConfig(this.themeConfig);
    }
    applyThemeConfig(themeConfig) {
        if (themeConfig) {
            for (const [key, value] of Object.entries(themeConfig)) {
                this.host.style.setProperty(`${key}`, value);
            }
        }
    }
    handleFormInputChanged(event) {
        debug(`[nylas-feedback-form] Form input changed`);
        if (!event.detail.value) {
            return;
        }
        this.triggerValidation.emit();
        if (event.detail.name === 'feedback') {
            this.feedbackMessage = event.detail.value;
        }
    }
    closeModal(e) {
        e.preventDefault();
        e.stopImmediatePropagation();
        this.feedbackModalClosed.emit();
    }
    render() {
        return (h(Host, { key: 'd9e46e03000d836deb146a105ad5af7314fc93d3', part: "nfbf" }, h("div", { key: '6924101eb302c203949378720eea5a6c0ecb34a3', class: "card-wrapper", part: "nfbfc__card" }, h("div", { key: '29ad795443767409c2d9812f55b20678d79188ba', class: "close-button" }, h("button-component", { key: '6a2c2454351b7d1fdff44e03e5557da13eb75ade', tooltip: "Close", id: "close", variant: 'invisible', onClick: e => this.closeModal(e), part: "nfbf__close-button" }, h("close-icon", { key: 'cad67f709930a3d4f439de83f15d2cc4bf48ded5' }))), h("div", { key: '4060b186ead7f5dceb82275ca7c5f3daae6603e8', class: "header" }, h("h2", { key: 'dc5d83a6e9d0c8898011d2f490940bc7e9bcce8b', slot: "card-title", part: "nfbf__title" }, instance.t('feedbackFormTitle'))), h("div", { key: '8f5519ff02c341fe26613e640b76556b6edcd4ac', class: "description title-desc" }, h("p", { key: '2bd0aa700560b3381ab3902be2da9bb6765ca1bb' }, instance.t('feedbackFormDescription'))), h("textarea-component", { key: '959082460c4ed316ba7d74e8ce2a95225b53deb4', label: instance.t('feedbackMessageLabel'), required: true, maxLength: 500, part: "nfbf__description", name: "feedback", placeholder: instance.t('feedbackMessagePlaceholder') }), h("div", { key: 'b0b085c8d8c7d5855ade8a9e91ed172e710f5b7d', class: 'description footer-desc' }, h("p", { key: 'd64ed3e021f0590ee0d792f8d37fcf860332d50d' }, "By sending this report, you agree to Nylas\u2019 access to the data necessary to respond to this issue, subject to Nylas\u2019", ' ', h("a", { key: '7dc7c199ee9f4abd84274cd5bfca904f6aa8d19c', href: "https://www.nylas.com/privacy-policy/", target: "_blank", rel: "nofollow noreferrer noopener" }, "Privacy Policy"), ".")), h("div", { key: 'aa907c4c9462c5218b279b0cb6495aa7177a68aa', class: {
                footer: true,
            } }, h("button-component", { key: '049883425d2a7657b84dfd9cf1829b61771c23bc', variant: 'invisible', onClick: e => this.closeModal(e), part: "nfbf__button-outline nfbf__cancel-cta" }, `${instance.t('feedbackModalCloseButton')}`), h("button-component", { key: '5a97be9f193e984237835f1707af9ce541a4f063', variant: 'primary', onClick: this.handleConfirmBookingButtonClicked, part: "nfbf__button-outline nfbf__reschedule-cta" }, `${instance.t('feedbackModalSubmitButton')}`)))));
    }
    get host() { return this; }
    static get style() { return nylasFeedbackFormCss; }
}, [1, "nylas-feedback-form", {
        "themeConfig": [8, "theme-config"],
        "feedbackMessage": [32]
    }, [[4, "nylasFormInputChanged", "handleFormInputChanged"]]]);
__decorate([
    RegisterComponent({
        name: 'nylas-feedback-form',
        stateToProps: new Map([]),
        eventToProps: {
            feedbackSubmitted: async (_event, _nylasSchedulerConnector) => {
            },
        },
        localPropsToProp: new Map([['themeConfig', 'themeConfig']]),
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasFeedbackForm.prototype, "render", null);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["nylas-feedback-form", "button-component", "close-icon", "info-icon", "loading-icon", "textarea-component", "tooltip-component"];
    components.forEach(tagName => { switch (tagName) {
        case "nylas-feedback-form":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, NylasFeedbackForm);
            }
            break;
        case "button-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "close-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "info-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "loading-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "textarea-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "tooltip-component":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { NylasFeedbackForm as N, defineCustomElement as d };
