// @nylas/web-elements/dist/esm/nylas-locale-switch.entry.js
import { r as registerInstance, a as createEvent, h, i as Host, e as getElement } from './index-b5cb18eb.js';
import { T as TIMEZONE_MAP, a as LANGUAGE_MAP } from './constants-335f5e87.js';
import { R as RegisterComponent } from './register-component-6db1d72d.js';
import { a as debug, x as getTimezoneOffset } from './utils-14cfef23.js';

const nylasLocaleSwitchCss = ":host{display:block;--nylas-primary:#2563eb;--nylas-error:#cc4841;--nylas-error-pressed:#992222;--nylas-wraning:#f06c00;--nylas-success:#16a392;--nylas-info:#2b8fc2;--nylas-base-0:#ffffff;--nylas-base-25:#fcfcfd;--nylas-base-50:#f8f9fc;--nylas-base-100:#eaecf5;--nylas-base-200:#d5d9eb;--nylas-base-300:#b3b8d8;--nylas-base-400:#717bbc;--nylas-base-500:#4e5ba6;--nylas-base-600:#3e4784;--nylas-base-700:#263f72;--nylas-base-800:#293056;--nylas-base-900:#101323;--nylas-base-950:#0e101b;--nylas-border-radius:0.25rem;--nylas-border-radius-2x:0.5rem;--nylas-border-radius-3x:0.75rem;--nylas-font-family:\"Inter\", sans-serif}.nylas-locale-switch{display:flex;flex-direction:row;justify-content:space-between;align-items:center;border-left:none;border-right:none;color:var(--nylas-base-50);font-size:14px;font-weight:600;cursor:pointer;transition:all 0.2s ease-in-out;font-family:inherit}@media screen and (max-width: 768px){.nylas-locale-switch{font-size:16px}}.select-wrapper{display:flex;align-items:center;gap:0.5rem}select{border:none;background:transparent;cursor:pointer;outline:none;background-color:transparent;margin:0 0 1px;padding:4px;vertical-align:middle}select#timezone{width:140px}select#language{width:80px}span.timezone-label{display:flex;width:-webkit-fill-available;justify-content:space-between}select-dropdown{width:auto}select-dropdown::part(sd_dropdown-button){border:none;padding:10px}select-dropdown::part(sd_dropdown-content){max-width:306px}";

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (undefined && undefined.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
const NylasLocaleSwitch = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.timezoneChanged = createEvent(this, "timezoneChanged", 7);
        this.languageChanged = createEvent(this, "languageChanged", 7);
        this.themeConfig = undefined;
        this.bookingInfo = undefined;
        this.selectedTimezone = this.bookingInfo?.timezone || Intl.DateTimeFormat().resolvedOptions().timeZone;
        this.selectedLanguage = this.bookingInfo?.language || navigator.language;
    }
    themeConfigChanged(newThemeConfig) {
        this.applyThemeConfig(newThemeConfig);
    }
    bookingInfoChangedHandler(newValue) {
        if (newValue && newValue?.timezone) {
            this.changeTimezone(newValue.timezone);
        }
        if (newValue && newValue?.language) {
            this.changeLanguage(newValue.language);
        }
    }
    connectedCallback() { }
    disconnectedCallback() { }
    componentWillLoad() { }
    componentDidLoad() {
        debug(`[nylas-locale-switch] Component did load`);
        this.applyThemeConfig(this.themeConfig);
        if (this.bookingInfo && this.bookingInfo?.timezone) {
            this.changeTimezone(this.bookingInfo.timezone);
        }
        if (this.bookingInfo && this.bookingInfo?.language) {
            this.changeLanguage(this.bookingInfo.language);
        }
    }
    applyThemeConfig(themeConfig) {
        if (themeConfig) {
            for (const [key, value] of Object.entries(themeConfig)) {
                this.host.style.setProperty(`${key}`, value);
            }
        }
    }
    changeTimezone(timezone) {
        this.selectedTimezone = timezone;
        this.timezoneChanged.emit(timezone);
    }
    changeLanguage(language) {
        this.selectedLanguage = language;
        this.languageChanged.emit(language);
    }
    nylasFormDropdownChangedHandler(event) {
        const { name, value } = event.detail;
        if (name === 'timezone') {
            this.changeTimezone(value);
        }
        else if (name === 'language') {
            this.changeLanguage(value);
        }
    }
    getTimezoneLabelHTML(timezone) {
        const offset = getTimezoneOffset(timezone);
        return (h("span", { class: "timezone-label", style: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'space-between',
                width: '-webkit-fill-available',
            } }, TIMEZONE_MAP[timezone], ' ', h("span", { class: "timezone-offset", style: {
                fontSize: '14px',
                fontWeight: '400',
                color: 'var(--nylas-base-500)',
            } }, offset)));
    }
    render() {
        const timezoneOptions = Object.keys(TIMEZONE_MAP).map(key => ({
            labelHTML: this.getTimezoneLabelHTML(key),
            label: TIMEZONE_MAP[key],
            value: key,
        }));
        const languageOptions = Object.keys(LANGUAGE_MAP).map(key => ({
            label: LANGUAGE_MAP[key],
            value: key,
        }));
        return (h(Host, { key: '00af320c95ef648f69c9ebde6d741df848190059' }, h("div", { key: '56998ac07ecebb57f763852ea1c46b11328d4d15', class: "nylas-locale-switch", part: "nls" }, h("div", { key: 'ba294660f0716060787c9034010ad26b133338f5', class: {
                'select-wrapper': true,
                'timezone': true,
            }, part: "nls__timezone" }, h("select-dropdown", { key: 'e3d1a121cb6230c32ad2e4cf9aeffa4395bd635c', name: "timezone", options: timezoneOptions, defaultSelectedOption: timezoneOptions.find(op => op.value == this.selectedTimezone), exportparts: "sd_dropdown: nls__timezone-dropdown, sd_dropdown-button: nls__timezone-drop-button, sd_dropdown-button-selected-label: nls__timezone-drop-button-selected-label, sd_dropdown-content: nls__timezone-drop-content, sd_dropdown_label: nls__timezone-drop-label" }, h("span", { key: 'cc7bb4f138747aab7c5cf673c1b7b3ed09fb9de5', slot: "select-icon" }, h("globe-icon", { key: '717894778dc82c415eab8f764c2c18afc3636ee3', width: "20", height: "20" })))), h("div", { key: 'd0a64b3910138c1764e17e2be841dbd3fb746f3d', class: {
                'select-wrapper': true,
                'language': true,
            }, part: "nls__language" }, h("select-dropdown", { key: '86b417f7b62a0e6c74383e3315e1c6dfdfa09868', name: "language", options: languageOptions, defaultSelectedOption: languageOptions.find(lang => lang.value == this.selectedLanguage), exportparts: "sd_dropdown: nls__language-dropdown, sd_dropdown-button: nls__language-drop-button, sd_dropdown-content: nls__language-drop-content, sd_dropdown_label: nls__language-drop-label" }, h("span", { key: '0f1c14fae74a2543fcc57358c507d1288b02ba9c', slot: "select-icon" }, h("translate-icon", { key: 'f5a373fff866cf48f39e5bb3b2aa0f6567044433', width: "20", height: "20" })))))));
    }
    get host() { return getElement(this); }
    static get watchers() { return {
        "themeConfig": ["themeConfigChanged"],
        "bookingInfo": ["bookingInfoChangedHandler"]
    }; }
};
__decorate([
    RegisterComponent({
        name: 'nylas-locale-switch',
        stateToProps: new Map([
            ['scheduler.selectedLanguage', 'selectedLanguage'],
            ['scheduler.selectedTimezone', 'selectedTimezone'],
            ['scheduler.bookingInfo', 'bookingInfo'],
            ['scheduler.themeConfig', 'themeConfig'],
        ]),
        eventToProps: {
            timezoneChanged: async (event, nylasSchedulerConnector) => {
                debug('nylas-locale-switch', 'timezoneChanged', event.detail);
                nylasSchedulerConnector.scheduler.selectTimezone(event.detail);
            },
            languageChanged: async (event, nylasSchedulerConnector) => {
                debug('nylas-locale-switch', 'languageChanged', event.detail);
                nylasSchedulerConnector.scheduler.selectLanguage(event.detail);
            },
        },
        fireRegisterEvent: true,
    }),
    __metadata("design:type", Function),
    __metadata("design:paramtypes", []),
    __metadata("design:returntype", void 0)
], NylasLocaleSwitch.prototype, "render", null);
NylasLocaleSwitch.style = nylasLocaleSwitchCss;

export { NylasLocaleSwitch as nylas_locale_switch };
